<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Shixiang Wang">
<meta name="author" content="Yankun Zhao">
<meta name="dcterms.date" content="2024-03-16">

<title>Methods – ImmunoFusion</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-7fbd9fe57f4d935bbf0703f32dfb278c.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">ImmunoFusion</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html"> 
<span class="menu-text">Tutorial</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./methods.html" aria-current="page"> 
<span class="menu-text">Supplementary Methods</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./about.html"> 
<span class="menu-text">About</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active" data-toc-expanded="2">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#gene-fusions-identification-and-differential-expression-analysis" id="toc-gene-fusions-identification-and-differential-expression-analysis" class="nav-link active" data-scroll-target="#gene-fusions-identification-and-differential-expression-analysis">Gene Fusions identification and differential expression analysis</a></li>
  <li><a href="#response-and-response2" id="toc-response-and-response2" class="nav-link" data-scroll-target="#response-and-response2">Response and Response2</a></li>
  <li><a href="#cancer-type-abbreviation" id="toc-cancer-type-abbreviation" class="nav-link" data-scroll-target="#cancer-type-abbreviation">Cancer Type Abbreviation</a></li>
  <li><a href="#immunofusion-cohort-naming-maping" id="toc-immunofusion-cohort-naming-maping" class="nav-link" data-scroll-target="#immunofusion-cohort-naming-maping">ImmunoFusion cohort naming maping</a></li>
  <li><a href="#clinical-data-preprocessing" id="toc-clinical-data-preprocessing" class="nav-link" data-scroll-target="#clinical-data-preprocessing">Clinical data preprocessing</a>
  <ul class="collapse">
  <li><a href="#tcga" id="toc-tcga" class="nav-link" data-scroll-target="#tcga">TCGA</a></li>
  <li><a href="#cptac" id="toc-cptac" class="nav-link" data-scroll-target="#cptac">CPTAC</a></li>
  <li><a href="#target" id="toc-target" class="nav-link" data-scroll-target="#target">TARGET</a></li>
  </ul></li>
  <li><a href="#tcgatargetcptac-clinicaliobrtmegene-data-divide-and-anotate-preprocessing" id="toc-tcgatargetcptac-clinicaliobrtmegene-data-divide-and-anotate-preprocessing" class="nav-link" data-scroll-target="#tcgatargetcptac-clinicaliobrtmegene-data-divide-and-anotate-preprocessing">TCGA/TARGET/CPTAC Clinical/IOBR/TME/GENE data divide and anotate preprocessing</a>
  <ul class="collapse">
  <li><a href="#target-1" id="toc-target-1" class="nav-link" data-scroll-target="#target-1">TARGET</a></li>
  <li><a href="#cptac-1" id="toc-cptac-1" class="nav-link" data-scroll-target="#cptac-1">CPTAC</a></li>
  <li><a href="#tcga-1" id="toc-tcga-1" class="nav-link" data-scroll-target="#tcga-1">TCGA</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Methods</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Authors</div>
    <div class="quarto-title-meta-contents">
             <p>Shixiang Wang </p>
             <p>Yankun Zhao </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">March 16, 2024</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="gene-fusions-identification-and-differential-expression-analysis" class="level2">
<h2 class="anchored" data-anchor-id="gene-fusions-identification-and-differential-expression-analysis">Gene Fusions identification and differential expression analysis</h2>
<p>We employed two Gene Fusions(GFs) detection tools, including arriba, STAR-Fusion, to identify, parse, and annotate GFs junctions within each sample, using the human genome version hg38 as the reference. In our pursuit of biologically robust GFs identification within each cohort, we undertook an approach metafusion that amalgamated results from two GFs detection tools. Firstly, we exclusively retained Fusions located on chromosomes 1 to 22, as well as the X and Y chromosomes. Secondly, GFs were mandated to exhibit genomic overlap with gene regions as defined in the reference genome annotation file, specifically “gencode.v34.annotation.gtf.”. Thirdly to ensure the precision and recall in GFs prediction with different tools, one can either rely on consensus predictions or all predctions and choosing junction-crossing reads and reads mates which flank (span) the breakpoint. Fourthly, we offer FusionAnnotator to identify and prioritize GFs that have been previously reported in cancer or normal samples. Lastly, the detected data can be divided into Cis-SAGes and other categories to further explore the debate on what should be considered true fusion RNAs. This meticulous methodological approach was instrumental in ensuring the acquisition of biologically sound and reliable GFs datasets, thereby fortifying the integrity of our analysis within each cohort. This analysis facilitated comparisons between patient groups, such as those who responded to checkpoint immunotherapy versus non-responders, as well as between samples collected before and during ICB treatment.</p>
</section>
<section id="response-and-response2" class="level2">
<h2 class="anchored" data-anchor-id="response-and-response2">Response and Response2</h2>
<p>We gathered and processed patient ‘Response’ data from various cohorts. However, due to inconsistencies among these cohorts, such as the usage of different terms like CR, PR, SD, PD, R, NR, DCB, and NDB, we harmonized the response variable based on the timing of treatment and treatment response. As a result, we introduced a new variable called ‘Response2’ with four distinct categories: Pre-R, Pre-NR, On-R, and On-NR. In cases where response data was not available, we assigned the category NE.</p>
</section>
<section id="cancer-type-abbreviation" class="level2">
<h2 class="anchored" data-anchor-id="cancer-type-abbreviation">Cancer Type Abbreviation</h2>
<table class="caption-top table">
<colgroup>
<col style="width: 23%">
<col style="width: 76%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Abbr.</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">BLCA</td>
<td style="text-align: left;">Bladder urothelial carcinoma</td>
</tr>
<tr class="even">
<td style="text-align: left;">SKCM</td>
<td style="text-align: left;">Skin cutaneous melanoma</td>
</tr>
<tr class="odd">
<td style="text-align: left;">KIRC &amp; CCRCC</td>
<td style="text-align: left;">Kidney renal clear cell carcinoma</td>
</tr>
<tr class="even">
<td style="text-align: left;">HNSC &amp; HNSCC</td>
<td style="text-align: left;">Head and neck squamous cell carcinoma</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NSCLC</td>
<td style="text-align: left;">Non-small cell lung cancer, including lung adenocarcinoma and lung squamous cell carcinoma</td>
</tr>
<tr class="even">
<td style="text-align: left;">SCLC</td>
<td style="text-align: left;">Small cell lung cancer</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SARC</td>
<td style="text-align: left;">Sarcoma</td>
</tr>
<tr class="even">
<td style="text-align: left;">BRCA</td>
<td style="text-align: left;">Breast cancer</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GBM</td>
<td style="text-align: left;">Glioblastoma Multiforme</td>
</tr>
<tr class="even">
<td style="text-align: left;">STAD</td>
<td style="text-align: left;">Stomach Adenocarcinoma</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SGC</td>
<td style="text-align: left;">Salivary gland cancer</td>
</tr>
<tr class="even">
<td style="text-align: left;">LUSC</td>
<td style="text-align: left;">Lung Squamous Cell Carcinoma</td>
</tr>
<tr class="odd">
<td style="text-align: left;">MESO</td>
<td style="text-align: left;">Mesothelioma</td>
</tr>
<tr class="even">
<td style="text-align: left;">PCPG</td>
<td style="text-align: left;">Pheochromocytoma and Paraganglioma</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LIHC</td>
<td style="text-align: left;">Liver Hepatocellular Carcinoma</td>
</tr>
<tr class="even">
<td style="text-align: left;">PRAD</td>
<td style="text-align: left;">Prostate Adenocarcinoma</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ACC</td>
<td style="text-align: left;">Adrenocortical Carcinoma</td>
</tr>
<tr class="even">
<td style="text-align: left;">TGCT</td>
<td style="text-align: left;">Testicular Germ Cell Tumors</td>
</tr>
<tr class="odd">
<td style="text-align: left;">PAAD</td>
<td style="text-align: left;">Pancreatic Adenocarcinoma</td>
</tr>
<tr class="even">
<td style="text-align: left;">UCEC</td>
<td style="text-align: left;">Uterine Corpus Endometrial Carcinoma</td>
</tr>
<tr class="odd">
<td style="text-align: left;">THCA</td>
<td style="text-align: left;">Thyroid carcinoma</td>
</tr>
<tr class="even">
<td style="text-align: left;">CESC</td>
<td style="text-align: left;">Cervical squamous cell carcinoma and endocervical adenocarcinoma</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ESCA</td>
<td style="text-align: left;">Esophageal carcinoma</td>
</tr>
<tr class="even">
<td style="text-align: left;">READ</td>
<td style="text-align: left;">Rectum adenocarcinoma</td>
</tr>
<tr class="odd">
<td style="text-align: left;">OV</td>
<td style="text-align: left;">Ovarian Serous Cystadenocarcinoma</td>
</tr>
<tr class="even">
<td style="text-align: left;">CHOL</td>
<td style="text-align: left;">Cholangiocarcinoma</td>
</tr>
<tr class="odd">
<td style="text-align: left;">UCS</td>
<td style="text-align: left;">Uterine Carcinosarcoma</td>
</tr>
<tr class="even">
<td style="text-align: left;">LUAD</td>
<td style="text-align: left;">Lung Adenocarcinoma</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LGG</td>
<td style="text-align: left;">Brain Lower Grade Glioma</td>
</tr>
<tr class="even">
<td style="text-align: left;">HGSC</td>
<td style="text-align: left;">High-Grade Serous Carcinoma</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LSCC &amp; LUSC</td>
<td style="text-align: left;">Lung Squamous Cell Carcinoma</td>
</tr>
<tr class="even">
<td style="text-align: left;">PDAC</td>
<td style="text-align: left;">Pancreatic Ductal Adenocarcinoma</td>
</tr>
<tr class="odd">
<td style="text-align: left;">THYM</td>
<td style="text-align: left;">Thymoma</td>
</tr>
<tr class="even">
<td style="text-align: left;">KICH</td>
<td style="text-align: left;">Kidney Chromophobe</td>
</tr>
<tr class="odd">
<td style="text-align: left;">KIRP</td>
<td style="text-align: left;">Kidney Renal Papillary Cell Carcinoma</td>
</tr>
<tr class="even">
<td style="text-align: left;">COAD</td>
<td style="text-align: left;">Colon Adenocarcinoma</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LAML &amp; AML</td>
<td style="text-align: left;">Acute Myeloid Leukemia</td>
</tr>
<tr class="even">
<td style="text-align: left;">NBL</td>
<td style="text-align: left;">Neuroblastoma</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GNB</td>
<td style="text-align: left;">Ganglioneuroblastoma</td>
</tr>
</tbody>
</table>
</section>
<section id="immunofusion-cohort-naming-maping" class="level2">
<h2 class="anchored" data-anchor-id="immunofusion-cohort-naming-maping">ImmunoFusion cohort naming maping</h2>
<table class="caption-top table">
<colgroup>
<col style="width: 14%">
<col style="width: 24%">
<col style="width: 22%">
<col style="width: 38%">
</colgroup>
<thead>
<tr class="header">
<th>Cancer</th>
<th>Sample_type</th>
<th>Project_id</th>
<th>Immunofusion-cohort</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>BRCA</td>
<td>Primary Tumor</td>
<td>CPTAC-2</td>
<td>CPTAC-BRCA</td>
</tr>
<tr class="even">
<td>COAD</td>
<td>Primary Tumor</td>
<td>CPTAC-2</td>
<td>CPTAC-COAD</td>
</tr>
<tr class="odd">
<td>HGSC</td>
<td>Primary Tumor</td>
<td>CPTAC-2</td>
<td>CPTAC-HGSC</td>
</tr>
<tr class="even">
<td>LSCC</td>
<td>Primary Tumor</td>
<td>CPTAC-3</td>
<td>CPTAC-LUSC</td>
</tr>
<tr class="odd">
<td>UCEC</td>
<td>Primary Tumor</td>
<td>CPTAC-3</td>
<td>CPTAC-UCEC</td>
</tr>
<tr class="even">
<td>LUAD</td>
<td>Primary Tumor</td>
<td>CPTAC-3</td>
<td>CPTAC-LUAD</td>
</tr>
<tr class="odd">
<td>GBM</td>
<td>Primary Tumor</td>
<td>CPTAC-3</td>
<td>CPTAC-GBM</td>
</tr>
<tr class="even">
<td>HNSCC</td>
<td>Primary Tumor</td>
<td>CPTAC-3</td>
<td>CPTAC-HNSC</td>
</tr>
<tr class="odd">
<td>CCRCC</td>
<td>Primary Tumor</td>
<td>CPTAC-3</td>
<td>CPTAC-KIRC</td>
</tr>
<tr class="even">
<td>PDAC</td>
<td>Primary Tumor</td>
<td>CPTAC-3</td>
<td>CPTAC-PDAC</td>
</tr>
<tr class="odd">
<td>PDAC</td>
<td>Solid Tissue Normal</td>
<td>CPTAC-3</td>
<td>CPTAC-NORMAL</td>
</tr>
<tr class="even">
<td>LUAD</td>
<td>Solid Tissue Normal</td>
<td>CPTAC-3</td>
<td>CPTAC-NORMAL</td>
</tr>
<tr class="odd">
<td>UCEC</td>
<td>Solid Tissue Normal</td>
<td>CPTAC-3</td>
<td>CPTAC-NORMAL</td>
</tr>
<tr class="even">
<td>CCRCC</td>
<td>Solid Tissue Normal</td>
<td>CPTAC-3</td>
<td>CPTAC-NORMAL</td>
</tr>
<tr class="odd">
<td>HNSCC</td>
<td>Solid Tissue Normal</td>
<td>CPTAC-3</td>
<td>CPTAC-NORMAL</td>
</tr>
<tr class="even">
<td>LSCC</td>
<td>Solid Tissue Normal</td>
<td>CPTAC-3</td>
<td>CPTAC-NORMAL</td>
</tr>
<tr class="odd">
<td>GTEX-Brain</td>
<td>Solid Tissue Normal</td>
<td>CPTAC-3</td>
<td>CPTAC-NORMAL</td>
</tr>
<tr class="even">
<td>–</td>
<td>Primary Blood Derived Cancer - Peripheral Blood/Recurrent Blood Derived Cancer - Peripheral Blood/Next Generation Cancer Model/Recurrent Blood Derived Cancer - Bone Marrow/Primary Blood Derived Cancer - Bone Marrow</td>
<td>TARGET-AML</td>
<td>TARGET-LAML</td>
</tr>
<tr class="odd">
<td>Acute myeloid leukemia, NOS</td>
<td>Primary Blood Derived Cancer - Bone Marrow/Recurrent Blood Derived Cancer - Bone Marrow/Primary Blood Derived Cancer - Peripheral Blood/ Blood Derived Cancer - Bone Marrow/Post-treatment/Recurrent Blood Derived Cancer - Peripheral Blood/ Blood Derived Cancer - Peripheral Blood</td>
<td>TARGET-AML</td>
<td>TARGET-LAML</td>
</tr>
<tr class="even">
<td>Neuroblastoma, NOS</td>
<td>Primary Tumor/Recurrent Tumor/Recurrent Blood Derived Cancer - Bone Marrow</td>
<td>TARGET-NBL</td>
<td>TARGET-NBL</td>
</tr>
<tr class="odd">
<td>Ganglioneuroblastoma</td>
<td>Primary Tumor/Recurrent Tumor</td>
<td>TARGET-NBL</td>
<td>TARGET-GNB</td>
</tr>
<tr class="even">
<td>–</td>
<td>Cell Lines</td>
<td>TARGET-AML</td>
<td>TARGET-CELL</td>
</tr>
<tr class="odd">
<td>–</td>
<td>Bone Marrow Normal</td>
<td>TARGET-AML</td>
<td>TARGET-NORMAL</td>
</tr>
<tr class="even">
<td>Acute myeloid leukemia, NOS</td>
<td>Blood Derived Normal/Bone Marrow Normal</td>
<td>TARGET-AML</td>
<td>TARGET-NORMAL</td>
</tr>
<tr class="odd">
<td>ACC</td>
<td>Primary Tumor</td>
<td>TCGA-ACC</td>
<td>TCGA-ACC</td>
</tr>
<tr class="even">
<td>BLCA</td>
<td>Primary Tumor</td>
<td>TCGA-BLCA</td>
<td>TCGA-BLCA</td>
</tr>
<tr class="odd">
<td>BRCA</td>
<td>Primary Tumor</td>
<td>TCGA-BRCA</td>
<td>TCGA-BRCA</td>
</tr>
<tr class="even">
<td>BRCA</td>
<td>Metastatic</td>
<td>TCGA-BRCA</td>
<td>TCGA-BRCA</td>
</tr>
<tr class="odd">
<td>CESC</td>
<td>Primary Tumor</td>
<td>TCGA-CESC</td>
<td>TCGA-CESC</td>
</tr>
<tr class="even">
<td>CESC</td>
<td>Metastatic</td>
<td>TCGA-CESC</td>
<td>TCGA-CESC</td>
</tr>
<tr class="odd">
<td>CHOL</td>
<td>Primary Tumor</td>
<td>TCGA-CHOL</td>
<td>TCGA-CHOL</td>
</tr>
<tr class="even">
<td>COAD</td>
<td>Primary Tumor</td>
<td>TCGA-COAD</td>
<td>TCGA-COAD</td>
</tr>
<tr class="odd">
<td>COAD</td>
<td>Recurrent Tumor</td>
<td>TCGA-COAD</td>
<td>TCGA-COAD</td>
</tr>
<tr class="even">
<td>COAD</td>
<td>Metastatic</td>
<td>TCGA-COAD</td>
<td>TCGA-COAD</td>
</tr>
<tr class="odd">
<td>DLBC</td>
<td>Primary Tumor</td>
<td>TCGA-DLBC</td>
<td>TCGA-DLBC</td>
</tr>
<tr class="even">
<td>ESCA</td>
<td>Primary Tumor</td>
<td>TCGA-ESCA</td>
<td>TCGA-ESCA</td>
</tr>
<tr class="odd">
<td>ESCA</td>
<td>Metastatic</td>
<td>TCGA-ESCA</td>
<td>TCGA-ESCA</td>
</tr>
<tr class="even">
<td>GBM</td>
<td>Primary Tumor</td>
<td>TCGA-GBM</td>
<td>TCGA-GBM</td>
</tr>
<tr class="odd">
<td>GBM</td>
<td>Recurrent Tumor</td>
<td>TCGA-GBM</td>
<td>TCGA-GBM</td>
</tr>
<tr class="even">
<td>HNSC</td>
<td>Primary Tumor</td>
<td>TCGA-HNSC</td>
<td>TCGA-HNSC</td>
</tr>
<tr class="odd">
<td>HNSC</td>
<td>Metastatic</td>
<td>TCGA-HNSC</td>
<td>TCGA-HNSC</td>
</tr>
<tr class="even">
<td>KICH</td>
<td>Primary Tumor</td>
<td>TCGA-KICH</td>
<td>TCGA-KICH</td>
</tr>
<tr class="odd">
<td>KIRC</td>
<td>Primary Tumor</td>
<td>TCGA-KIRC</td>
<td>TCGA-KIRC</td>
</tr>
<tr class="even">
<td>KIRC</td>
<td>Additional - New Primary</td>
<td>TCGA-KIRC</td>
<td>TCGA-KIRC</td>
</tr>
<tr class="odd">
<td>KIRP</td>
<td>Primary Tumor</td>
<td>TCGA-KIRP</td>
<td>TCGA-KIRP</td>
</tr>
<tr class="even">
<td>KIRP</td>
<td>Additional - New Primary</td>
<td>TCGA-KIRP</td>
<td>TCGA-KIRP</td>
</tr>
<tr class="odd">
<td>LAML</td>
<td>Primary Blood Derived Cancer - Peripheral Blood</td>
<td>TCGA-LAML</td>
<td>TCGA-LAML</td>
</tr>
<tr class="even">
<td>LGG</td>
<td>Primary Tumor</td>
<td>TCGA-LGG</td>
<td>TCGA-LGG</td>
</tr>
<tr class="odd">
<td>LGG</td>
<td>Recurrent Tumor</td>
<td>TCGA-LGG</td>
<td>TCGA-LGG</td>
</tr>
<tr class="even">
<td>LIHC</td>
<td>Primary Tumor</td>
<td>TCGA-LIHC</td>
<td>TCGA-LIHC</td>
</tr>
<tr class="odd">
<td>LIHC</td>
<td>Recurrent Tumor</td>
<td>TCGA-LIHC</td>
<td>TCGA-LIHC</td>
</tr>
<tr class="even">
<td>LUAD</td>
<td>Primary Tumor</td>
<td>TCGA-LUAD</td>
<td>TCGA-LUAD</td>
</tr>
<tr class="odd">
<td>LUAD</td>
<td>Recurrent Tumor</td>
<td>TCGA-LUAD</td>
<td>TCGA-LUAD</td>
</tr>
<tr class="even">
<td>LUSC</td>
<td>Primary Tumor</td>
<td>TCGA-LUSC</td>
<td>TCGA-LUSC</td>
</tr>
<tr class="odd">
<td>MESO</td>
<td>Primary Tumor</td>
<td>TCGA-MESO</td>
<td>TCGA-MESO</td>
</tr>
<tr class="even">
<td>OV</td>
<td>Primary Tumor</td>
<td>TCGA-OV</td>
<td>TCGA-OV</td>
</tr>
<tr class="odd">
<td>OV</td>
<td>Recurrent Tumor</td>
<td>TCGA-OV</td>
<td>TCGA-OV</td>
</tr>
<tr class="even">
<td>PAAD</td>
<td>Primary Tumor</td>
<td>TCGA-PAAD</td>
<td>TCGA-PAAD</td>
</tr>
<tr class="odd">
<td>PAAD</td>
<td>Metastatic</td>
<td>TCGA-PAAD</td>
<td>TCGA-PAAD</td>
</tr>
<tr class="even">
<td>PCPG</td>
<td>Primary Tumor</td>
<td>TCGA-PCPG</td>
<td>TCGA-PCPG</td>
</tr>
<tr class="odd">
<td>PCPG</td>
<td>Additional - New Primary</td>
<td>TCGA-PCPG</td>
<td>TCGA-PCPG</td>
</tr>
<tr class="even">
<td>PCPG</td>
<td>Metastatic</td>
<td>TCGA-PCPG</td>
<td>TCGA-PCPG</td>
</tr>
<tr class="odd">
<td>PRAD</td>
<td>Primary Tumor</td>
<td>TCGA-PRAD</td>
<td>TCGA-PRAD</td>
</tr>
<tr class="even">
<td>PRAD</td>
<td>Metastatic</td>
<td>TCGA-PRAD</td>
<td>TCGA-PRAD</td>
</tr>
<tr class="odd">
<td>READ</td>
<td>Primary Tumor</td>
<td>TCGA-READ</td>
<td>TCGA-READ</td>
</tr>
<tr class="even">
<td>READ</td>
<td>Recurrent Tumor</td>
<td>TCGA-READ</td>
<td>TCGA-READ</td>
</tr>
<tr class="odd">
<td>SARC</td>
<td>Primary Tumor</td>
<td>TCGA-SARC</td>
<td>TCGA-SARC</td>
</tr>
<tr class="even">
<td>SARC</td>
<td>Recurrent Tumor</td>
<td>TCGA-SARC</td>
<td>TCGA-SARC</td>
</tr>
<tr class="odd">
<td>SARC</td>
<td>Metastatic</td>
<td>TCGA-SARC</td>
<td>TCGA-SARC</td>
</tr>
<tr class="even">
<td>SKCM</td>
<td>Metastatic</td>
<td>TCGA-SKCM</td>
<td>TCGA-SKCM</td>
</tr>
<tr class="odd">
<td>SKCM</td>
<td>Primary Tumor</td>
<td>TCGA-SKCM</td>
<td>TCGA-SKCM</td>
</tr>
<tr class="even">
<td>SKCM</td>
<td>Additional Metastatic</td>
<td>TCGA-SKCM</td>
<td>TCGA-SKCM</td>
</tr>
<tr class="odd">
<td>STAD</td>
<td>Primary Tumor</td>
<td>TCGA-STAD</td>
<td>TCGA-STAD</td>
</tr>
<tr class="even">
<td>TGCT</td>
<td>Primary Tumor</td>
<td>TCGA-TGCT</td>
<td>TCGA-TGCT</td>
</tr>
<tr class="odd">
<td>TGCT</td>
<td>Additional - New Primary</td>
<td>TCGA-TGCT</td>
<td>TCGA-TGCT</td>
</tr>
<tr class="even">
<td>THCA</td>
<td>Primary Tumor</td>
<td>TCGA-THCA</td>
<td>TCGA-THCA</td>
</tr>
<tr class="odd">
<td>THCA</td>
<td>Metastatic</td>
<td>TCGA-THCA</td>
<td>TCGA-THCA</td>
</tr>
<tr class="even">
<td>THYM</td>
<td>Primary Tumor</td>
<td>TCGA-THYM</td>
<td>TCGA-THYM</td>
</tr>
<tr class="odd">
<td>UCEC</td>
<td>Primary Tumor</td>
<td>TCGA-UCEC</td>
<td>TCGA-UCEC</td>
</tr>
<tr class="even">
<td>UCEC</td>
<td>Recurrent Tumor</td>
<td>TCGA-UCEC</td>
<td>TCGA-UCEC</td>
</tr>
<tr class="odd">
<td>UCS</td>
<td>Primary Tumor</td>
<td>TCGA-UCS</td>
<td>TCGA-UCS</td>
</tr>
<tr class="even">
<td>UVM</td>
<td>Primary Tumor</td>
<td>TCGA-UVM</td>
<td>TCGA-UVM</td>
</tr>
<tr class="odd">
<td>BLCA</td>
<td>Solid Tissue Normal</td>
<td>TCGA-BLCA</td>
<td>TCGA-NORMAL</td>
</tr>
<tr class="even">
<td>BRCA</td>
<td>Solid Tissue Normal</td>
<td>TCGA-BRCA</td>
<td>TCGA-NORMAL</td>
</tr>
<tr class="odd">
<td>CESC</td>
<td>Solid Tissue Normal</td>
<td>TCGA-CESC</td>
<td>TCGA-NORMAL</td>
</tr>
<tr class="even">
<td>CHOL</td>
<td>Solid Tissue Normal</td>
<td>TCGA-CHOL</td>
<td>TCGA-NORMAL</td>
</tr>
<tr class="odd">
<td>COAD</td>
<td>Solid Tissue Normal</td>
<td>TCGA-COAD</td>
<td>TCGA-NORMAL</td>
</tr>
<tr class="even">
<td>ESCA</td>
<td>Solid Tissue Normal</td>
<td>TCGA-ESCA</td>
<td>TCGA-NORMAL</td>
</tr>
<tr class="odd">
<td>HNSC</td>
<td>Solid Tissue Normal</td>
<td>TCGA-HNSC</td>
<td>TCGA-NORMAL</td>
</tr>
<tr class="even">
<td>KICH</td>
<td>Solid Tissue Normal</td>
<td>TCGA-KICH</td>
<td>TCGA-NORMAL</td>
</tr>
<tr class="odd">
<td>KIRC</td>
<td>Solid Tissue Normal</td>
<td>TCGA-KIRC</td>
<td>TCGA-NORMAL</td>
</tr>
<tr class="even">
<td>KIRP</td>
<td>Solid Tissue Normal</td>
<td>TCGA-KIRP</td>
<td>TCGA-NORMAL</td>
</tr>
<tr class="odd">
<td>LIHC</td>
<td>Solid Tissue Normal</td>
<td>TCGA-LIHC</td>
<td>TCGA-NORMAL</td>
</tr>
<tr class="even">
<td>LUAD</td>
<td>Solid Tissue Normal</td>
<td>TCGA-LUAD</td>
<td>TCGA-NORMAL</td>
</tr>
<tr class="odd">
<td>LUSC</td>
<td>Solid Tissue Normal</td>
<td>TCGA-LUSC</td>
<td>TCGA-NORMAL</td>
</tr>
<tr class="even">
<td>PAAD</td>
<td>Solid Tissue Normal</td>
<td>TCGA-PAAD</td>
<td>TCGA-NORMAL</td>
</tr>
<tr class="odd">
<td>PCPG</td>
<td>Solid Tissue Normal</td>
<td>TCGA-PCPG</td>
<td>TCGA-NORMAL</td>
</tr>
<tr class="even">
<td>PRAD</td>
<td>Solid Tissue Normal</td>
<td>TCGA-PRAD</td>
<td>TCGA-NORMAL</td>
</tr>
<tr class="odd">
<td>READ</td>
<td>Solid Tissue Normal</td>
<td>TCGA-READ</td>
<td>TCGA-NORMAL</td>
</tr>
<tr class="even">
<td>SARC</td>
<td>Solid Tissue Normal</td>
<td>TCGA-SARC</td>
<td>TCGA-NORMAL</td>
</tr>
<tr class="odd">
<td>SKCM</td>
<td>Solid Tissue Normal</td>
<td>TCGA-SKCM</td>
<td>TCGA-NORMAL</td>
</tr>
<tr class="even">
<td>STAD</td>
<td>Solid Tissue Normal</td>
<td>TCGA-STAD</td>
<td>TCGA-NORMAL</td>
</tr>
<tr class="odd">
<td>THCA</td>
<td>Solid Tissue Normal</td>
<td>TCGA-THCA</td>
<td>TCGA-NORMAL</td>
</tr>
<tr class="even">
<td>THYM</td>
<td>Solid Tissue Normal</td>
<td>TCGA-THYM</td>
<td>TCGA-NORMAL</td>
</tr>
<tr class="odd">
<td>UCEC</td>
<td>Solid Tissue Normal</td>
<td>TCGA-UCEC</td>
<td>TCGA-NORMAL</td>
</tr>
</tbody>
</table>
</section>
<section id="clinical-data-preprocessing" class="level2">
<h2 class="anchored" data-anchor-id="clinical-data-preprocessing">Clinical data preprocessing</h2>
<section id="tcga" class="level3">
<h3 class="anchored" data-anchor-id="tcga">TCGA</h3>
<div class="sourceCode" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co"># load("/home/data2/Projects/Fusion/unclean/TCGA_ALL_clinical.rdata")</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="co"># TCGA_ALL_clinical |&gt;</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co">#   rename(Patient_ID = bcr_patient_barcode,</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="co">#          Age = age_at_initial_pathologic_diagnosis,</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="co">#          Sex = gender,</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="co">#          Race = race,</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="co">#          )</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="co"># [Discrepancy] [Not Applicable]  [Not Available]</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="co"># [Discrepancy] [Not Available]       [Unknown]</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="co"># # [Discrepancy]                                        [Not Applicable]</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="co"># 29                                                          105</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="co"># [Not Available]                                         [Not Evaluated]</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="co"># 5140                                                       4</span></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="co"># [Unknown]</span></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(UCSCXenaShiny)</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(tcga_clinical_fine<span class="sc">$</span>Stage_ajcc)</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>tcga_clinical</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>tcga_clinical_fine</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>tcga_genome_instability</span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>tcga_purity</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>tcga_subtypes</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>tcga_surv</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(tcga_clinical_fine)</span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(<span class="fu">unique</span>(tcga_clinical_fine))</span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a>tcga_clinical <span class="ot">=</span> <span class="fu">unique</span>(tcga_clinical_fine) <span class="sc">|&gt;</span></span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(tcga_surv, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Sample"</span><span class="ot">=</span><span class="st">"sample"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(tcga_purity, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Sample"</span><span class="ot">=</span><span class="st">"sample"</span>))</span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a>tcga_clinical[<span class="dv">551</span>, ]<span class="sc">$</span>Sample</span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">startsWith</span>(tcga_genome_instability<span class="sc">$</span>sample, <span class="st">"TCGA"</span>))</span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a>tcga_clinical <span class="ot">=</span> tcga_clinical <span class="sc">|&gt;</span></span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(tcga_genome_instability[<span class="fu">startsWith</span>(tcga_genome_instability<span class="sc">$</span>sample, <span class="st">"TCGA"</span>) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">duplicated</span>(tcga_genome_instability<span class="sc">$</span>sample), ],  <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Sample"</span><span class="ot">=</span><span class="st">"sample"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a>  <span class="co">#left_join(tcga_genome_instability[duplicated(tcga_genome_instability$sample), ],  by = c("Sample"="sample")) |&gt;</span></span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(tcga_subtypes,  <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Sample"</span><span class="ot">=</span><span class="st">"sampleID"</span>))</span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">is.na</span>(tcga_clinical<span class="sc">$</span>Sample))</span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">duplicated</span>(tcga_clinical<span class="sc">$</span>Sample))</span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a>tcga_clinical[<span class="fu">duplicated</span>(tcga_clinical<span class="sc">$</span>Sample), ]<span class="sc">$</span>Sample</span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a>tcga_clinical.final <span class="ot">=</span> tcga_clinical[<span class="sc">!</span><span class="fu">duplicated</span>(tcga_clinical<span class="sc">$</span>Sample), ] <span class="sc">|&gt;</span></span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Sample_ID =</span> Sample, <span class="at">Sex =</span> Gender, <span class="at">OS_Time =</span> OS.time, <span class="at">OS_Status =</span> OS, <span class="at">DSS_Time =</span> DSS.time, <span class="at">DSS_Status =</span> DSS, <span class="at">DFS_Time =</span> DFI.time, <span class="at">DFS_Status =</span> DFI, <span class="at">PFS_Time =</span> PFI.time, <span class="at">PFS_Status =</span> PFI) <span class="sc">|&gt;</span></span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>cancer_type) <span class="sc">|&gt;</span></span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">ifelse</span>(Sex <span class="sc">==</span> <span class="st">"FEMALE"</span>, <span class="st">"F"</span>, <span class="st">"M"</span>),</span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a>         <span class="at">Patient_ID =</span> <span class="fu">substr</span>(Sample_ID, <span class="dv">1</span>, <span class="dv">12</span>))</span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">is.na</span>(tcga_clinical.final<span class="sc">$</span>Sample_ID))</span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">is.na</span>(tcga_clinical.final<span class="sc">$</span>Cancer))</span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">duplicated</span>(tcga_clinical.final<span class="sc">$</span>Sample_ID))</span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a>tcga_clinical.final <span class="ot">=</span> tcga_clinical.final <span class="sc">|&gt;</span></span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Patient_ID, Sample_ID, <span class="fu">everything</span>()) <span class="sc">|&gt;</span></span>
<span id="cb1-67"><a href="#cb1-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.table</span>()</span>
<span id="cb1-68"><a href="#cb1-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-69"><a href="#cb1-69" aria-hidden="true" tabindex="-1"></a>tcga_clinical.final</span>
<span id="cb1-70"><a href="#cb1-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-71"><a href="#cb1-71" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(tcga_clinical.final, <span class="at">file =</span> <span class="st">"/home/data2/Projects/Fusion/fusiondb/Clininfo/TCGA_info.tsv"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="cptac" class="level3">
<h3 class="anchored" data-anchor-id="cptac">CPTAC</h3>
<div class="sourceCode" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Clinical data from CPTAC paper ------------------------------------------</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>cli_list <span class="ot">=</span> fs<span class="sc">::</span><span class="fu">dir_ls</span>(<span class="st">"CPTAC_Clinical_meta_data_v1/"</span>)</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>cli <span class="ot">=</span> purrr<span class="sc">::</span><span class="fu">map</span>(cli_list, fread) <span class="sc">|&gt;</span> <span class="fu">rbindlist</span>(<span class="at">fill =</span> <span class="cn">TRUE</span>, <span class="at">use.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(cli)</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Problem: Need to skip the second row</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="co"># This also includes samples from CPTAC-2/3, but the total number is fewer</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>cli2 <span class="ot">=</span> purrr<span class="sc">::</span><span class="fu">map</span>(cli_list, <span class="cf">function</span>(x) {</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Since the column names are also different, read them twice—once for the column names and once for the data</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>  data1 <span class="ot">=</span> <span class="fu">fread</span>(x)</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>  data2 <span class="ot">=</span> <span class="fu">fread</span>(x, <span class="at">skip =</span> <span class="dv">2</span>, <span class="at">header =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Update the column names</span></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(data2) <span class="ot">=</span> <span class="fu">colnames</span>(data1)</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>  data2<span class="sc">$</span>Cancer <span class="ot">=</span> stringr<span class="sc">::</span><span class="fu">str_remove</span>(<span class="fu">basename</span>(x), <span class="st">"_meta.txt"</span>)</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>  data2 <span class="sc">|&gt;</span> <span class="fu">as.data.table</span>()</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>}) <span class="sc">|&gt;</span> <span class="fu">rbindlist</span>(<span class="at">fill =</span> <span class="cn">TRUE</span>, <span class="at">use.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a><span class="co">#colnames(cli2) = colnames(cli)</span></span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(cli2)</span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>cli2[Tumor <span class="sc">==</span> <span class="st">"Yes"</span> <span class="sc">&amp;</span> Normal <span class="sc">==</span> <span class="st">"Yes"</span>] <span class="co"># This is to mark whether the patient has both tumor and normal samples</span></span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>cli_list2 <span class="ot">=</span> fs<span class="sc">::</span><span class="fu">dir_ls</span>(<span class="st">"meta_data_BCM/"</span>)</span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>cli3 <span class="ot">=</span> purrr<span class="sc">::</span><span class="fu">map</span>(cli_list2, <span class="cf">function</span>(x) {</span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Since the column names are also different, read them twice—once for the column names and once for the data</span></span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>  data1 <span class="ot">=</span> <span class="fu">fread</span>(x)</span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>  data2 <span class="ot">=</span> <span class="fu">fread</span>(x, <span class="at">skip =</span> <span class="dv">2</span>, <span class="at">header =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>  <span class="co"># 更新列名</span></span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(data2) <span class="ot">=</span> <span class="fu">colnames</span>(data1)</span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a>  data2<span class="sc">$</span>Cancer <span class="ot">=</span> stringr<span class="sc">::</span><span class="fu">str_remove</span>(<span class="fu">basename</span>(x), <span class="st">"_meta.txt"</span>)</span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a>  data2 <span class="sc">|&gt;</span> <span class="fu">as.data.table</span>()</span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a>}) <span class="sc">|&gt;</span> <span class="fu">rbindlist</span>(<span class="at">fill =</span> <span class="cn">TRUE</span>, <span class="at">use.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(cli3)</span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a><span class="fu">all.equal</span>(cli2, cli3)</span>
<span id="cb2-43"><a href="#cb2-43" aria-hidden="true" tabindex="-1"></a><span class="fu">all.equal</span>(<span class="fu">sort</span>(<span class="fu">colnames</span>(cli2)), <span class="fu">sort</span>(<span class="fu">colnames</span>(cli3)))</span>
<span id="cb2-44"><a href="#cb2-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-45"><a href="#cb2-45" aria-hidden="true" tabindex="-1"></a>cli4 <span class="ot">=</span> cli3[, <span class="fu">colnames</span>(cli2), with <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb2-46"><a href="#cb2-46" aria-hidden="true" tabindex="-1"></a><span class="fu">all.equal</span>(cli2, cli4)</span>
<span id="cb2-47"><a href="#cb2-47" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(cli2, cli4)</span>
<span id="cb2-48"><a href="#cb2-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-49"><a href="#cb2-49" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">setdiff</span>(cli2, cli4) <span class="sc">|&gt;</span> <span class="fu">View</span>()</span>
<span id="cb2-50"><a href="#cb2-50" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">setdiff</span>(cli4, cli2) <span class="sc">|&gt;</span> <span class="fu">nrow</span>()</span>
<span id="cb2-51"><a href="#cb2-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-52"><a href="#cb2-52" aria-hidden="true" tabindex="-1"></a><span class="co"># rbind(cli2[idx == "01OV007"],</span></span>
<span id="cb2-53"><a href="#cb2-53" aria-hidden="true" tabindex="-1"></a><span class="co">#       cli4[idx == "01OV007"]) |&gt; View()</span></span>
<span id="cb2-54"><a href="#cb2-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-55"><a href="#cb2-55" aria-hidden="true" tabindex="-1"></a><span class="co"># The files in the directories of the two data sources are basically identical</span></span>
<span id="cb2-56"><a href="#cb2-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-57"><a href="#cb2-57" aria-hidden="true" tabindex="-1"></a>cli.final <span class="ot">=</span> cli2 <span class="sc">|&gt;</span></span>
<span id="cb2-58"><a href="#cb2-58" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(</span>
<span id="cb2-59"><a href="#cb2-59" aria-hidden="true" tabindex="-1"></a>    <span class="at">Patient_ID =</span> idx,</span>
<span id="cb2-60"><a href="#cb2-60" aria-hidden="true" tabindex="-1"></a>    <span class="at">OS_Status =</span> OS_event,</span>
<span id="cb2-61"><a href="#cb2-61" aria-hidden="true" tabindex="-1"></a>    <span class="at">OS_Time =</span> OS_days,</span>
<span id="cb2-62"><a href="#cb2-62" aria-hidden="true" tabindex="-1"></a>    <span class="at">PFS_Status =</span> PFS_event,</span>
<span id="cb2-63"><a href="#cb2-63" aria-hidden="true" tabindex="-1"></a>    <span class="at">PFS_Time =</span> PFS_days</span>
<span id="cb2-64"><a href="#cb2-64" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb2-65"><a href="#cb2-65" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb2-66"><a href="#cb2-66" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sample_ID =</span> Patient_ID,</span>
<span id="cb2-67"><a href="#cb2-67" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sex =</span> <span class="fu">ifelse</span>(Sex <span class="sc">==</span> <span class="st">"Female"</span>, <span class="st">"F"</span>, <span class="st">"M"</span>)</span>
<span id="cb2-68"><a href="#cb2-68" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb2-69"><a href="#cb2-69" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb2-70"><a href="#cb2-70" aria-hidden="true" tabindex="-1"></a>    Patient_ID, Sample_ID, Cancer, Age, Sex, OS_Time, OS_Status, PFS_Time, PFS_Status, dplyr<span class="sc">::</span><span class="fu">everything</span>()</span>
<span id="cb2-71"><a href="#cb2-71" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb2-72"><a href="#cb2-72" aria-hidden="true" tabindex="-1"></a>  <span class="co"># dplyr::select(</span></span>
<span id="cb2-73"><a href="#cb2-73" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   - dplyr::starts_with("CIBERSORT"), - dplyr::starts_with("xCell")</span></span>
<span id="cb2-74"><a href="#cb2-74" aria-hidden="true" tabindex="-1"></a>  <span class="co"># )</span></span>
<span id="cb2-75"><a href="#cb2-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-76"><a href="#cb2-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-77"><a href="#cb2-77" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(cli.final, <span class="at">file =</span> <span class="st">"CPTAC_cli_paper_combined.tsv"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb2-78"><a href="#cb2-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-79"><a href="#cb2-79" aria-hidden="true" tabindex="-1"></a><span class="co"># Clinical data from GDC --------------------------------------------------</span></span>
<span id="cb2-80"><a href="#cb2-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-81"><a href="#cb2-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-82"><a href="#cb2-82" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(<span class="st">"</span></span>
<span id="cb2-83"><a href="#cb2-83" aria-hidden="true" tabindex="-1"></a><span class="st">cd /home/data2/Projects/Fusion/unclean/gdc_cli/CPTAC</span></span>
<span id="cb2-84"><a href="#cb2-84" aria-hidden="true" tabindex="-1"></a><span class="st">mkdir CPTAC-3 CPTAC-2</span></span>
<span id="cb2-85"><a href="#cb2-85" aria-hidden="true" tabindex="-1"></a><span class="st">tar zxvf clinical.project-cptac-2.2024-08-07.tar.gz -C CPTAC-2</span></span>
<span id="cb2-86"><a href="#cb2-86" aria-hidden="true" tabindex="-1"></a><span class="st">tar zxvf clinical.project-cptac-3.2024-08-07.tar.gz -C CPTAC-3</span></span>
<span id="cb2-87"><a href="#cb2-87" aria-hidden="true" tabindex="-1"></a><span class="st">       "</span>)</span>
<span id="cb2-88"><a href="#cb2-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-89"><a href="#cb2-89" aria-hidden="true" tabindex="-1"></a>cptac2 <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">"gdc_cli/CPTAC/CPTAC-2/clinical.tsv"</span>)</span>
<span id="cb2-90"><a href="#cb2-90" aria-hidden="true" tabindex="-1"></a>cptac3 <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">"gdc_cli/CPTAC/CPTAC-3/clinical.tsv"</span>)</span>
<span id="cb2-91"><a href="#cb2-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-92"><a href="#cb2-92" aria-hidden="true" tabindex="-1"></a><span class="fu">which</span>(<span class="fu">colnames</span>(cptac2) <span class="sc">%in%</span> <span class="st">"residual_disease"</span>)</span>
<span id="cb2-93"><a href="#cb2-93" aria-hidden="true" tabindex="-1"></a><span class="fu">which</span>(<span class="fu">colnames</span>(cptac3) <span class="sc">%in%</span> <span class="st">"residual_disease"</span>)</span>
<span id="cb2-94"><a href="#cb2-94" aria-hidden="true" tabindex="-1"></a><span class="fu">all.equal</span>(cptac2[[<span class="dv">119</span>]], cptac2[[<span class="dv">192</span>]])</span>
<span id="cb2-95"><a href="#cb2-95" aria-hidden="true" tabindex="-1"></a><span class="fu">all.equal</span>(cptac3[[<span class="dv">119</span>]], cptac3[[<span class="dv">192</span>]])</span>
<span id="cb2-96"><a href="#cb2-96" aria-hidden="true" tabindex="-1"></a>cptac2[[<span class="dv">192</span>]] <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb2-97"><a href="#cb2-97" aria-hidden="true" tabindex="-1"></a>cptac3[[<span class="dv">192</span>]] <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb2-98"><a href="#cb2-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-99"><a href="#cb2-99" aria-hidden="true" tabindex="-1"></a><span class="fu">setdiff</span>(cptac2<span class="sc">$</span>case_submitter_id, cli.final<span class="sc">$</span>Sample_ID)</span>
<span id="cb2-100"><a href="#cb2-100" aria-hidden="true" tabindex="-1"></a><span class="fu">setdiff</span>(cptac3<span class="sc">$</span>case_submitter_id, cli.final<span class="sc">$</span>Sample_ID)</span>
<span id="cb2-101"><a href="#cb2-101" aria-hidden="true" tabindex="-1"></a><span class="co"># &gt; setdiff(cptac2$case_submitter_id, cli.final$Sample_ID)</span></span>
<span id="cb2-102"><a href="#cb2-102" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] "01OV049"   "26OV010"   "100004028" "05BR058"   "05BR055"   "17OV019"   "11BR069"   "02OV042"   "100004012" "05BR051"   "03BR012"   "02OV045"   "04OV041"</span></span>
<span id="cb2-103"><a href="#cb2-103" aria-hidden="true" tabindex="-1"></a><span class="co"># [14] "05BR052"   "05BR031"   "1488"      "11OV009"   "02OV040"   "01OV002"   "100002921" "22OV001"   "13OV004"   "100003304" "02OV035"   "01OV046"   "01OV045"</span></span>
<span id="cb2-104"><a href="#cb2-104" aria-hidden="true" tabindex="-1"></a><span class="co"># [27] "17OV034"</span></span>
<span id="cb2-105"><a href="#cb2-105" aria-hidden="true" tabindex="-1"></a><span class="co"># &gt; setdiff(cptac3$case_submitter_id, cli.final$Sample_ID)</span></span>
<span id="cb2-106"><a href="#cb2-106" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] "C3N-02672"                    "C3L-01929"                    "C3N-02682"                    "C3N-02012"                    "C3N-02070"</span></span>
<span id="cb2-107"><a href="#cb2-107" aria-hidden="true" tabindex="-1"></a><span class="co"># [6] "C3N-02696"                    "C3L-02354"                    "C3N-03026"                    "C3L-01739"                    "C3N-02088"</span></span>
<span id="cb2-108"><a href="#cb2-108" aria-hidden="true" tabindex="-1"></a><span class="co"># [11] "C3L-04213"                    "C3N-03205"                    "C3L-04081"                    "C3N-03791"                    "C3L-03679"</span></span>
<span id="cb2-109"><a href="#cb2-109" aria-hidden="true" tabindex="-1"></a><span class="co"># [16] "C3L-02643"                    "C3N-04686"                    "C3L-01558"                    "C3L-03268"                    "C3L-01633"</span></span>
<span id="cb2-110"><a href="#cb2-110" aria-hidden="true" tabindex="-1"></a><span class="co"># [21] "C3L-02220"                    "C3N-02439"                    "C3N-02028"                    "C3N-03800"                    "C3L-04392"</span></span>
<span id="cb2-111"><a href="#cb2-111" aria-hidden="true" tabindex="-1"></a><span class="co"># [26] "C3L-02556"                    "C3N-02721"                    "C3L-01672"                    "C3L-00938"                    "C3L-02201"</span></span>
<span id="cb2-112"><a href="#cb2-112" aria-hidden="true" tabindex="-1"></a><span class="co"># [31] "C3N-02146"                    "C3N-02639"                    "C3N-03788"                    "C3L-03733"                    "C3L-04037"</span></span>
<span id="cb2-113"><a href="#cb2-113" aria-hidden="true" tabindex="-1"></a><span class="co"># [36] "C3N-01091"                    "C3L-02747"                    "C3N-01879"                    "C3N-02996"                    "C3N-02296"</span></span>
<span id="cb2-114"><a href="#cb2-114" aria-hidden="true" tabindex="-1"></a><span class="co"># [41] "C3L-02553"                    "C3L-03632"                    "C3N-03755"                    "GTEX-NPJ7-0011-R10A-SM-HAKXW" "C3L-02746"</span></span>
<span id="cb2-115"><a href="#cb2-115" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">setdiff</span>(cptac3<span class="sc">$</span>case_submitter_id, cli.final<span class="sc">$</span>Sample_ID))  <span class="co"># 586</span></span>
<span id="cb2-116"><a href="#cb2-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-117"><a href="#cb2-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-118"><a href="#cb2-118" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(cptac2[case_submitter_id <span class="sc">%in%</span> <span class="fu">setdiff</span>(cptac2<span class="sc">$</span>case_submitter_id, cli.final<span class="sc">$</span>Sample_ID)])</span>
<span id="cb2-119"><a href="#cb2-119" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(cptac3[case_submitter_id <span class="sc">%in%</span> <span class="fu">setdiff</span>(cptac3<span class="sc">$</span>case_submitter_id, cli.final<span class="sc">$</span>Sample_ID)])</span>
<span id="cb2-120"><a href="#cb2-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-121"><a href="#cb2-121" aria-hidden="true" tabindex="-1"></a>data.table<span class="sc">::</span><span class="fu">fwrite</span>(</span>
<span id="cb2-122"><a href="#cb2-122" aria-hidden="true" tabindex="-1"></a>  cptac3[case_submitter_id <span class="sc">%in%</span> <span class="fu">setdiff</span>(cptac3<span class="sc">$</span>case_submitter_id, cli.final<span class="sc">$</span>Sample_ID)],</span>
<span id="cb2-123"><a href="#cb2-123" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">"CPTAC3_non_included.tsv"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span></span>
<span id="cb2-124"><a href="#cb2-124" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-125"><a href="#cb2-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-126"><a href="#cb2-126" aria-hidden="true" tabindex="-1"></a>cptac3</span>
<span id="cb2-127"><a href="#cb2-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-128"><a href="#cb2-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-129"><a href="#cb2-129" aria-hidden="true" tabindex="-1"></a><span class="co"># Supplement the missing sample information in the paper based on the existing information in the GDC. ---------------------------------------------</span></span>
<span id="cb2-130"><a href="#cb2-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-131"><a href="#cb2-131" aria-hidden="true" tabindex="-1"></a><span class="co"># might be columns that have labeled cancer types.</span></span>
<span id="cb2-132"><a href="#cb2-132" aria-hidden="true" tabindex="-1"></a><span class="co"># primary_diagnosis site_of_resection_or_biopsy tissue_or_organ_of_origin</span></span>
<span id="cb2-133"><a href="#cb2-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-134"><a href="#cb2-134" aria-hidden="true" tabindex="-1"></a>cptac <span class="ot">=</span> <span class="fu">rbind</span>(cptac2, cptac3)[, .(case_submitter_id, primary_diagnosis, site_of_resection_or_biopsy, tissue_or_organ_of_origin)] <span class="sc">|&gt;</span> <span class="fu">unique</span>()</span>
<span id="cb2-135"><a href="#cb2-135" aria-hidden="true" tabindex="-1"></a>cptac <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(cptac, cli.final[, .(Sample_ID, Cancer)], <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"case_submitter_id"</span><span class="ot">=</span><span class="st">"Sample_ID"</span>))</span>
<span id="cb2-136"><a href="#cb2-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-137"><a href="#cb2-137" aria-hidden="true" tabindex="-1"></a><span class="fu">setdiff</span>(cli.final<span class="sc">$</span>Sample_ID, cptac<span class="sc">$</span>case_submitter_id)</span>
<span id="cb2-138"><a href="#cb2-138" aria-hidden="true" tabindex="-1"></a><span class="co"># 29 cases are not in the GDC data.</span></span>
<span id="cb2-139"><a href="#cb2-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-140"><a href="#cb2-140" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(cptac<span class="sc">$</span>Cancer, cptac<span class="sc">$</span>primary_diagnosis) <span class="sc">|&gt;</span> pheatmap<span class="sc">::</span><span class="fu">pheatmap</span>()</span>
<span id="cb2-141"><a href="#cb2-141" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(cptac<span class="sc">$</span>Cancer, cptac<span class="sc">$</span>site_of_resection_or_biopsy) <span class="sc">|&gt;</span> pheatmap<span class="sc">::</span><span class="fu">pheatmap</span>()</span>
<span id="cb2-142"><a href="#cb2-142" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(cptac<span class="sc">$</span>Cancer, cptac<span class="sc">$</span>tissue_or_organ_of_origin) <span class="sc">|&gt;</span> pheatmap<span class="sc">::</span><span class="fu">pheatmap</span>()</span>
<span id="cb2-143"><a href="#cb2-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-144"><a href="#cb2-144" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(cptac<span class="sc">$</span>Cancer,</span>
<span id="cb2-145"><a href="#cb2-145" aria-hidden="true" tabindex="-1"></a>      <span class="fu">paste</span>(</span>
<span id="cb2-146"><a href="#cb2-146" aria-hidden="true" tabindex="-1"></a>        cptac<span class="sc">$</span>primary_diagnosis,</span>
<span id="cb2-147"><a href="#cb2-147" aria-hidden="true" tabindex="-1"></a>        cptac<span class="sc">$</span>site_of_resection_or_biopsy,</span>
<span id="cb2-148"><a href="#cb2-148" aria-hidden="true" tabindex="-1"></a>        cptac<span class="sc">$</span>tissue_or_organ_of_origin, <span class="at">sep =</span> <span class="st">"_"</span></span>
<span id="cb2-149"><a href="#cb2-149" aria-hidden="true" tabindex="-1"></a>      )) <span class="sc">|&gt;</span> pheatmap<span class="sc">::</span><span class="fu">pheatmap</span>(<span class="at">show_colnames =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-150"><a href="#cb2-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-151"><a href="#cb2-151" aria-hidden="true" tabindex="-1"></a><span class="co"># Solve them one by one.</span></span>
<span id="cb2-152"><a href="#cb2-152" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sub</span>(<span class="st">"_meta.txt"</span>, <span class="st">""</span>, <span class="fu">basename</span>(<span class="fu">names</span>(cli_list))))</span>
<span id="cb2-153"><a href="#cb2-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-154"><a href="#cb2-154" aria-hidden="true" tabindex="-1"></a>typelist <span class="ot">=</span> <span class="fu">sub</span>(<span class="st">"_meta.txt"</span>, <span class="st">""</span>, <span class="fu">basename</span>(<span class="fu">names</span>(cli_list)))</span>
<span id="cb2-155"><a href="#cb2-155" aria-hidden="true" tabindex="-1"></a><span class="co"># BRCA CCRCC COAD GBM HGSC HNSCC LSCC LUAD PDAC UCEC</span></span>
<span id="cb2-156"><a href="#cb2-156" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> typelist) {</span>
<span id="cb2-157"><a href="#cb2-157" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">"check type "</span>, i)</span>
<span id="cb2-158"><a href="#cb2-158" aria-hidden="true" tabindex="-1"></a>  cptac_part <span class="ot">=</span> cptac <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(Cancer <span class="sc">==</span> i)</span>
<span id="cb2-159"><a href="#cb2-159" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">"  primary_diagnosis"</span>)</span>
<span id="cb2-160"><a href="#cb2-160" aria-hidden="true" tabindex="-1"></a>  <span class="fu">table</span>(cptac_part<span class="sc">$</span>Cancer, cptac_part<span class="sc">$</span>primary_diagnosis) <span class="sc">|&gt;</span> <span class="fu">print</span>()</span>
<span id="cb2-161"><a href="#cb2-161" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">"  site_of_resection_or_biopsy"</span>)</span>
<span id="cb2-162"><a href="#cb2-162" aria-hidden="true" tabindex="-1"></a>  <span class="fu">table</span>(cptac_part<span class="sc">$</span>Cancer, cptac_part<span class="sc">$</span>site_of_resection_or_biopsy) <span class="sc">|&gt;</span> <span class="fu">print</span>()</span>
<span id="cb2-163"><a href="#cb2-163" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">"  tissue_or_organ_of_origin"</span>)</span>
<span id="cb2-164"><a href="#cb2-164" aria-hidden="true" tabindex="-1"></a>  <span class="fu">table</span>(cptac_part<span class="sc">$</span>Cancer, cptac_part<span class="sc">$</span>tissue_or_organ_of_origin) <span class="sc">|&gt;</span> <span class="fu">print</span>()</span>
<span id="cb2-165"><a href="#cb2-165" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-166"><a href="#cb2-166" aria-hidden="true" tabindex="-1"></a><span class="co"># BRCA (122): tissue_or_organ_of_origin "Breast, NOS"</span></span>
<span id="cb2-167"><a href="#cb2-167" aria-hidden="true" tabindex="-1"></a><span class="co"># CCRCC (103): primary_diagnosis "Renal cell carcinoma, NOS"</span></span>
<span id="cb2-168"><a href="#cb2-168" aria-hidden="true" tabindex="-1"></a><span class="co">#        tissue_or_organ_of_origin "Kidney, NOS"</span></span>
<span id="cb2-169"><a href="#cb2-169" aria-hidden="true" tabindex="-1"></a><span class="co"># COAD (106): tissue_or_organ_of_origin "Colon, NOS", "Rectum, NOS"</span></span>
<span id="cb2-170"><a href="#cb2-170" aria-hidden="true" tabindex="-1"></a><span class="co"># GBM (99): primary_diagnosis "Glioblastoma"</span></span>
<span id="cb2-171"><a href="#cb2-171" aria-hidden="true" tabindex="-1"></a><span class="co"># HGSC (87): primary_diagnosis "Serous adenocarcinoma, NOS" AND</span></span>
<span id="cb2-172"><a href="#cb2-172" aria-hidden="true" tabindex="-1"></a><span class="co">#  tissue_or_organ_of_origin Fallopian tube Ovary Peritoneum, NOS</span></span>
<span id="cb2-173"><a href="#cb2-173" aria-hidden="true" tabindex="-1"></a><span class="co"># HNSCC (108): Other situation: primary_diagnosis "Squamous cell carcinoma, NOS"</span></span>
<span id="cb2-174"><a href="#cb2-174" aria-hidden="true" tabindex="-1"></a><span class="co"># Base of tongue, NOS Cheek mucosa Floor of mouth, NOS Gum, NOS</span></span>
<span id="cb2-175"><a href="#cb2-175" aria-hidden="true" tabindex="-1"></a><span class="co"># HNSCC                   1            2                  18        3</span></span>
<span id="cb2-176"><a href="#cb2-176" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb2-177"><a href="#cb2-177" aria-hidden="true" tabindex="-1"></a><span class="co"># Head, face or neck, NOS Larynx, NOS Lip, NOS Oropharynx, NOS</span></span>
<span id="cb2-178"><a href="#cb2-178" aria-hidden="true" tabindex="-1"></a><span class="co"># HNSCC                       2          47        4               4</span></span>
<span id="cb2-179"><a href="#cb2-179" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb2-180"><a href="#cb2-180" aria-hidden="true" tabindex="-1"></a><span class="co"># Overlapping lesion of lip, oral cavity and pharynx Tongue, NOS Tonsil, NOS</span></span>
<span id="cb2-181"><a href="#cb2-181" aria-hidden="true" tabindex="-1"></a><span class="co"># HNSCC</span></span>
<span id="cb2-182"><a href="#cb2-182" aria-hidden="true" tabindex="-1"></a><span class="co"># LSCC (108): primary_diagnosis "Squamous cell carcinoma, NOS" AND</span></span>
<span id="cb2-183"><a href="#cb2-183" aria-hidden="true" tabindex="-1"></a><span class="co">#   tissue_or_organ_of_origin "lung" pattern</span></span>
<span id="cb2-184"><a href="#cb2-184" aria-hidden="true" tabindex="-1"></a><span class="co"># LUAD (110): primary_diagnosis "Adenocarcinoma, NOS" AND</span></span>
<span id="cb2-185"><a href="#cb2-185" aria-hidden="true" tabindex="-1"></a><span class="co">#   tissue_or_organ_of_origin "lung" pattern</span></span>
<span id="cb2-186"><a href="#cb2-186" aria-hidden="true" tabindex="-1"></a><span class="co"># PDAC (105): tissue_or_organ_of_origin "pancreas" pattern</span></span>
<span id="cb2-187"><a href="#cb2-187" aria-hidden="true" tabindex="-1"></a><span class="co"># UCEC (95): primary_diagnosis "Endometrioid adenocarcinoma, NOS"</span></span>
<span id="cb2-188"><a href="#cb2-188" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="sc">!</span><span class="fu">is.na</span>(cptac<span class="sc">$</span>Cancer))</span>
<span id="cb2-189"><a href="#cb2-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-190"><a href="#cb2-190" aria-hidden="true" tabindex="-1"></a><span class="co"># rbind(cptac2, cptac3)[,192] # There are duplicate columns, need to go back and handle this earlier (fixed).</span></span>
<span id="cb2-191"><a href="#cb2-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-192"><a href="#cb2-192" aria-hidden="true" tabindex="-1"></a><span class="co"># Actual testing and updating</span></span>
<span id="cb2-193"><a href="#cb2-193" aria-hidden="true" tabindex="-1"></a>cptac_updated <span class="ot">=</span> <span class="fu">rbind</span>(cptac2, cptac3) <span class="sc">|&gt;</span></span>
<span id="cb2-194"><a href="#cb2-194" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>() <span class="sc">|&gt;</span></span>
<span id="cb2-195"><a href="#cb2-195" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb2-196"><a href="#cb2-196" aria-hidden="true" tabindex="-1"></a>    <span class="at">Cancer =</span> dplyr<span class="sc">::</span><span class="fu">case_when</span>(</span>
<span id="cb2-197"><a href="#cb2-197" aria-hidden="true" tabindex="-1"></a>      tissue_or_organ_of_origin <span class="sc">%in%</span> <span class="st">"Breast, NOS"</span> <span class="sc">~</span> <span class="st">"BRCA"</span>,</span>
<span id="cb2-198"><a href="#cb2-198" aria-hidden="true" tabindex="-1"></a>      primary_diagnosis <span class="sc">%in%</span> <span class="st">"Renal cell carcinoma, NOS"</span> <span class="sc">~</span> <span class="st">"CCRCC"</span>,</span>
<span id="cb2-199"><a href="#cb2-199" aria-hidden="true" tabindex="-1"></a>      tissue_or_organ_of_origin <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Colon, NOS"</span>, <span class="st">"Rectum, NOS"</span>) <span class="sc">~</span> <span class="st">"COAD"</span>,</span>
<span id="cb2-200"><a href="#cb2-200" aria-hidden="true" tabindex="-1"></a>      <span class="fu">grepl</span>(<span class="st">"Glioblastoma"</span>, primary_diagnosis, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>) <span class="sc">|</span> <span class="fu">grepl</span>(<span class="st">"Oligodendroglioma"</span>, primary_diagnosis, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>) <span class="sc">~</span> <span class="st">"GBM"</span>,</span>
<span id="cb2-201"><a href="#cb2-201" aria-hidden="true" tabindex="-1"></a>      <span class="fu">grepl</span>(<span class="st">"GTEX"</span>, case_submitter_id, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>) <span class="sc">~</span> <span class="st">"GTEX-Brain"</span>,</span>
<span id="cb2-202"><a href="#cb2-202" aria-hidden="true" tabindex="-1"></a>      case_submitter_id <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"C3L-06912"</span>, <span class="st">"C3L-07212"</span>) <span class="sc">~</span> <span class="st">"GBM"</span>,</span>
<span id="cb2-203"><a href="#cb2-203" aria-hidden="true" tabindex="-1"></a>      <span class="co">#primary_diagnosis %in% "Glioblastoma" ~ "GBM",</span></span>
<span id="cb2-204"><a href="#cb2-204" aria-hidden="true" tabindex="-1"></a>      primary_diagnosis <span class="sc">%in%</span> <span class="st">"Serous adenocarcinoma, NOS"</span> <span class="sc">&amp;</span> tissue_or_organ_of_origin <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Fallopian tube"</span>, <span class="st">"Ovary"</span>, <span class="st">"Peritoneum, NOS"</span>) <span class="sc">~</span> <span class="st">"HGSC"</span>,</span>
<span id="cb2-205"><a href="#cb2-205" aria-hidden="true" tabindex="-1"></a>      primary_diagnosis <span class="sc">%in%</span> <span class="st">"Squamous cell carcinoma, NOS"</span> <span class="sc">&amp;</span> <span class="fu">grepl</span>(<span class="st">"lung"</span>, tissue_or_organ_of_origin, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>) <span class="sc">~</span> <span class="st">"LSCC"</span>,</span>
<span id="cb2-206"><a href="#cb2-206" aria-hidden="true" tabindex="-1"></a>      primary_diagnosis <span class="sc">%in%</span> <span class="st">"Adenocarcinoma, NOS"</span> <span class="sc">&amp;</span> <span class="fu">grepl</span>(<span class="st">"lung"</span>, tissue_or_organ_of_origin, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>) <span class="sc">~</span> <span class="st">"LUAD"</span>,</span>
<span id="cb2-207"><a href="#cb2-207" aria-hidden="true" tabindex="-1"></a>      <span class="fu">grepl</span>(<span class="st">"pancreas"</span>, tissue_or_organ_of_origin, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>) <span class="sc">~</span> <span class="st">"PDAC"</span>,</span>
<span id="cb2-208"><a href="#cb2-208" aria-hidden="true" tabindex="-1"></a>      primary_diagnosis <span class="sc">%in%</span> <span class="st">"Endometrioid adenocarcinoma, NOS"</span> <span class="sc">~</span> <span class="st">"UCEC"</span>,</span>
<span id="cb2-209"><a href="#cb2-209" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"HNSCC"</span></span>
<span id="cb2-210"><a href="#cb2-210" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-211"><a href="#cb2-211" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span> <span class="fu">as.data.table</span>()</span>
<span id="cb2-212"><a href="#cb2-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-213"><a href="#cb2-213" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="st">"HNSCC"</span>) { <span class="co"># typelist</span></span>
<span id="cb2-214"><a href="#cb2-214" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">"check type "</span>, i)</span>
<span id="cb2-215"><a href="#cb2-215" aria-hidden="true" tabindex="-1"></a>  cptac_part <span class="ot">=</span> cptac_updated <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(Cancer <span class="sc">==</span> i)</span>
<span id="cb2-216"><a href="#cb2-216" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">"  primary_diagnosis"</span>)</span>
<span id="cb2-217"><a href="#cb2-217" aria-hidden="true" tabindex="-1"></a>  <span class="fu">table</span>(cptac_part<span class="sc">$</span>Cancer, cptac_part<span class="sc">$</span>primary_diagnosis) <span class="sc">|&gt;</span> <span class="fu">print</span>()</span>
<span id="cb2-218"><a href="#cb2-218" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">"  site_of_resection_or_biopsy"</span>)</span>
<span id="cb2-219"><a href="#cb2-219" aria-hidden="true" tabindex="-1"></a>  <span class="fu">table</span>(cptac_part<span class="sc">$</span>Cancer, cptac_part<span class="sc">$</span>site_of_resection_or_biopsy) <span class="sc">|&gt;</span> <span class="fu">print</span>()</span>
<span id="cb2-220"><a href="#cb2-220" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">"  tissue_or_organ_of_origin"</span>)</span>
<span id="cb2-221"><a href="#cb2-221" aria-hidden="true" tabindex="-1"></a>  <span class="fu">table</span>(cptac_part<span class="sc">$</span>Cancer, cptac_part<span class="sc">$</span>tissue_or_organ_of_origin) <span class="sc">|&gt;</span> <span class="fu">print</span>()</span>
<span id="cb2-222"><a href="#cb2-222" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-223"><a href="#cb2-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-224"><a href="#cb2-224" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(cptac<span class="sc">$</span>Cancer)</span>
<span id="cb2-225"><a href="#cb2-225" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(cptac_updated<span class="sc">$</span>Cancer)</span>
<span id="cb2-226"><a href="#cb2-226" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-227"><a href="#cb2-227" aria-hidden="true" tabindex="-1"></a>cptac_updated <span class="sc">|&gt;</span></span>
<span id="cb2-228"><a href="#cb2-228" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb2-229"><a href="#cb2-229" aria-hidden="true" tabindex="-1"></a>    primary_diagnosis <span class="sc">==</span> <span class="st">"'--"</span> <span class="sc">|</span> site_of_resection_or_biopsy <span class="sc">==</span> <span class="st">"'--"</span> <span class="sc">|</span></span>
<span id="cb2-230"><a href="#cb2-230" aria-hidden="true" tabindex="-1"></a>    tissue_or_organ_of_origin <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"'--"</span>, <span class="st">"Unknown"</span>)</span>
<span id="cb2-231"><a href="#cb2-231" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb2-232"><a href="#cb2-232" aria-hidden="true" tabindex="-1"></a><span class="co"># 1-9 from GTEX-Brain</span></span>
<span id="cb2-233"><a href="#cb2-233" aria-hidden="true" tabindex="-1"></a><span class="co"># C3L-06912 C3L-07212 Gliomas # https://portal.gdc.cancer.gov/cases/e6027623-a5b2-4d82-a9ca-f37c948dd3ed</span></span>
<span id="cb2-234"><a href="#cb2-234" aria-hidden="true" tabindex="-1"></a>cptac_updated<span class="sc">$</span>case_submitter_id[<span class="fu">grepl</span>(<span class="st">"GTEX"</span>, cptac_updated<span class="sc">$</span>case_submitter_id)]</span>
<span id="cb2-235"><a href="#cb2-235" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-236"><a href="#cb2-236" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(cptac_updated)</span>
<span id="cb2-237"><a href="#cb2-237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-238"><a href="#cb2-238" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(cptac_updated, <span class="at">file =</span> <span class="st">"CPTAC_combined_with_cancer_annotated.tsv"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb2-239"><a href="#cb2-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-240"><a href="#cb2-240" aria-hidden="true" tabindex="-1"></a>cptac_unmatched <span class="ot">=</span> cptac_updated[<span class="sc">!</span> case_submitter_id <span class="sc">%in%</span> cli.final<span class="sc">$</span>Sample_ID]</span>
<span id="cb2-241"><a href="#cb2-241" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(cptac_unmatched)</span>
<span id="cb2-242"><a href="#cb2-242" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(cli.final)[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb2-243"><a href="#cb2-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-244"><a href="#cb2-244" aria-hidden="true" tabindex="-1"></a>cptac_unmatched2 <span class="ot">=</span> cptac_unmatched <span class="sc">|&gt;</span></span>
<span id="cb2-245"><a href="#cb2-245" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb2-246"><a href="#cb2-246" aria-hidden="true" tabindex="-1"></a>    case_submitter_id, age_at_diagnosis, gender,</span>
<span id="cb2-247"><a href="#cb2-247" aria-hidden="true" tabindex="-1"></a>    days_to_death, vital_status,</span>
<span id="cb2-248"><a href="#cb2-248" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">starts_with</span>(<span class="st">"ajcc"</span>),</span>
<span id="cb2-249"><a href="#cb2-249" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">starts_with</span>(<span class="st">"days"</span>),</span>
<span id="cb2-250"><a href="#cb2-250" aria-hidden="true" tabindex="-1"></a>    last_known_disease_status, progression_or_recurrence, residual_disease,</span>
<span id="cb2-251"><a href="#cb2-251" aria-hidden="true" tabindex="-1"></a>    tumor_grade, Cancer</span>
<span id="cb2-252"><a href="#cb2-252" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb2-253"><a href="#cb2-253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-254"><a href="#cb2-254" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(cptac_unmatched2<span class="sc">$</span>days_to_last_follow_up <span class="sc">!=</span> <span class="st">"'--"</span>)</span>
<span id="cb2-255"><a href="#cb2-255" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(cptac_unmatched2<span class="sc">$</span>days_to_death <span class="sc">!=</span> <span class="st">"'--"</span>)</span>
<span id="cb2-256"><a href="#cb2-256" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(cptac_unmatched2<span class="sc">$</span>days_to_recurrence <span class="sc">!=</span> <span class="st">"'--"</span>)</span>
<span id="cb2-257"><a href="#cb2-257" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(cptac_unmatched2<span class="sc">$</span>days_to_last_known_disease_status <span class="sc">!=</span> <span class="st">"'--"</span>)</span>
<span id="cb2-258"><a href="#cb2-258" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-259"><a href="#cb2-259" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(cptac_unmatched2<span class="sc">$</span>progression_or_recurrence <span class="sc">!=</span> <span class="st">"'--"</span>)</span>
<span id="cb2-260"><a href="#cb2-260" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-261"><a href="#cb2-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-262"><a href="#cb2-262" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="sc">!</span><span class="fu">is.na</span>(cli.final<span class="sc">$</span>OS_Time))</span>
<span id="cb2-263"><a href="#cb2-263" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="sc">!</span><span class="fu">is.na</span>(cli.final<span class="sc">$</span>PFS_Time))</span>
<span id="cb2-264"><a href="#cb2-264" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(cli.final<span class="sc">$</span>OS_Time <span class="sc">&gt;=</span> cli.final<span class="sc">$</span>PFS_Time, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-265"><a href="#cb2-265" aria-hidden="true" tabindex="-1"></a>cli.final[cli.final<span class="sc">$</span>OS_Time <span class="sc">&lt;</span> cli.final<span class="sc">$</span>PFS_Time] <span class="sc">|&gt;</span> <span class="fu">View</span>()</span>
<span id="cb2-266"><a href="#cb2-266" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-267"><a href="#cb2-267" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">as.integer</span>(cptac_unmatched2<span class="sc">$</span>days_to_last_follow_up) <span class="sc">&gt;=</span> <span class="fu">as.integer</span>(cptac_unmatched2<span class="sc">$</span>days_to_last_known_disease_status), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-268"><a href="#cb2-268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-269"><a href="#cb2-269" aria-hidden="true" tabindex="-1"></a>cptac_unmatched3 <span class="ot">=</span> cptac_unmatched2 <span class="sc">|&gt;</span></span>
<span id="cb2-270"><a href="#cb2-270" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(</span>
<span id="cb2-271"><a href="#cb2-271" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sample_ID =</span> case_submitter_id,</span>
<span id="cb2-272"><a href="#cb2-272" aria-hidden="true" tabindex="-1"></a>    <span class="at">Age =</span> age_at_diagnosis,</span>
<span id="cb2-273"><a href="#cb2-273" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sex =</span> gender,</span>
<span id="cb2-274"><a href="#cb2-274" aria-hidden="true" tabindex="-1"></a>    <span class="at">OS_Time =</span> days_to_last_follow_up,</span>
<span id="cb2-275"><a href="#cb2-275" aria-hidden="true" tabindex="-1"></a>    <span class="at">OS_Status =</span> vital_status,</span>
<span id="cb2-276"><a href="#cb2-276" aria-hidden="true" tabindex="-1"></a>    <span class="at">PFS_Time =</span> days_to_last_known_disease_status,</span>
<span id="cb2-277"><a href="#cb2-277" aria-hidden="true" tabindex="-1"></a>    <span class="at">PFS_Status =</span> progression_or_recurrence,</span>
<span id="cb2-278"><a href="#cb2-278" aria-hidden="true" tabindex="-1"></a>    <span class="at">Histologic_Grade =</span> tumor_grade,</span>
<span id="cb2-279"><a href="#cb2-279" aria-hidden="true" tabindex="-1"></a>    <span class="at">Path_Stage_pT =</span> ajcc_pathologic_t,</span>
<span id="cb2-280"><a href="#cb2-280" aria-hidden="true" tabindex="-1"></a>    <span class="at">Path_Stage_pN =</span> ajcc_pathologic_n,</span>
<span id="cb2-281"><a href="#cb2-281" aria-hidden="true" tabindex="-1"></a>    <span class="at">Stage =</span> ajcc_pathologic_stage</span>
<span id="cb2-282"><a href="#cb2-282" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb2-283"><a href="#cb2-283" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb2-284"><a href="#cb2-284" aria-hidden="true" tabindex="-1"></a>    Sample_ID, Age, Sex, OS_Time, OS_Status, PFS_Time, PFS_Status,</span>
<span id="cb2-285"><a href="#cb2-285" aria-hidden="true" tabindex="-1"></a>    Path_Stage_pN<span class="sc">:</span>Path_Stage_pT,</span>
<span id="cb2-286"><a href="#cb2-286" aria-hidden="true" tabindex="-1"></a>    Histologic_Grade, Cancer</span>
<span id="cb2-287"><a href="#cb2-287" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb2-288"><a href="#cb2-288" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb2-289"><a href="#cb2-289" aria-hidden="true" tabindex="-1"></a>    <span class="at">Patient_ID =</span> Sample_ID,</span>
<span id="cb2-290"><a href="#cb2-290" aria-hidden="true" tabindex="-1"></a>    <span class="at">Age =</span> <span class="fu">as.integer</span>(<span class="fu">round</span>(<span class="fu">as.integer</span>(Age) <span class="sc">/</span> <span class="dv">365</span>)),</span>
<span id="cb2-291"><a href="#cb2-291" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sex =</span> <span class="fu">fcase</span>(Sex <span class="sc">%in%</span> <span class="st">"female"</span>, <span class="st">"F"</span>,</span>
<span id="cb2-292"><a href="#cb2-292" aria-hidden="true" tabindex="-1"></a>                Sex <span class="sc">%in%</span> <span class="st">"male"</span>, <span class="st">"M"</span>, <span class="at">default =</span> <span class="cn">NA_character_</span>),</span>
<span id="cb2-293"><a href="#cb2-293" aria-hidden="true" tabindex="-1"></a>    <span class="at">OS_Time =</span> <span class="fu">as.integer</span>(OS_Time),</span>
<span id="cb2-294"><a href="#cb2-294" aria-hidden="true" tabindex="-1"></a>    <span class="at">OS_Status =</span> <span class="fu">fcase</span>(OS_Status <span class="sc">%in%</span> <span class="st">"Alive"</span>, <span class="dv">0</span>L,</span>
<span id="cb2-295"><a href="#cb2-295" aria-hidden="true" tabindex="-1"></a>                      OS_Status <span class="sc">%in%</span> <span class="st">"Dead"</span>, <span class="dv">1</span>L, <span class="at">default =</span> <span class="cn">NA_integer_</span>),</span>
<span id="cb2-296"><a href="#cb2-296" aria-hidden="true" tabindex="-1"></a>    <span class="at">PFS_Time =</span> <span class="fu">as.integer</span>(PFS_Time),</span>
<span id="cb2-297"><a href="#cb2-297" aria-hidden="true" tabindex="-1"></a>    <span class="at">PFS_Status =</span> <span class="fu">fcase</span>(PFS_Status <span class="sc">%in%</span> <span class="st">"no"</span>, <span class="dv">0</span>L,</span>
<span id="cb2-298"><a href="#cb2-298" aria-hidden="true" tabindex="-1"></a>                       PFS_Status <span class="sc">%in%</span> <span class="st">"yes"</span>, <span class="dv">1</span>L, <span class="at">default =</span> <span class="cn">NA_integer_</span>),</span>
<span id="cb2-299"><a href="#cb2-299" aria-hidden="true" tabindex="-1"></a>    <span class="at">OS_Time =</span> <span class="fu">ifelse</span>(OS_Time <span class="sc">&lt;</span> <span class="dv">0</span>, <span class="cn">NA_integer_</span>, OS_Time),</span>
<span id="cb2-300"><a href="#cb2-300" aria-hidden="true" tabindex="-1"></a>    <span class="at">Path_Stage_pN =</span> <span class="fu">fcase</span>(</span>
<span id="cb2-301"><a href="#cb2-301" aria-hidden="true" tabindex="-1"></a>      Path_Stage_pN <span class="sc">%in%</span> <span class="st">"N0"</span>, <span class="st">"pN0"</span>,</span>
<span id="cb2-302"><a href="#cb2-302" aria-hidden="true" tabindex="-1"></a>      Path_Stage_pN <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"N1"</span>, <span class="st">"N1a"</span>), <span class="st">"pN1"</span>,</span>
<span id="cb2-303"><a href="#cb2-303" aria-hidden="true" tabindex="-1"></a>      Path_Stage_pN <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"N2"</span>, <span class="st">"N2a"</span>), <span class="st">"pN2"</span>,</span>
<span id="cb2-304"><a href="#cb2-304" aria-hidden="true" tabindex="-1"></a>      <span class="at">default =</span> <span class="cn">NA_character_</span></span>
<span id="cb2-305"><a href="#cb2-305" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb2-306"><a href="#cb2-306" aria-hidden="true" tabindex="-1"></a>    <span class="at">Path_Stage_pT =</span> <span class="fu">fcase</span>(</span>
<span id="cb2-307"><a href="#cb2-307" aria-hidden="true" tabindex="-1"></a>      Path_Stage_pT <span class="sc">%in%</span> <span class="st">"T0"</span>, <span class="st">"pT0"</span>,</span>
<span id="cb2-308"><a href="#cb2-308" aria-hidden="true" tabindex="-1"></a>      Path_Stage_pT <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"T1"</span>, <span class="st">"T1a"</span>, <span class="st">"T1b"</span>, <span class="st">"T1c"</span>), <span class="st">"pT1"</span>,</span>
<span id="cb2-309"><a href="#cb2-309" aria-hidden="true" tabindex="-1"></a>      Path_Stage_pT <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"T2"</span>, <span class="st">"T2a"</span>, <span class="st">"T2b"</span>), <span class="st">"pT2"</span>,</span>
<span id="cb2-310"><a href="#cb2-310" aria-hidden="true" tabindex="-1"></a>      Path_Stage_pT <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"T3"</span>, <span class="st">"T3a"</span>, <span class="st">"T3b"</span>), <span class="st">"pT3"</span>,</span>
<span id="cb2-311"><a href="#cb2-311" aria-hidden="true" tabindex="-1"></a>      Path_Stage_pT <span class="sc">%in%</span> <span class="st">"T4"</span>, <span class="st">"pT4"</span>,</span>
<span id="cb2-312"><a href="#cb2-312" aria-hidden="true" tabindex="-1"></a>      <span class="at">default =</span> <span class="cn">NA_character_</span></span>
<span id="cb2-313"><a href="#cb2-313" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb2-314"><a href="#cb2-314" aria-hidden="true" tabindex="-1"></a>    <span class="at">Stage =</span> <span class="fu">fcase</span>(</span>
<span id="cb2-315"><a href="#cb2-315" aria-hidden="true" tabindex="-1"></a>      Stage <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Stage I"</span>, <span class="st">"Stage IA"</span>, <span class="st">"Stage IA1"</span>, <span class="st">"Stage IA2"</span>, <span class="st">"Stage IA3"</span>, <span class="st">"Stage IB"</span>), <span class="st">"Stage I"</span>,</span>
<span id="cb2-316"><a href="#cb2-316" aria-hidden="true" tabindex="-1"></a>      Stage <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Stage II"</span>, <span class="st">"Stage IIA"</span>, <span class="st">"Stage IIB"</span>), <span class="st">"Stage II"</span>,</span>
<span id="cb2-317"><a href="#cb2-317" aria-hidden="true" tabindex="-1"></a>      Stage <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Stage III"</span>, <span class="st">"Stage IIIA"</span>, <span class="st">"Stage IIIB"</span>, <span class="st">"Stage IIIC"</span>, <span class="st">"Stage IIIC1"</span>, <span class="st">"Stage IIIC2"</span>), <span class="st">"Stage III"</span>,</span>
<span id="cb2-318"><a href="#cb2-318" aria-hidden="true" tabindex="-1"></a>      Stage <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Stage IV"</span>, <span class="st">"Stage IVB"</span>), <span class="st">"Stage IV"</span>,</span>
<span id="cb2-319"><a href="#cb2-319" aria-hidden="true" tabindex="-1"></a>      <span class="at">default =</span> <span class="cn">NA_character_</span></span>
<span id="cb2-320"><a href="#cb2-320" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb2-321"><a href="#cb2-321" aria-hidden="true" tabindex="-1"></a>    <span class="at">Histologic_Grade =</span> <span class="fu">fcase</span>(</span>
<span id="cb2-322"><a href="#cb2-322" aria-hidden="true" tabindex="-1"></a>      Histologic_Grade <span class="sc">%in%</span> <span class="st">"G1"</span>, <span class="st">"G1 Well differentiated"</span>,</span>
<span id="cb2-323"><a href="#cb2-323" aria-hidden="true" tabindex="-1"></a>      Histologic_Grade <span class="sc">%in%</span> <span class="st">"G2"</span>, <span class="st">"G2 Moderately differentiated"</span>,</span>
<span id="cb2-324"><a href="#cb2-324" aria-hidden="true" tabindex="-1"></a>      Histologic_Grade <span class="sc">%in%</span> <span class="st">"G3"</span>, <span class="st">"G3 Poorly differentiated"</span>,</span>
<span id="cb2-325"><a href="#cb2-325" aria-hidden="true" tabindex="-1"></a>      Histologic_Grade <span class="sc">%in%</span> <span class="st">"G4"</span>, <span class="st">"G4 Undifferentiated"</span>,</span>
<span id="cb2-326"><a href="#cb2-326" aria-hidden="true" tabindex="-1"></a>      <span class="at">default =</span> <span class="cn">NA_character_</span></span>
<span id="cb2-327"><a href="#cb2-327" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-328"><a href="#cb2-328" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb2-329"><a href="#cb2-329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-330"><a href="#cb2-330" aria-hidden="true" tabindex="-1"></a>cptac_unmatched3</span>
<span id="cb2-331"><a href="#cb2-331" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-332"><a href="#cb2-332" aria-hidden="true" tabindex="-1"></a><span class="co"># X: cannot be assessed</span></span>
<span id="cb2-333"><a href="#cb2-333" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-334"><a href="#cb2-334" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(cptac_unmatched3<span class="sc">$</span>Path_Stage_pN)</span>
<span id="cb2-335"><a href="#cb2-335" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(cli.final<span class="sc">$</span>Path_Stage_pN)</span>
<span id="cb2-336"><a href="#cb2-336" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-337"><a href="#cb2-337" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(cptac_unmatched3<span class="sc">$</span>Path_Stage_pT)</span>
<span id="cb2-338"><a href="#cb2-338" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(cli.final<span class="sc">$</span>Path_Stage_pT)</span>
<span id="cb2-339"><a href="#cb2-339" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-340"><a href="#cb2-340" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(cptac_unmatched3<span class="sc">$</span>Stage)</span>
<span id="cb2-341"><a href="#cb2-341" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(cli.final<span class="sc">$</span>Stage)</span>
<span id="cb2-342"><a href="#cb2-342" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-343"><a href="#cb2-343" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(cptac_unmatched3<span class="sc">$</span>Histologic_Grade)</span>
<span id="cb2-344"><a href="#cb2-344" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(cli.final<span class="sc">$</span>Histologic_Grade)</span>
<span id="cb2-345"><a href="#cb2-345" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-346"><a href="#cb2-346" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-347"><a href="#cb2-347" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-348"><a href="#cb2-348" aria-hidden="true" tabindex="-1"></a><span class="co"># Final output ------------------------------------------------------------</span></span>
<span id="cb2-349"><a href="#cb2-349" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-350"><a href="#cb2-350" aria-hidden="true" tabindex="-1"></a>cli.final<span class="fl">.2</span> <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(</span>
<span id="cb2-351"><a href="#cb2-351" aria-hidden="true" tabindex="-1"></a>  cli.final, cptac_unmatched3</span>
<span id="cb2-352"><a href="#cb2-352" aria-hidden="true" tabindex="-1"></a>) <span class="sc">|&gt;</span> <span class="fu">as.data.table</span>()</span>
<span id="cb2-353"><a href="#cb2-353" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-354"><a href="#cb2-354" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(cli.final<span class="fl">.2</span><span class="sc">$</span>Cancer)</span>
<span id="cb2-355"><a href="#cb2-355" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">table</span>(cli.final<span class="fl">.2</span><span class="sc">$</span>Cancer))</span>
<span id="cb2-356"><a href="#cb2-356" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-357"><a href="#cb2-357" aria-hidden="true" tabindex="-1"></a><span class="co"># Include all Race information?</span></span>
<span id="cb2-358"><a href="#cb2-358" aria-hidden="true" tabindex="-1"></a>cli.final<span class="fl">.2</span> <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(</span>
<span id="cb2-359"><a href="#cb2-359" aria-hidden="true" tabindex="-1"></a>  cli.final<span class="fl">.2</span>, <span class="fu">unique</span>(<span class="fu">rbind</span>(cptac2, cptac3)[, .(case_submitter_id, race)]),</span>
<span id="cb2-360"><a href="#cb2-360" aria-hidden="true" tabindex="-1"></a>  <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Sample_ID"</span><span class="ot">=</span><span class="st">"case_submitter_id"</span>)</span>
<span id="cb2-361"><a href="#cb2-361" aria-hidden="true" tabindex="-1"></a>) <span class="sc">|&gt;</span></span>
<span id="cb2-362"><a href="#cb2-362" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb2-363"><a href="#cb2-363" aria-hidden="true" tabindex="-1"></a>    <span class="at">race =</span> <span class="fu">ifelse</span>(race <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Unknown"</span>, <span class="st">"not reported"</span>), <span class="cn">NA_character_</span>,</span>
<span id="cb2-364"><a href="#cb2-364" aria-hidden="true" tabindex="-1"></a>                  race)</span>
<span id="cb2-365"><a href="#cb2-365" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb2-366"><a href="#cb2-366" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.table</span>()</span>
<span id="cb2-367"><a href="#cb2-367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-368"><a href="#cb2-368" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">is.na</span>(cli.final<span class="fl">.2</span><span class="sc">$</span>race))</span>
<span id="cb2-369"><a href="#cb2-369" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-370"><a href="#cb2-370" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(cli.final<span class="fl">.2</span>, <span class="at">file =</span> <span class="st">"/home/data2/Projects/Fusion/fusiondb/Clininfo/CPTAC_info.tsv"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="target" class="level3">
<h3 class="anchored" data-anchor-id="target">TARGET</h3>
<div class="sourceCode" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>cli1 <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">"TARGET_donor_allprojects_transfer_to_sample.gz"</span>)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>cli2 <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">"TARGET_phenotype.gz"</span>)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>target_cli <span class="ot">=</span> cli2 <span class="sc">|&gt;</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="st">`</span><span class="at">_cohort</span><span class="st">`</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">Patient_ID =</span> <span class="st">`</span><span class="at">_PATIENT</span><span class="st">`</span>,</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">Sample_ID =</span> sample_id,</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">disease_code =</span> primary_disease_code,</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">disease =</span> <span class="st">`</span><span class="at">_primary_disease</span><span class="st">`</span>,</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">sample_type =</span> <span class="st">`</span><span class="at">_sample_type</span><span class="st">`</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">full_join</span>(</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>    cli1 <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>      <span class="sc">-</span><span class="st">`</span><span class="at">_OS_UNIT</span><span class="st">`</span>, <span class="sc">-</span><span class="st">`</span><span class="at">_EVENT</span><span class="st">`</span>, <span class="sc">-</span><span class="st">`</span><span class="at">_TIME_TO_EVENT</span><span class="st">`</span></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">|&gt;</span></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">rename</span>(</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>        <span class="at">Sample_ID =</span> xena_sample,</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>        <span class="at">Sex =</span> <span class="st">`</span><span class="at">_gender</span><span class="st">`</span>,</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>        <span class="at">Age =</span> <span class="st">`</span><span class="at">_age_at_diagnosis</span><span class="st">`</span>,</span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>        <span class="at">OS_Status =</span> <span class="st">`</span><span class="at">_OS_IND</span><span class="st">`</span>,</span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>        <span class="at">OS_Time =</span> <span class="st">`</span><span class="at">_OS</span><span class="st">`</span></span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>      ), <span class="at">by =</span> <span class="st">"Sample_ID"</span></span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sex =</span> <span class="fu">ifelse</span>(Sex <span class="sc">==</span> <span class="st">"Male"</span>, <span class="st">"M"</span>, <span class="st">"F"</span>),</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">OS_Status =</span> <span class="fu">ifelse</span>(OS_Status <span class="sc">==</span> <span class="st">"Alive"</span>, <span class="dv">0</span>L, <span class="dv">1</span>L)</span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Patient_ID, Sample_ID, Age, Sex, OS_Time, OS_Status, dplyr<span class="sc">::</span><span class="fu">everything</span>()) <span class="sc">|&gt;</span></span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.table</span>()</span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a>cli1</span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a>target_cli</span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(target_cli, <span class="at">file =</span> <span class="st">"/home/data2/Projects/Fusion/fusiondb/Clininfo/TARGET_xena_info.tsv"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a><span class="co"># Using cli from GDC ------------------------------------------------------</span></span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(<span class="st">"</span></span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a><span class="st">cd /home/data2/Projects/Fusion/unclean/gdc_cli/TARGET</span></span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a><span class="st">mkdir ALL_P1 ALL_P2 ALL_P3 AML CCSK NBL OS RT WT</span></span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a><span class="st">tar zxvf clinical.project-target-all-p1.2024-08-07.tar.gz -C ALL_P1</span></span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a><span class="st">tar zxvf clinical.project-target-all-p2.2024-08-07.tar.gz -C ALL_P2</span></span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a><span class="st">tar zxvf clinical.project-target-all-p3.2024-08-07.tar.gz -C ALL_P3</span></span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a><span class="st">tar zxvf clinical.project-target-aml.2024-08-07.tar.gz -C AML</span></span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a><span class="st">tar zxvf clinical.project-target-ccsk.2024-08-07.tar.gz -C CCSK</span></span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a><span class="st">tar zxvf clinical.project-target-nbl.2024-08-07.tar.gz -C NBL</span></span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a><span class="st">tar zxvf clinical.project-target-os.2024-08-07.tar.gz -C OS</span></span>
<span id="cb3-51"><a href="#cb3-51" aria-hidden="true" tabindex="-1"></a><span class="st">tar zxvf clinical.project-target-rt.2024-08-07.tar.gz -C RT</span></span>
<span id="cb3-52"><a href="#cb3-52" aria-hidden="true" tabindex="-1"></a><span class="st">tar zxvf clinical.project-target-wt.2024-08-07.tar.gz -C WT</span></span>
<span id="cb3-53"><a href="#cb3-53" aria-hidden="true" tabindex="-1"></a><span class="st">       "</span>)</span>
<span id="cb3-54"><a href="#cb3-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-55"><a href="#cb3-55" aria-hidden="true" tabindex="-1"></a>gdc_target <span class="ot">=</span> <span class="fu">rbindlist</span>(</span>
<span id="cb3-56"><a href="#cb3-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(</span>
<span id="cb3-57"><a href="#cb3-57" aria-hidden="true" tabindex="-1"></a>    <span class="fu">fread</span>(<span class="st">"gdc_cli/TARGET/ALL_P1/clinical.tsv"</span>),</span>
<span id="cb3-58"><a href="#cb3-58" aria-hidden="true" tabindex="-1"></a>    <span class="fu">fread</span>(<span class="st">"gdc_cli/TARGET/ALL_P2/clinical.tsv"</span>),</span>
<span id="cb3-59"><a href="#cb3-59" aria-hidden="true" tabindex="-1"></a>    <span class="fu">fread</span>(<span class="st">"gdc_cli/TARGET/ALL_P3/clinical.tsv"</span>),</span>
<span id="cb3-60"><a href="#cb3-60" aria-hidden="true" tabindex="-1"></a>    <span class="fu">fread</span>(<span class="st">"gdc_cli/TARGET/AML/clinical.tsv"</span>),</span>
<span id="cb3-61"><a href="#cb3-61" aria-hidden="true" tabindex="-1"></a>    <span class="fu">fread</span>(<span class="st">"gdc_cli/TARGET/CCSK/clinical.tsv"</span>),</span>
<span id="cb3-62"><a href="#cb3-62" aria-hidden="true" tabindex="-1"></a>    <span class="fu">fread</span>(<span class="st">"gdc_cli/TARGET/NBL/clinical.tsv"</span>),</span>
<span id="cb3-63"><a href="#cb3-63" aria-hidden="true" tabindex="-1"></a>    <span class="fu">fread</span>(<span class="st">"gdc_cli/TARGET/OS/clinical.tsv"</span>),</span>
<span id="cb3-64"><a href="#cb3-64" aria-hidden="true" tabindex="-1"></a>    <span class="fu">fread</span>(<span class="st">"gdc_cli/TARGET/RT/clinical.tsv"</span>),</span>
<span id="cb3-65"><a href="#cb3-65" aria-hidden="true" tabindex="-1"></a>    <span class="fu">fread</span>(<span class="st">"gdc_cli/TARGET/WT/clinical.tsv"</span>)</span>
<span id="cb3-66"><a href="#cb3-66" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb3-67"><a href="#cb3-67" aria-hidden="true" tabindex="-1"></a>  <span class="at">use.names =</span> <span class="cn">TRUE</span>, <span class="at">fill =</span> <span class="cn">TRUE</span></span>
<span id="cb3-68"><a href="#cb3-68" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-69"><a href="#cb3-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-70"><a href="#cb3-70" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(gdc_target<span class="sc">$</span>project_id)</span>
<span id="cb3-71"><a href="#cb3-71" aria-hidden="true" tabindex="-1"></a><span class="fu">any</span>(<span class="fu">duplicated</span>(gdc_target))</span>
<span id="cb3-72"><a href="#cb3-72" aria-hidden="true" tabindex="-1"></a>gdc_target <span class="ot">=</span> gdc_target[<span class="sc">!</span><span class="fu">duplicated</span>(gdc_target)]</span>
<span id="cb3-73"><a href="#cb3-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-74"><a href="#cb3-74" aria-hidden="true" tabindex="-1"></a><span class="fu">all.equal</span>(gdc_target[[<span class="dv">119</span>]], gdc_target[[<span class="dv">192</span>]])</span>
<span id="cb3-75"><a href="#cb3-75" aria-hidden="true" tabindex="-1"></a>gdc_target[[<span class="dv">192</span>]] <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb3-76"><a href="#cb3-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-77"><a href="#cb3-77" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(gdc_target<span class="sc">$</span>days_to_last_follow_up <span class="sc">!=</span> <span class="st">"'--"</span>)</span>
<span id="cb3-78"><a href="#cb3-78" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(gdc_target<span class="sc">$</span>days_to_death <span class="sc">!=</span> <span class="st">"'--"</span>)</span>
<span id="cb3-79"><a href="#cb3-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-80"><a href="#cb3-80" aria-hidden="true" tabindex="-1"></a>gdc_target2 <span class="ot">=</span> gdc_target <span class="sc">|&gt;</span></span>
<span id="cb3-81"><a href="#cb3-81" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(</span>
<span id="cb3-82"><a href="#cb3-82" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sample_ID =</span> case_submitter_id,</span>
<span id="cb3-83"><a href="#cb3-83" aria-hidden="true" tabindex="-1"></a>    <span class="at">Age =</span> age_at_diagnosis,</span>
<span id="cb3-84"><a href="#cb3-84" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sex =</span> gender,</span>
<span id="cb3-85"><a href="#cb3-85" aria-hidden="true" tabindex="-1"></a>    <span class="at">OS_Time =</span> days_to_death,</span>
<span id="cb3-86"><a href="#cb3-86" aria-hidden="true" tabindex="-1"></a>    <span class="at">OS_Status =</span> vital_status</span>
<span id="cb3-87"><a href="#cb3-87" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb3-88"><a href="#cb3-88" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb3-89"><a href="#cb3-89" aria-hidden="true" tabindex="-1"></a>    Sample_ID, Age, Sex, OS_Time, OS_Status,</span>
<span id="cb3-90"><a href="#cb3-90" aria-hidden="true" tabindex="-1"></a>    project_id, race, primary_diagnosis</span>
<span id="cb3-91"><a href="#cb3-91" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb3-92"><a href="#cb3-92" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb3-93"><a href="#cb3-93" aria-hidden="true" tabindex="-1"></a>    <span class="at">Patient_ID =</span> Sample_ID,</span>
<span id="cb3-94"><a href="#cb3-94" aria-hidden="true" tabindex="-1"></a>    <span class="at">Age =</span> <span class="fu">as.integer</span>(<span class="fu">round</span>(<span class="fu">as.integer</span>(Age) <span class="sc">/</span> <span class="dv">365</span>)),</span>
<span id="cb3-95"><a href="#cb3-95" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sex =</span> <span class="fu">fcase</span>(Sex <span class="sc">%in%</span> <span class="st">"female"</span>, <span class="st">"F"</span>,</span>
<span id="cb3-96"><a href="#cb3-96" aria-hidden="true" tabindex="-1"></a>                Sex <span class="sc">%in%</span> <span class="st">"male"</span>, <span class="st">"M"</span>, <span class="at">default =</span> <span class="cn">NA_character_</span>),</span>
<span id="cb3-97"><a href="#cb3-97" aria-hidden="true" tabindex="-1"></a>    <span class="at">OS_Time =</span> <span class="fu">as.integer</span>(OS_Time),</span>
<span id="cb3-98"><a href="#cb3-98" aria-hidden="true" tabindex="-1"></a>    <span class="at">OS_Status =</span> <span class="fu">fcase</span>(OS_Status <span class="sc">%in%</span> <span class="st">"Alive"</span>, <span class="dv">0</span>L,</span>
<span id="cb3-99"><a href="#cb3-99" aria-hidden="true" tabindex="-1"></a>                      OS_Status <span class="sc">%in%</span> <span class="st">"Dead"</span>, <span class="dv">1</span>L, <span class="at">default =</span> <span class="cn">NA_integer_</span>),</span>
<span id="cb3-100"><a href="#cb3-100" aria-hidden="true" tabindex="-1"></a>    <span class="at">OS_Time =</span> <span class="fu">ifelse</span>(OS_Time <span class="sc">&lt;</span> <span class="dv">0</span>, <span class="cn">NA_integer_</span>, OS_Time)</span>
<span id="cb3-101"><a href="#cb3-101" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb3-102"><a href="#cb3-102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.table</span>()</span>
<span id="cb3-103"><a href="#cb3-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-104"><a href="#cb3-104" aria-hidden="true" tabindex="-1"></a><span class="co">#unique()</span></span>
<span id="cb3-105"><a href="#cb3-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-106"><a href="#cb3-106" aria-hidden="true" tabindex="-1"></a><span class="co"># exist  a large number of case level  replicate</span></span>
<span id="cb3-107"><a href="#cb3-107" aria-hidden="true" tabindex="-1"></a>gdc_target2[<span class="fu">duplicated</span>(gdc_target2), ]</span>
<span id="cb3-108"><a href="#cb3-108" aria-hidden="true" tabindex="-1"></a><span class="fu">duplicated</span>(gdc_target[case_submitter_id <span class="sc">==</span> <span class="st">"TARGET-20-PANTIV"</span>])</span>
<span id="cb3-109"><a href="#cb3-109" aria-hidden="true" tabindex="-1"></a>gdc_target[case_submitter_id <span class="sc">==</span> <span class="st">"TARGET-20-PANTIV"</span>] <span class="sc">|&gt;</span> <span class="fu">View</span>() <span class="co"># A few records that have been updated？</span></span>
<span id="cb3-110"><a href="#cb3-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-111"><a href="#cb3-111" aria-hidden="true" tabindex="-1"></a>gdc_target2 <span class="ot">=</span> gdc_target2[<span class="sc">!</span><span class="fu">duplicated</span>(gdc_target2)]</span>
<span id="cb3-112"><a href="#cb3-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-113"><a href="#cb3-113" aria-hidden="true" tabindex="-1"></a><span class="co">#table(gdc_target2$PFS_Time)</span></span>
<span id="cb3-114"><a href="#cb3-114" aria-hidden="true" tabindex="-1"></a><span class="fu">any</span>(<span class="fu">duplicated</span>(gdc_target2))</span>
<span id="cb3-115"><a href="#cb3-115" aria-hidden="true" tabindex="-1"></a><span class="fu">any</span>(<span class="fu">duplicated</span>(gdc_target2<span class="sc">$</span>Sample_ID))</span>
<span id="cb3-116"><a href="#cb3-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-117"><a href="#cb3-117" aria-hidden="true" tabindex="-1"></a>gdc_target2[<span class="fu">duplicated</span>(gdc_target2<span class="sc">$</span>Sample_ID)]</span>
<span id="cb3-118"><a href="#cb3-118" aria-hidden="true" tabindex="-1"></a><span class="co"># They may exist in different ALL projects.</span></span>
<span id="cb3-119"><a href="#cb3-119" aria-hidden="true" tabindex="-1"></a>gdc_dup <span class="ot">=</span> gdc_target2[Sample_ID <span class="sc">%in%</span> Sample_ID[<span class="fu">duplicated</span>(Sample_ID)]]</span>
<span id="cb3-120"><a href="#cb3-120" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(gdc_dup<span class="sc">$</span>project_id)</span>
<span id="cb3-121"><a href="#cb3-121" aria-hidden="true" tabindex="-1"></a><span class="co"># TARGET-ALL-P1 TARGET-ALL-P2 TARGET-ALL-P3</span></span>
<span id="cb3-122"><a href="#cb3-122" aria-hidden="true" tabindex="-1"></a><span class="co"># 10            57            47</span></span>
<span id="cb3-123"><a href="#cb3-123" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb3-124"><a href="#cb3-124" aria-hidden="true" tabindex="-1"></a><span class="co"># For now, we won't address this issue. If we conduct a combined analysis of ALL, we will consider removing duplicates.</span></span>
<span id="cb3-125"><a href="#cb3-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-126"><a href="#cb3-126" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(gdc_target2)</span>
<span id="cb3-127"><a href="#cb3-127" aria-hidden="true" tabindex="-1"></a>gdc_target2 <span class="ot">=</span> gdc_target2[, <span class="fu">c</span>(<span class="st">"Patient_ID"</span>, <span class="st">"Sample_ID"</span>, <span class="st">"Age"</span>, <span class="st">"Sex"</span>, <span class="st">"race"</span>, <span class="st">"primary_diagnosis"</span>, <span class="st">"OS_Time"</span>, <span class="st">"OS_Status"</span>, <span class="st">"project_id"</span>)]</span>
<span id="cb3-128"><a href="#cb3-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-129"><a href="#cb3-129" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(gdc_target2, <span class="at">file =</span> <span class="st">"/home/data2/Projects/Fusion/fusiondb/Clininfo/TARGET_info.tsv"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb3-130"><a href="#cb3-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-131"><a href="#cb3-131" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="sc">!</span><span class="fu">is.na</span>(gdc_target2<span class="sc">$</span>OS_Time))</span>
<span id="cb3-132"><a href="#cb3-132" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="sc">!</span><span class="fu">is.na</span>(cli1<span class="sc">$</span><span class="st">`</span><span class="at">_OS</span><span class="st">`</span>))</span>
<span id="cb3-133"><a href="#cb3-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-134"><a href="#cb3-134" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(<span class="fu">na.omit</span>(<span class="fu">unique</span>(target_cli[, .(Patient_ID, OS_Time, OS_Status)])))</span>
<span id="cb3-135"><a href="#cb3-135" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(<span class="fu">na.omit</span>(<span class="fu">unique</span>(gdc_target2[, .(Patient_ID, OS_Time, OS_Status)])))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="tcgatargetcptac-clinicaliobrtmegene-data-divide-and-anotate-preprocessing" class="level2">
<h2 class="anchored" data-anchor-id="tcgatargetcptac-clinicaliobrtmegene-data-divide-and-anotate-preprocessing">TCGA/TARGET/CPTAC Clinical/IOBR/TME/GENE data divide and anotate preprocessing</h2>
<section id="target-1" class="level3">
<h3 class="anchored" data-anchor-id="target-1">TARGET</h3>
<div class="sourceCode" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>file_meta <span class="ot">&lt;-</span> <span class="st">"/home/data2/Projects/Fusion/fusiondb_v3/metafusion_calls_matched.cptac.tsv"</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>metafusion <span class="ot">&lt;-</span> <span class="fu">fread</span>(file_meta, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="co"># TARGET</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="do">## info</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="do">### Because some projects are not included in the target_sheet, we should abandon them.</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>TARGET_sheet <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/home/data2/Projects/Fusion/unclean/TARGET_gdc_sample_sheet.2023-11-15.tsv"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>project_choose <span class="ot">&lt;-</span> <span class="fu">substr</span>(TARGET_sheet<span class="sc">$</span><span class="st">`</span><span class="at">Sample ID</span><span class="st">`</span>, <span class="dv">1</span>, <span class="dv">9</span>) <span class="sc">|&gt;</span> <span class="fu">unique</span>()</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>project_choose</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a><span class="do">### "TARGET-20" "TARGET-00" "TARGET-21" "TARGET-30"</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a><span class="do">### TARGET_sheet pretreatment</span></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a><span class="do">### Composite Sample </span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a><span class="do">### Composite Sample use comma to separate like TARGET-30-PAPEAV, TARGET-30-PAPTFZ</span></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>TARGET_Composite <span class="ot">&lt;-</span> TARGET_sheet <span class="sc">|&gt;</span> </span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">", "</span>, <span class="st">`</span><span class="at">Sample ID</span><span class="st">`</span>)) <span class="sc">|&gt;</span>  </span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="st">`</span><span class="at">Sample ID</span><span class="st">`</span>) <span class="sc">|&gt;</span></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>TARGET_Composite</span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a><span class="do">### 1: TARGET-30-PAPEAV-01A, TARGET-30-PAPTFZ-01A</span></span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a><span class="do">### 2: TARGET-30-PANUKV-01A, TARGET-30-PASUML-01A</span></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a><span class="do">### 3: TARGET-30-PANKFE-01A, TARGET-30-PAPUAR-01A</span></span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a><span class="do">### 4: TARGET-30-PASYPX-01A, TARGET-30-PAIXIF-01A</span></span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a><span class="do">### For the CPTAC databases, check whether there is any overlap between composite and normal Sample_ID in the info data. </span></span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a>check_TARGET_overlap <span class="ot">&lt;-</span> TARGET_sheet <span class="sc">|&gt;</span>  </span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(<span class="st">`</span><span class="at">Case ID</span><span class="st">`</span>, <span class="st">`</span><span class="at">Sample ID</span><span class="st">`</span>, <span class="st">`</span><span class="at">Sample Type</span><span class="st">`</span>, <span class="st">`</span><span class="at">Project ID</span><span class="st">`</span>)) <span class="sc">|&gt;</span></span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>() <span class="sc">|&gt;</span> </span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate_rows</span>(<span class="st">`</span><span class="at">Sample ID</span><span class="st">`</span>, <span class="at">sep =</span> <span class="st">",</span><span class="sc">\\</span><span class="st">s*"</span>) </span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a>check_TARGET_overlap <span class="ot">&lt;-</span> check_TARGET_overlap <span class="sc">|&gt;</span></span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample_Type =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">", "</span>, <span class="st">`</span><span class="at">Sample ID</span><span class="st">`</span>), <span class="st">"Composite"</span>, <span class="st">"Normal"</span>))</span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a>overlap_samples <span class="ot">&lt;-</span> check_TARGET_overlap <span class="sc">|&gt;</span></span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(<span class="st">`</span><span class="at">Sample ID</span><span class="st">`</span>) <span class="sc">|&gt;</span></span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">Count =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">'drop'</span>) <span class="sc">|&gt;</span></span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Count <span class="sc">&gt;</span> <span class="dv">1</span>)</span>
<span id="cb4-45"><a href="#cb4-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-46"><a href="#cb4-46" aria-hidden="true" tabindex="-1"></a>overlap_samples</span>
<span id="cb4-47"><a href="#cb4-47" aria-hidden="true" tabindex="-1"></a><span class="do">###  A tibble: 4 × 2</span></span>
<span id="cb4-48"><a href="#cb4-48" aria-hidden="true" tabindex="-1"></a><span class="do">###  `Sample ID`          Count</span></span>
<span id="cb4-49"><a href="#cb4-49" aria-hidden="true" tabindex="-1"></a><span class="do">###  &lt;chr&gt;                &lt;int&gt;</span></span>
<span id="cb4-50"><a href="#cb4-50" aria-hidden="true" tabindex="-1"></a><span class="do">###  1 TARGET-30-PANKFE-01A     2</span></span>
<span id="cb4-51"><a href="#cb4-51" aria-hidden="true" tabindex="-1"></a><span class="do">###  2 TARGET-30-PAPTFZ-01A     2</span></span>
<span id="cb4-52"><a href="#cb4-52" aria-hidden="true" tabindex="-1"></a><span class="do">###  3 TARGET-30-PASUML-01A     2</span></span>
<span id="cb4-53"><a href="#cb4-53" aria-hidden="true" tabindex="-1"></a><span class="do">###  4 TARGET-30-PASYPX-01A     2</span></span>
<span id="cb4-54"><a href="#cb4-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-55"><a href="#cb4-55" aria-hidden="true" tabindex="-1"></a><span class="do">### So we choose abandon them.</span></span>
<span id="cb4-56"><a href="#cb4-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-57"><a href="#cb4-57" aria-hidden="true" tabindex="-1"></a>TARGET_pretreatment <span class="ot">&lt;-</span> TARGET_sheet <span class="sc">|&gt;</span> </span>
<span id="cb4-58"><a href="#cb4-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(<span class="st">`</span><span class="at">Case ID</span><span class="st">`</span>, <span class="st">`</span><span class="at">Sample ID</span><span class="st">`</span>, <span class="st">`</span><span class="at">Sample Type</span><span class="st">`</span>)) <span class="sc">|&gt;</span></span>
<span id="cb4-59"><a href="#cb4-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">"Patient_ID"</span> <span class="ot">=</span> <span class="st">`</span><span class="at">Case ID</span><span class="st">`</span>,</span>
<span id="cb4-60"><a href="#cb4-60" aria-hidden="true" tabindex="-1"></a>         <span class="st">"Sample_ID"</span> <span class="ot">=</span> <span class="st">`</span><span class="at">Sample ID</span><span class="st">`</span>,</span>
<span id="cb4-61"><a href="#cb4-61" aria-hidden="true" tabindex="-1"></a>         <span class="st">"Sample_type"</span>  <span class="ot">=</span>  <span class="st">`</span><span class="at">Sample Type</span><span class="st">`</span>) <span class="sc">|&gt;</span> </span>
<span id="cb4-62"><a href="#cb4-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">","</span>, Sample_ID)) <span class="sc">|&gt;</span> </span>
<span id="cb4-63"><a href="#cb4-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"^TARGET-(20|00|21|30)"</span>, Sample_ID)) <span class="sc">|&gt;</span> </span>
<span id="cb4-64"><a href="#cb4-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb4-65"><a href="#cb4-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-66"><a href="#cb4-66" aria-hidden="true" tabindex="-1"></a><span class="do">### TARGET_xena_info pretreatment</span></span>
<span id="cb4-67"><a href="#cb4-67" aria-hidden="true" tabindex="-1"></a><span class="do">### This data use</span></span>
<span id="cb4-68"><a href="#cb4-68" aria-hidden="true" tabindex="-1"></a>TARGET_xena_info <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/home/data2/Projects/Fusion/fusiondb/Clininfo/TARGET_xena_info.tsv"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>) </span>
<span id="cb4-69"><a href="#cb4-69" aria-hidden="true" tabindex="-1"></a>TARGET_xena_pretreatment   <span class="ot">&lt;-</span> TARGET_xena_info <span class="sc">|&gt;</span></span>
<span id="cb4-70"><a href="#cb4-70" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">mutate</span>(<span class="at">Patient_ID =</span> <span class="fu">str_extract</span>(Sample_ID, <span class="st">"^[^-]+-[^-]+-[^-]+"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb4-71"><a href="#cb4-71" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"^TARGET-(20|00|21|30)"</span>, Sample_ID)) <span class="sc">|&gt;</span></span>
<span id="cb4-72"><a href="#cb4-72" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">mutate</span>(<span class="at">Age =</span> <span class="fu">round</span>(Age, <span class="at">digits =</span> <span class="dv">0</span>),</span>
<span id="cb4-73"><a href="#cb4-73" aria-hidden="true" tabindex="-1"></a>                                     <span class="fu">across</span>(<span class="fu">c</span>(Age, OS_Time, OS_Status), <span class="sc">~</span> <span class="fu">as.integer</span>(.))) <span class="sc">|&gt;</span></span>
<span id="cb4-74"><a href="#cb4-74" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">rename</span>(<span class="at">Cancer =</span> disease_code)</span>
<span id="cb4-75"><a href="#cb4-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-76"><a href="#cb4-76" aria-hidden="true" tabindex="-1"></a><span class="do">### TARGET_info pretreatment</span></span>
<span id="cb4-77"><a href="#cb4-77" aria-hidden="true" tabindex="-1"></a><span class="do">### In the TARGET database, rows with the info column marked as '--' are annotated as NA.</span></span>
<span id="cb4-78"><a href="#cb4-78" aria-hidden="true" tabindex="-1"></a>filter_lose <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb4-79"><a href="#cb4-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">is.na</span>(x) <span class="sc">|</span> x <span class="sc">==</span> <span class="st">""</span> <span class="sc">|</span> x <span class="sc">==</span> <span class="st">"--"</span> <span class="sc">|</span> x <span class="sc">==</span> <span class="st">"Unknown"</span> <span class="sc">|</span> x <span class="sc">==</span> <span class="st">"'--"</span></span>
<span id="cb4-80"><a href="#cb4-80" aria-hidden="true" tabindex="-1"></a> }</span>
<span id="cb4-81"><a href="#cb4-81" aria-hidden="true" tabindex="-1"></a>TARGET_info <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/home/data2/Projects/Fusion/fusiondb/Clininfo/TARGET_info.tsv"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>) </span>
<span id="cb4-82"><a href="#cb4-82" aria-hidden="true" tabindex="-1"></a>replacement_table2 <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb4-83"><a href="#cb4-83" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Acute lymphocytic leukemia"</span> <span class="ot">=</span>  <span class="st">"TARGET-ALL"</span>,             </span>
<span id="cb4-84"><a href="#cb4-84" aria-hidden="true" tabindex="-1"></a>  <span class="st">"'--"</span>   <span class="ot">=</span> <span class="st">"TARGET-Unkown"</span>,                                  </span>
<span id="cb4-85"><a href="#cb4-85" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Precursor B-cell lymphoblastic leukemia"</span> <span class="ot">=</span> <span class="st">"TARGET-B-ALL"</span>, </span>
<span id="cb4-86"><a href="#cb4-86" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Not Reported"</span>  <span class="ot">=</span> <span class="st">"TARGET-Unkown"</span>,                         </span>
<span id="cb4-87"><a href="#cb4-87" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Acute myeloid leukemia, NOS"</span> <span class="ot">=</span> <span class="st">"TARGET-LAML"</span>,             </span>
<span id="cb4-88"><a href="#cb4-88" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Clear cell sarcoma of kidney"</span> <span class="ot">=</span> <span class="st">"TARGET-CCSK"</span>,           </span>
<span id="cb4-89"><a href="#cb4-89" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Neuroblastoma, NOS"</span>  <span class="ot">=</span> <span class="st">"TARGET-NBL"</span>,                     </span>
<span id="cb4-90"><a href="#cb4-90" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Ganglioneuroblastoma"</span> <span class="ot">=</span> <span class="st">"TARGET-GNB"</span>,                   </span>
<span id="cb4-91"><a href="#cb4-91" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Osteosarcoma, NOS"</span>   <span class="ot">=</span> <span class="st">"TARGET-OS"</span>,                    </span>
<span id="cb4-92"><a href="#cb4-92" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Malignant rhabdoid tumor"</span> <span class="ot">=</span> <span class="st">"TARGET-MRT"</span>,               </span>
<span id="cb4-93"><a href="#cb4-93" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Wilms tumor"</span> <span class="ot">=</span> <span class="st">"TARGET-WT"</span>  </span>
<span id="cb4-94"><a href="#cb4-94" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-95"><a href="#cb4-95" aria-hidden="true" tabindex="-1"></a>matched2 <span class="ot">&lt;-</span> TARGET_info<span class="sc">$</span>primary_diagnosis <span class="sc">%in%</span> <span class="fu">names</span>(replacement_table2)</span>
<span id="cb4-96"><a href="#cb4-96" aria-hidden="true" tabindex="-1"></a>TARGET_info_pretreatment <span class="ot">&lt;-</span> TARGET_info <span class="sc">|&gt;</span></span>
<span id="cb4-97"><a href="#cb4-97" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">mutate</span>(<span class="at">cohort =</span> <span class="fu">case_when</span>(</span>
<span id="cb4-98"><a href="#cb4-98" aria-hidden="true" tabindex="-1"></a>                                   matched2 <span class="sc">~</span> replacement_table2[primary_diagnosis],</span>
<span id="cb4-99"><a href="#cb4-99" aria-hidden="true" tabindex="-1"></a>                                   <span class="cn">TRUE</span> <span class="sc">~</span> <span class="fu">as.character</span>(<span class="cn">NA</span>)  </span>
<span id="cb4-100"><a href="#cb4-100" aria-hidden="true" tabindex="-1"></a>                                   )) <span class="sc">|&gt;</span></span>
<span id="cb4-101"><a href="#cb4-101" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">rename</span>(<span class="at">Cancer =</span> primary_diagnosis,</span>
<span id="cb4-102"><a href="#cb4-102" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">Project_id =</span> project_id) <span class="sc">|&gt;</span> </span>
<span id="cb4-103"><a href="#cb4-103" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"^TARGET-(20|00|21|30)"</span>, Sample_ID)) <span class="sc">|&gt;</span> </span>
<span id="cb4-104"><a href="#cb4-104" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(<span class="st">"Age"</span>, <span class="st">"Sex"</span>, <span class="st">"OS_Time"</span>, <span class="st">"OS_Status"</span>, <span class="st">"race"</span>), <span class="sc">~</span> <span class="fu">ifelse</span>(<span class="fu">filter_lose</span>(.x), <span class="cn">NA</span>, .x))) <span class="sc">|&gt;</span> </span>
<span id="cb4-105"><a href="#cb4-105" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">mutate</span>(<span class="at">Cancer =</span> <span class="fu">ifelse</span>(Cancer <span class="sc">==</span> <span class="st">"'--"</span>, <span class="cn">NA</span>, Cancer))</span>
<span id="cb4-106"><a href="#cb4-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-107"><a href="#cb4-107" aria-hidden="true" tabindex="-1"></a><span class="do">### Check if the valid rows in TARGET_info match the valid rows in TARGET_xena_info.</span></span>
<span id="cb4-108"><a href="#cb4-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-109"><a href="#cb4-109" aria-hidden="true" tabindex="-1"></a><span class="do">### 1.Filter the valid rows.</span></span>
<span id="cb4-110"><a href="#cb4-110" aria-hidden="true" tabindex="-1"></a>filter_valid <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb4-111"><a href="#cb4-111" aria-hidden="true" tabindex="-1"></a>  <span class="sc">!</span><span class="fu">is.na</span>(x) <span class="sc">&amp;</span> x <span class="sc">!=</span> <span class="st">""</span> <span class="sc">&amp;</span> x <span class="sc">!=</span> <span class="st">"--"</span> <span class="sc">&amp;</span> x <span class="sc">!=</span> <span class="st">"Unknown"</span> <span class="sc">&amp;</span> x <span class="sc">==</span> <span class="st">"'--"</span></span>
<span id="cb4-112"><a href="#cb4-112" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-113"><a href="#cb4-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-114"><a href="#cb4-114" aria-hidden="true" tabindex="-1"></a><span class="do">### Filter the valid rows in TARGET_info.</span></span>
<span id="cb4-115"><a href="#cb4-115" aria-hidden="true" tabindex="-1"></a>TARGET_info_valid <span class="ot">&lt;-</span> TARGET_info_pretreatment <span class="sc">%&gt;%</span></span>
<span id="cb4-116"><a href="#cb4-116" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">filter</span>(<span class="fu">filter_valid</span>(Age) <span class="sc">&amp;</span> <span class="fu">filter_valid</span>(OS_Time) <span class="sc">&amp;</span> <span class="fu">filter_valid</span>(OS_Status) <span class="sc">&amp;</span> <span class="fu">filter_valid</span>(Sex)) <span class="sc">|&gt;</span></span>
<span id="cb4-117"><a href="#cb4-117" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">select</span>(<span class="fu">c</span>(Patient_ID, Age, OS_Time, OS_Status, Sex))</span>
<span id="cb4-118"><a href="#cb4-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-119"><a href="#cb4-119" aria-hidden="true" tabindex="-1"></a><span class="do">### Filter the valid rows in TARGET_xena_info.</span></span>
<span id="cb4-120"><a href="#cb4-120" aria-hidden="true" tabindex="-1"></a>TARGET_xena_info_valid <span class="ot">&lt;-</span> TARGET_xena_pretreatment <span class="sc">%&gt;%</span></span>
<span id="cb4-121"><a href="#cb4-121" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">filter</span>(<span class="fu">filter_valid</span>(Age) <span class="sc">&amp;</span> <span class="fu">filter_valid</span>(OS_Time) <span class="sc">&amp;</span> <span class="fu">filter_valid</span>(OS_Status) <span class="sc">&amp;</span> <span class="fu">filter_valid</span>(Sex)) <span class="sc">|&gt;</span></span>
<span id="cb4-122"><a href="#cb4-122" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">select</span>(<span class="fu">c</span>(Patient_ID, Age, OS_Time, OS_Status, Sex))</span>
<span id="cb4-123"><a href="#cb4-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-124"><a href="#cb4-124" aria-hidden="true" tabindex="-1"></a><span class="do">### 2. Merge the two data frames based on Patient_ID.</span></span>
<span id="cb4-125"><a href="#cb4-125" aria-hidden="true" tabindex="-1"></a>merged_data <span class="ot">&lt;-</span> TARGET_info_valid <span class="sc">%&gt;%</span></span>
<span id="cb4-126"><a href="#cb4-126" aria-hidden="true" tabindex="-1"></a>               <span class="fu">inner_join</span>(TARGET_xena_info_valid, <span class="at">by =</span> <span class="st">"Patient_ID"</span>, <span class="at">suffix =</span> <span class="fu">c</span>(<span class="st">"_info"</span>, <span class="st">"_xena"</span>))</span>
<span id="cb4-127"><a href="#cb4-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-128"><a href="#cb4-128" aria-hidden="true" tabindex="-1"></a><span class="do">### 3. Check whether the valid values in the merged data frame are consistent.</span></span>
<span id="cb4-129"><a href="#cb4-129" aria-hidden="true" tabindex="-1"></a>is_consistent <span class="ot">&lt;-</span> merged_data <span class="sc">%&gt;%</span></span>
<span id="cb4-130"><a href="#cb4-130" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-131"><a href="#cb4-131" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb4-132"><a href="#cb4-132" aria-hidden="true" tabindex="-1"></a>    <span class="at">Age_consistent =</span> Age_info <span class="sc">==</span> Age_xena,</span>
<span id="cb4-133"><a href="#cb4-133" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sex_consistent =</span> Sex_info <span class="sc">==</span> Sex_xena,</span>
<span id="cb4-134"><a href="#cb4-134" aria-hidden="true" tabindex="-1"></a>    <span class="at">OS_Time_consistent =</span> OS_Time_info <span class="sc">==</span> OS_Time_xena,</span>
<span id="cb4-135"><a href="#cb4-135" aria-hidden="true" tabindex="-1"></a>    <span class="at">OS_Status_consistent =</span> OS_Status_info <span class="sc">==</span> OS_Status_xena</span>
<span id="cb4-136"><a href="#cb4-136" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb4-137"><a href="#cb4-137" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb4-138"><a href="#cb4-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-139"><a href="#cb4-139" aria-hidden="true" tabindex="-1"></a><span class="do">### 4. Age not onsistent</span></span>
<span id="cb4-140"><a href="#cb4-140" aria-hidden="true" tabindex="-1"></a>Age_not_consistent <span class="ot">&lt;-</span> is_consistent <span class="sc">|&gt;</span> </span>
<span id="cb4-141"><a href="#cb4-141" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">filter</span>(Age_consistent <span class="sc">==</span> <span class="st">"FALSE"</span>) </span>
<span id="cb4-142"><a href="#cb4-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-143"><a href="#cb4-143" aria-hidden="true" tabindex="-1"></a><span class="do">### 5. Sex not onsistent</span></span>
<span id="cb4-144"><a href="#cb4-144" aria-hidden="true" tabindex="-1"></a>Sex_not_consistent <span class="ot">&lt;-</span> is_consistent <span class="sc">|&gt;</span> </span>
<span id="cb4-145"><a href="#cb4-145" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">filter</span>(Sex_consistent <span class="sc">==</span> <span class="st">"FALSE"</span>) </span>
<span id="cb4-146"><a href="#cb4-146" aria-hidden="true" tabindex="-1"></a><span class="do">### Sex_not_consistent</span></span>
<span id="cb4-147"><a href="#cb4-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-148"><a href="#cb4-148" aria-hidden="true" tabindex="-1"></a><span class="do">###  A tibble: 0 × 13</span></span>
<span id="cb4-149"><a href="#cb4-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-150"><a href="#cb4-150" aria-hidden="true" tabindex="-1"></a><span class="do">### 6. OS_Time not onsistent</span></span>
<span id="cb4-151"><a href="#cb4-151" aria-hidden="true" tabindex="-1"></a>OS_Time_not_consistent <span class="ot">&lt;-</span> is_consistent <span class="sc">|&gt;</span> </span>
<span id="cb4-152"><a href="#cb4-152" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">filter</span>(OS_Time_consistent <span class="sc">==</span> <span class="st">"FALSE"</span>) <span class="sc">|&gt;</span></span>
<span id="cb4-153"><a href="#cb4-153" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">mutate</span>(</span>
<span id="cb4-154"><a href="#cb4-154" aria-hidden="true" tabindex="-1"></a>                          <span class="at">OS_Time_info_greater =</span> OS_Time_info <span class="sc">&gt;</span> OS_Time_xena,  </span>
<span id="cb4-155"><a href="#cb4-155" aria-hidden="true" tabindex="-1"></a>                          <span class="at">OS_Time_xena_greater =</span> OS_Time_xena <span class="sc">&gt;</span> OS_Time_info,  </span>
<span id="cb4-156"><a href="#cb4-156" aria-hidden="true" tabindex="-1"></a>                          <span class="at">OS_Time_equal =</span> OS_Time_info <span class="sc">==</span> OS_Time_xena         </span>
<span id="cb4-157"><a href="#cb4-157" aria-hidden="true" tabindex="-1"></a>                          )</span>
<span id="cb4-158"><a href="#cb4-158" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(OS_Time_not_consistent <span class="sc">|&gt;</span> <span class="fu">filter</span>(OS_Time_info_greater <span class="sc">==</span> <span class="st">"FALSE"</span>))</span>
<span id="cb4-159"><a href="#cb4-159" aria-hidden="true" tabindex="-1"></a><span class="do">### [1]0</span></span>
<span id="cb4-160"><a href="#cb4-160" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(OS_Time_not_consistent <span class="sc">|&gt;</span> <span class="fu">filter</span>(OS_Time_xena_greater <span class="sc">==</span> <span class="st">"TRUE"</span>))</span>
<span id="cb4-161"><a href="#cb4-161" aria-hidden="true" tabindex="-1"></a><span class="do">### [1]0</span></span>
<span id="cb4-162"><a href="#cb4-162" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(OS_Time_not_consistent <span class="sc">|&gt;</span> <span class="fu">filter</span>(OS_Time_equal <span class="sc">==</span> <span class="st">"TRUE"</span>))</span>
<span id="cb4-163"><a href="#cb4-163" aria-hidden="true" tabindex="-1"></a><span class="do">### [1]0</span></span>
<span id="cb4-164"><a href="#cb4-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-165"><a href="#cb4-165" aria-hidden="true" tabindex="-1"></a><span class="do">### 7. OS_Status not onsistent</span></span>
<span id="cb4-166"><a href="#cb4-166" aria-hidden="true" tabindex="-1"></a>OS_Status_not_consistent <span class="ot">&lt;-</span> is_consistent <span class="sc">|&gt;</span> </span>
<span id="cb4-167"><a href="#cb4-167" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">filter</span>(OS_Status_consistent <span class="sc">==</span> <span class="st">"FALSE"</span>) </span>
<span id="cb4-168"><a href="#cb4-168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-169"><a href="#cb4-169" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(OS_Status_not_consistent<span class="sc">$</span>OS_Status_info)</span>
<span id="cb4-170"><a href="#cb4-170" aria-hidden="true" tabindex="-1"></a><span class="do">### [1] 1</span></span>
<span id="cb4-171"><a href="#cb4-171" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(OS_Status_not_consistent<span class="sc">$</span>OS_Status_xena)</span>
<span id="cb4-172"><a href="#cb4-172" aria-hidden="true" tabindex="-1"></a><span class="do">### [1] 0</span></span>
<span id="cb4-173"><a href="#cb4-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-174"><a href="#cb4-174" aria-hidden="true" tabindex="-1"></a><span class="do">### All evidence shows that the GDC data has been updated more recently than the Xena data.</span></span>
<span id="cb4-175"><a href="#cb4-175" aria-hidden="true" tabindex="-1"></a><span class="do">### Therefore, if there is a conflict between GDC data and Xena data, we should use the GDC data.</span></span>
<span id="cb4-176"><a href="#cb4-176" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-177"><a href="#cb4-177" aria-hidden="true" tabindex="-1"></a><span class="do">### Use the valid values from TARGET_xena_info to fill the missing values in TARGET_info.</span></span>
<span id="cb4-178"><a href="#cb4-178" aria-hidden="true" tabindex="-1"></a><span class="do">### 1: Determine the matching Patient_ID.</span></span>
<span id="cb4-179"><a href="#cb4-179" aria-hidden="true" tabindex="-1"></a>matched_ids <span class="ot">&lt;-</span> <span class="fu">intersect</span>(TARGET_xena_pretreatment<span class="sc">$</span>Patient_ID, TARGET_info_pretreatment<span class="sc">$</span>Patient_ID)</span>
<span id="cb4-180"><a href="#cb4-180" aria-hidden="true" tabindex="-1"></a><span class="fu">setDT</span>(TARGET_xena_pretreatment)</span>
<span id="cb4-181"><a href="#cb4-181" aria-hidden="true" tabindex="-1"></a><span class="fu">setDT</span>(TARGET_info_pretreatment)</span>
<span id="cb4-182"><a href="#cb4-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-183"><a href="#cb4-183" aria-hidden="true" tabindex="-1"></a><span class="do">### 2: Find the columns that require updates.</span></span>
<span id="cb4-184"><a href="#cb4-184" aria-hidden="true" tabindex="-1"></a>cols_to_update <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">names</span>(TARGET_info_pretreatment), <span class="fu">c</span>(<span class="st">"Sample_ID"</span>, <span class="st">"Patient_ID"</span>, <span class="st">"Cancer"</span>,<span class="st">"project_id"</span>,<span class="st">"cohort"</span>,<span class="st">"race"</span>))</span>
<span id="cb4-185"><a href="#cb4-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-186"><a href="#cb4-186" aria-hidden="true" tabindex="-1"></a><span class="do">### 3.Iterate through each matching Patient_ID and update TARGET_info_pretreatment using TARGET_xena_pretreatment.</span></span>
<span id="cb4-187"><a href="#cb4-187" aria-hidden="true" tabindex="-1"></a><span class="do">### update Age</span></span>
<span id="cb4-188"><a href="#cb4-188" aria-hidden="true" tabindex="-1"></a>TARGET_xena_can  <span class="ot">&lt;-</span>  TARGET_xena_pretreatment <span class="sc">|&gt;</span> </span>
<span id="cb4-189"><a href="#cb4-189" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">filter</span>(<span class="fu">if_any</span>(<span class="st">"Age"</span>, <span class="sc">~</span> <span class="sc">!</span><span class="fu">filter_lose</span>(.x))) </span>
<span id="cb4-190"><a href="#cb4-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-191"><a href="#cb4-191" aria-hidden="true" tabindex="-1"></a>TARGET_info_match <span class="ot">&lt;-</span> TARGET_info_pretreatment <span class="sc">|&gt;</span></span>
<span id="cb4-192"><a href="#cb4-192" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">mutate</span>(<span class="at">Age =</span> <span class="fu">case_when</span>(</span>
<span id="cb4-193"><a href="#cb4-193" aria-hidden="true" tabindex="-1"></a>                                                   <span class="fu">filter_lose</span>(Age) <span class="sc">&amp;</span> Patient_ID <span class="sc">%in%</span> TARGET_xena_can<span class="sc">$</span>Patient_ID <span class="sc">~</span>  TARGET_xena_can<span class="sc">$</span>Age[<span class="fu">match</span>(Patient_ID, TARGET_xena_can<span class="sc">$</span>Patient_ID)],</span>
<span id="cb4-194"><a href="#cb4-194" aria-hidden="true" tabindex="-1"></a>                                                   <span class="cn">TRUE</span> <span class="sc">~</span> Age))</span>
<span id="cb4-195"><a href="#cb4-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-196"><a href="#cb4-196" aria-hidden="true" tabindex="-1"></a><span class="do">### update Sex</span></span>
<span id="cb4-197"><a href="#cb4-197" aria-hidden="true" tabindex="-1"></a>TARGET_xena_can  <span class="ot">&lt;-</span>  TARGET_xena_pretreatment <span class="sc">|&gt;</span> </span>
<span id="cb4-198"><a href="#cb4-198" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">filter</span>(<span class="fu">if_any</span>(<span class="st">"Sex"</span>, <span class="sc">~</span> <span class="sc">!</span><span class="fu">filter_lose</span>(.x))) </span>
<span id="cb4-199"><a href="#cb4-199" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-200"><a href="#cb4-200" aria-hidden="true" tabindex="-1"></a>TARGET_info_match1 <span class="ot">&lt;-</span> TARGET_info_match <span class="sc">|&gt;</span></span>
<span id="cb4-201"><a href="#cb4-201" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">mutate</span>(<span class="at">Sex =</span> <span class="fu">case_when</span>(</span>
<span id="cb4-202"><a href="#cb4-202" aria-hidden="true" tabindex="-1"></a>                                                   <span class="fu">filter_lose</span>(Sex) <span class="sc">&amp;</span> Patient_ID <span class="sc">%in%</span> TARGET_xena_can<span class="sc">$</span>Patient_ID <span class="sc">~</span>  TARGET_xena_can<span class="sc">$</span>Sex[<span class="fu">match</span>(Patient_ID, TARGET_xena_can<span class="sc">$</span>Patient_ID)],</span>
<span id="cb4-203"><a href="#cb4-203" aria-hidden="true" tabindex="-1"></a>                                                   <span class="cn">TRUE</span> <span class="sc">~</span> Sex))</span>
<span id="cb4-204"><a href="#cb4-204" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-205"><a href="#cb4-205" aria-hidden="true" tabindex="-1"></a><span class="do">### update OS_Time</span></span>
<span id="cb4-206"><a href="#cb4-206" aria-hidden="true" tabindex="-1"></a>TARGET_xena_can  <span class="ot">&lt;-</span>  TARGET_xena_pretreatment <span class="sc">|&gt;</span> </span>
<span id="cb4-207"><a href="#cb4-207" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">filter</span>(<span class="fu">if_any</span>(<span class="st">"OS_Time"</span>, <span class="sc">~</span> <span class="sc">!</span><span class="fu">filter_lose</span>(.x))) </span>
<span id="cb4-208"><a href="#cb4-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-209"><a href="#cb4-209" aria-hidden="true" tabindex="-1"></a>TARGET_info_match2 <span class="ot">&lt;-</span> TARGET_info_match1 <span class="sc">|&gt;</span></span>
<span id="cb4-210"><a href="#cb4-210" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">mutate</span>(<span class="at">OS_Time =</span> <span class="fu">case_when</span>(</span>
<span id="cb4-211"><a href="#cb4-211" aria-hidden="true" tabindex="-1"></a>                                                   <span class="fu">filter_lose</span>(OS_Time) <span class="sc">&amp;</span> Patient_ID <span class="sc">%in%</span> TARGET_xena_can<span class="sc">$</span>Patient_ID  <span class="sc">&amp;</span> OS_Status <span class="sc">%in%</span> TARGET_xena_can<span class="sc">$</span>OS_Status <span class="sc">~</span>  TARGET_xena_can<span class="sc">$</span>OS_Time[<span class="fu">match</span>(Patient_ID, TARGET_xena_can<span class="sc">$</span>Patient_ID)],</span>
<span id="cb4-212"><a href="#cb4-212" aria-hidden="true" tabindex="-1"></a>                                                   <span class="cn">TRUE</span> <span class="sc">~</span> OS_Time))</span>
<span id="cb4-213"><a href="#cb4-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-214"><a href="#cb4-214" aria-hidden="true" tabindex="-1"></a><span class="do">### update OS_Status</span></span>
<span id="cb4-215"><a href="#cb4-215" aria-hidden="true" tabindex="-1"></a>TARGET_xena_can  <span class="ot">&lt;-</span>  TARGET_xena_pretreatment <span class="sc">|&gt;</span> </span>
<span id="cb4-216"><a href="#cb4-216" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">filter</span>(<span class="fu">if_any</span>(<span class="st">"OS_Status"</span>, <span class="sc">~</span> <span class="sc">!</span><span class="fu">filter_lose</span>(.x))) </span>
<span id="cb4-217"><a href="#cb4-217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-218"><a href="#cb4-218" aria-hidden="true" tabindex="-1"></a>TARGET_info_after_match <span class="ot">&lt;-</span> TARGET_info_match2 <span class="sc">|&gt;</span></span>
<span id="cb4-219"><a href="#cb4-219" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">mutate</span>(<span class="at">OS_Status =</span> <span class="fu">case_when</span>(</span>
<span id="cb4-220"><a href="#cb4-220" aria-hidden="true" tabindex="-1"></a>                                                   <span class="fu">filter_lose</span>(OS_Status) <span class="sc">&amp;</span> Patient_ID <span class="sc">%in%</span> TARGET_xena_can<span class="sc">$</span>Patient_ID  <span class="sc">&amp;</span> OS_Time <span class="sc">%in%</span> TARGET_xena_can<span class="sc">$</span>OS_Time <span class="sc">~</span>  TARGET_xena_can<span class="sc">$</span>OS_Status[<span class="fu">match</span>(Patient_ID, TARGET_xena_can<span class="sc">$</span>Patient_ID)],</span>
<span id="cb4-221"><a href="#cb4-221" aria-hidden="true" tabindex="-1"></a>                                                   <span class="cn">TRUE</span> <span class="sc">~</span> OS_Status)) </span>
<span id="cb4-222"><a href="#cb4-222" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-223"><a href="#cb4-223" aria-hidden="true" tabindex="-1"></a><span class="do">### Use TARGET_sheet to perform a left join, ensuring that the Sample_ID is correct.</span></span>
<span id="cb4-224"><a href="#cb4-224" aria-hidden="true" tabindex="-1"></a>TARGET_info_after_match <span class="ot">&lt;-</span> TARGET_info_after_match <span class="sc">|&gt;</span> <span class="fu">select</span>(<span class="sc">-</span> Sample_ID)</span>
<span id="cb4-225"><a href="#cb4-225" aria-hidden="true" tabindex="-1"></a>TARGET_correct      <span class="ot">&lt;-</span> TARGET_pretreatment <span class="sc">|&gt;</span></span>
<span id="cb4-226"><a href="#cb4-226" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">left_join</span>(TARGET_info_after_match, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Patient_ID"</span> <span class="ot">=</span> <span class="st">"Patient_ID"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb4-227"><a href="#cb4-227" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">mutate</span>(<span class="at">Run =</span> Sample_ID)</span>
<span id="cb4-228"><a href="#cb4-228" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-229"><a href="#cb4-229" aria-hidden="true" tabindex="-1"></a><span class="do">### Replace the cohort column values with Project_ID where the row is TARGET-Unkown.</span></span>
<span id="cb4-230"><a href="#cb4-230" aria-hidden="true" tabindex="-1"></a><span class="do">### TARGET_correct$project_id |&gt; unique()</span></span>
<span id="cb4-231"><a href="#cb4-231" aria-hidden="true" tabindex="-1"></a><span class="do">### [1] "TARGET-AML" "TARGET-NBL"</span></span>
<span id="cb4-232"><a href="#cb4-232" aria-hidden="true" tabindex="-1"></a>TARGET_correct_after_pro <span class="ot">&lt;-</span> TARGET_correct  <span class="sc">%&gt;%</span></span>
<span id="cb4-233"><a href="#cb4-233" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb4-234"><a href="#cb4-234" aria-hidden="true" tabindex="-1"></a>    <span class="at">cohort =</span> <span class="fu">case_when</span>(</span>
<span id="cb4-235"><a href="#cb4-235" aria-hidden="true" tabindex="-1"></a>             cohort <span class="sc">==</span> <span class="st">"TARGET-Unkown"</span> <span class="sc">~</span> Project_id,</span>
<span id="cb4-236"><a href="#cb4-236" aria-hidden="true" tabindex="-1"></a>             <span class="cn">TRUE</span> <span class="sc">~</span> cohort),</span>
<span id="cb4-237"><a href="#cb4-237" aria-hidden="true" tabindex="-1"></a>    <span class="at">cohort =</span> <span class="fu">case_when</span>(</span>
<span id="cb4-238"><a href="#cb4-238" aria-hidden="true" tabindex="-1"></a>             cohort <span class="sc">==</span> <span class="st">"TARGET-AML"</span> <span class="sc">~</span> <span class="st">"TARGET-LAML"</span>,</span>
<span id="cb4-239"><a href="#cb4-239" aria-hidden="true" tabindex="-1"></a>             <span class="cn">TRUE</span> <span class="sc">~</span> cohort)</span>
<span id="cb4-240"><a href="#cb4-240" aria-hidden="true" tabindex="-1"></a>  ) </span>
<span id="cb4-241"><a href="#cb4-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-242"><a href="#cb4-242" aria-hidden="true" tabindex="-1"></a><span class="do">### Divide a part of data into a group based on whether the Sample_type is cell</span></span>
<span id="cb4-243"><a href="#cb4-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-244"><a href="#cb4-244" aria-hidden="true" tabindex="-1"></a>target_cell <span class="ot">&lt;-</span> TARGET_correct_after_pro <span class="sc">|&gt;</span> <span class="fu">filter</span>(<span class="st">`</span><span class="at">Sample_type</span><span class="st">`</span> <span class="sc">==</span> <span class="st">"Cell Lines"</span>)</span>
<span id="cb4-245"><a href="#cb4-245" aria-hidden="true" tabindex="-1"></a>target_cell_sample <span class="ot">&lt;-</span> target_cell<span class="sc">$</span>Sample_ID <span class="sc">|&gt;</span> <span class="fu">unique</span>()</span>
<span id="cb4-246"><a href="#cb4-246" aria-hidden="true" tabindex="-1"></a>target_cell_patient <span class="ot">&lt;-</span> target_cell<span class="sc">$</span>Patient_ID <span class="sc">|&gt;</span> <span class="fu">unique</span>()</span>
<span id="cb4-247"><a href="#cb4-247" aria-hidden="true" tabindex="-1"></a><span class="do">### target_cell_sample</span></span>
<span id="cb4-248"><a href="#cb4-248" aria-hidden="true" tabindex="-1"></a><span class="do">### [1] "TARGET-20-MUTZ3-50A"         "TARGET-20-ML1-50A"           "TARGET-20-MV411D1-50A"       "TARGET-20-KasumiD1-50A"    "TARGET-20-HEK293-50A"      "TARGET-20-MOLM14CBFGLIS-50A"     </span></span>
<span id="cb4-249"><a href="#cb4-249" aria-hidden="true" tabindex="-1"></a><span class="do">### [7] "TARGET-20-TF1-50A"           "TARGET-20-IGROV1-50A"        "TARGET-20-CMS-50A"           "TARGET-20-WSUAML-50A"      "TARGET-20-KasumiAZAD5-50A" "TARGET-20-KasumiAZAD11-50A"</span></span>
<span id="cb4-250"><a href="#cb4-250" aria-hidden="true" tabindex="-1"></a><span class="do">### [13]"TARGET-20-REH-50A"           "TARGET-20-MV411AZAD11-50A"   "TARGET-20-MV411AZAD5-50A"    "TARGET-20-THP1-50A"        "TARGET-20-OCIAML2-50A"     "TARGET-20-HL60-50A"  </span></span>
<span id="cb4-251"><a href="#cb4-251" aria-hidden="true" tabindex="-1"></a><span class="do">### target_cell_patient</span></span>
<span id="cb4-252"><a href="#cb4-252" aria-hidden="true" tabindex="-1"></a><span class="do">### [1] "TARGET-20-MUTZ3"         "TARGET-20-ML1"           "TARGET-20-MV411D1"       "TARGET-20-KasumiD1"      "TARGET-20-HEK293"        "TARGET-20-MOLM14CBFGLIS"</span></span>
<span id="cb4-253"><a href="#cb4-253" aria-hidden="true" tabindex="-1"></a><span class="do">### [7] "TARGET-20-TF1"           "TARGET-20-IGROV1"        "TARGET-20-CMS"           "TARGET-20-WSUAML"        "TARGET-20-KasumiAZAD5"   "TARGET-20-KasumiAZAD11" </span></span>
<span id="cb4-254"><a href="#cb4-254" aria-hidden="true" tabindex="-1"></a><span class="do">### [13]"TARGET-20-REH"           "TARGET-20-MV411AZAD11"   "TARGET-20-MV411AZAD5"    "TARGET-20-THP1"          "TARGET-20-OCIAML2"       "TARGET-20-HL60"  </span></span>
<span id="cb4-255"><a href="#cb4-255" aria-hidden="true" tabindex="-1"></a><span class="do">### reference : https://www.cellosaurus.org/</span></span>
<span id="cb4-256"><a href="#cb4-256" aria-hidden="true" tabindex="-1"></a><span class="do">### MUTZ3：Acute myeloid leukemia</span></span>
<span id="cb4-257"><a href="#cb4-257" aria-hidden="true" tabindex="-1"></a><span class="do">### ML1：    Adult acute myeloid leukemia</span></span>
<span id="cb4-258"><a href="#cb4-258" aria-hidden="true" tabindex="-1"></a><span class="do">### MV411D1：biphenotypic B-myelomonocytic leukemia</span></span>
<span id="cb4-259"><a href="#cb4-259" aria-hidden="true" tabindex="-1"></a><span class="do">### KasumiD1：Acute myeloblastic leukemia</span></span>
<span id="cb4-260"><a href="#cb4-260" aria-hidden="true" tabindex="-1"></a><span class="do">### TF1：Acute erythroid leukemia </span></span>
<span id="cb4-261"><a href="#cb4-261" aria-hidden="true" tabindex="-1"></a><span class="do">### IGROV1： Ovarian endometrioid adenocarcinoma</span></span>
<span id="cb4-262"><a href="#cb4-262" aria-hidden="true" tabindex="-1"></a><span class="do">### WSUAML：Acute myeloid leukemia。</span></span>
<span id="cb4-263"><a href="#cb4-263" aria-hidden="true" tabindex="-1"></a><span class="do">### KasumiAZAD5：can't find, so KasumiD1：Acute myeloblastic leukemia</span></span>
<span id="cb4-264"><a href="#cb4-264" aria-hidden="true" tabindex="-1"></a><span class="do">### KasumiAZAD11：can't find, so KasumiD1：Acute myeloblastic leukemia</span></span>
<span id="cb4-265"><a href="#cb4-265" aria-hidden="true" tabindex="-1"></a><span class="do">### REH：Childhood B acute lymphoblastic leukemia</span></span>
<span id="cb4-266"><a href="#cb4-266" aria-hidden="true" tabindex="-1"></a><span class="do">### MV411AZAD11：can't find, so MV411D1：biphenotypic B-myelomonocytic leukemia</span></span>
<span id="cb4-267"><a href="#cb4-267" aria-hidden="true" tabindex="-1"></a><span class="do">### MV411AZAD5：can't find, so MV411D1：biphenotypic B-myelomonocytic leukemia</span></span>
<span id="cb4-268"><a href="#cb4-268" aria-hidden="true" tabindex="-1"></a><span class="do">### THP1：Childhood acute monocytic leukemia</span></span>
<span id="cb4-269"><a href="#cb4-269" aria-hidden="true" tabindex="-1"></a><span class="do">### OCIAML2：Adult acute myeloid leukemia</span></span>
<span id="cb4-270"><a href="#cb4-270" aria-hidden="true" tabindex="-1"></a><span class="do">### HL60：Adult acute myeloid leukemia</span></span>
<span id="cb4-271"><a href="#cb4-271" aria-hidden="true" tabindex="-1"></a><span class="do">### HEK293：Vaccine production cell line.</span></span>
<span id="cb4-272"><a href="#cb4-272" aria-hidden="true" tabindex="-1"></a><span class="do">### CML：Childhood acute megakaryoblastic leukemia</span></span>
<span id="cb4-273"><a href="#cb4-273" aria-hidden="true" tabindex="-1"></a><span class="do">### MOLM14CBFGLIS：can't find, so MOLM14：Adult acute myeloid leukemia</span></span>
<span id="cb4-274"><a href="#cb4-274" aria-hidden="true" tabindex="-1"></a>cell_data <span class="ot">&lt;-</span> metafusion[metafusion<span class="sc">$</span>Sample_ID <span class="sc">%in%</span> target_cell_sample]</span>
<span id="cb4-275"><a href="#cb4-275" aria-hidden="true" tabindex="-1"></a><span class="do">### &gt; nrow(cell_data)</span></span>
<span id="cb4-276"><a href="#cb4-276" aria-hidden="true" tabindex="-1"></a><span class="do">###  [1] 1853</span></span>
<span id="cb4-277"><a href="#cb4-277" aria-hidden="true" tabindex="-1"></a>cell_info <span class="ot">&lt;-</span> TARGET_info_pretreatment[TARGET_info_pretreatment<span class="sc">$</span>Patient_ID <span class="sc">%in%</span> target_cell_patient]</span>
<span id="cb4-278"><a href="#cb4-278" aria-hidden="true" tabindex="-1"></a><span class="do">###  nrow(cell_info)</span></span>
<span id="cb4-279"><a href="#cb4-279" aria-hidden="true" tabindex="-1"></a><span class="do">###  [1] 18</span></span>
<span id="cb4-280"><a href="#cb4-280" aria-hidden="true" tabindex="-1"></a><span class="do">### no os.etc</span></span>
<span id="cb4-281"><a href="#cb4-281" aria-hidden="true" tabindex="-1"></a>cell_info2 <span class="ot">&lt;-</span> TARGET_xena_pretreatment[TARGET_xena_pretreatment<span class="sc">$</span>Patient_ID <span class="sc">%in%</span> target_cell_patient]</span>
<span id="cb4-282"><a href="#cb4-282" aria-hidden="true" tabindex="-1"></a><span class="do">### nrow(cell_info2)</span></span>
<span id="cb4-283"><a href="#cb4-283" aria-hidden="true" tabindex="-1"></a><span class="do">### [1] 0</span></span>
<span id="cb4-284"><a href="#cb4-284" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-285"><a href="#cb4-285" aria-hidden="true" tabindex="-1"></a>TARGET_info_after_cell <span class="ot">&lt;-</span> TARGET_correct_after_pro <span class="sc">|&gt;</span></span>
<span id="cb4-286"><a href="#cb4-286" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">cohort =</span> <span class="fu">case_when</span>(</span>
<span id="cb4-287"><a href="#cb4-287" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">Sample_type</span><span class="st">`</span> <span class="sc">%in%</span> <span class="st">"Cell Lines"</span> <span class="sc">~</span> <span class="st">"TARGET-CELL"</span>,</span>
<span id="cb4-288"><a href="#cb4-288" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> cohort  </span>
<span id="cb4-289"><a href="#cb4-289" aria-hidden="true" tabindex="-1"></a>  ))</span>
<span id="cb4-290"><a href="#cb4-290" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-291"><a href="#cb4-291" aria-hidden="true" tabindex="-1"></a><span class="do">### The criteria for distinguishing normal samples based on Sample_type column in the TARGET databases are as follows:</span></span>
<span id="cb4-292"><a href="#cb4-292" aria-hidden="true" tabindex="-1"></a>TARGET_info_after_cell<span class="sc">$</span>Sample_type <span class="sc">|&gt;</span> <span class="fu">unique</span>()</span>
<span id="cb4-293"><a href="#cb4-293" aria-hidden="true" tabindex="-1"></a><span class="do">### [1] "Primary Blood Derived Cancer - Bone Marrow"              "Recurrent Blood Derived Cancer - Bone Marrow"           </span></span>
<span id="cb4-294"><a href="#cb4-294" aria-hidden="true" tabindex="-1"></a><span class="do">### [3] "Blood Derived Normal"                                    "Primary Blood Derived Cancer - Peripheral Blood"        </span></span>
<span id="cb4-295"><a href="#cb4-295" aria-hidden="true" tabindex="-1"></a><span class="do">### [5] "Bone Marrow Normal"                                      "Cell Lines"                                             </span></span>
<span id="cb4-296"><a href="#cb4-296" aria-hidden="true" tabindex="-1"></a><span class="do">### [7] "Next Generation Cancer Model"                            "Blood Derived Cancer - Bone Marrow, Post-treatment"     </span></span>
<span id="cb4-297"><a href="#cb4-297" aria-hidden="true" tabindex="-1"></a><span class="do">### [9] "Primary Tumor"                                           "Recurrent Tumor"                                        </span></span>
<span id="cb4-298"><a href="#cb4-298" aria-hidden="true" tabindex="-1"></a><span class="do">### [11] "Recurrent Blood Derived Cancer - Peripheral Blood"       "Blood Derived Cancer - Peripheral Blood, Post-treatment"</span></span>
<span id="cb4-299"><a href="#cb4-299" aria-hidden="true" tabindex="-1"></a><span class="do">### </span></span>
<span id="cb4-300"><a href="#cb4-300" aria-hidden="true" tabindex="-1"></a><span class="do">### If the sample type is "Blood Derived Normal" or "Bone Marrow Normal", then it is a normal sample. Otherwise, it is a tumor sample.</span></span>
<span id="cb4-301"><a href="#cb4-301" aria-hidden="true" tabindex="-1"></a>TARGET_info_after_normal <span class="ot">&lt;-</span> TARGET_info_after_cell <span class="sc">|&gt;</span></span>
<span id="cb4-302"><a href="#cb4-302" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">cohort =</span> <span class="fu">case_when</span>(</span>
<span id="cb4-303"><a href="#cb4-303" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">Sample_type</span><span class="st">`</span> <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Blood Derived Normal"</span>, <span class="st">"Bone Marrow Normal"</span>) <span class="sc">~</span> <span class="st">"TARGET-NORMAL"</span>,</span>
<span id="cb4-304"><a href="#cb4-304" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> cohort  </span>
<span id="cb4-305"><a href="#cb4-305" aria-hidden="true" tabindex="-1"></a>  ))</span>
<span id="cb4-306"><a href="#cb4-306" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-307"><a href="#cb4-307" aria-hidden="true" tabindex="-1"></a><span class="do">### Adjust the column order.</span></span>
<span id="cb4-308"><a href="#cb4-308" aria-hidden="true" tabindex="-1"></a>new_order <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Patient_ID"</span>, <span class="st">"Sample_ID"</span>, <span class="st">"Run"</span>, <span class="st">"Sample_type"</span>, <span class="st">"Age"</span>, <span class="st">"Sex"</span>, <span class="st">"race"</span>, <span class="st">"Cancer"</span>, </span>
<span id="cb4-309"><a href="#cb4-309" aria-hidden="true" tabindex="-1"></a>               <span class="st">"OS_Time"</span>, <span class="st">"OS_Status"</span>, <span class="st">"Project_id"</span>, <span class="st">"cohort"</span>)</span>
<span id="cb4-310"><a href="#cb4-310" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-311"><a href="#cb4-311" aria-hidden="true" tabindex="-1"></a>TARGET_info_after_processing <span class="ot">&lt;-</span> TARGET_info_after_normal <span class="sc">|&gt;</span></span>
<span id="cb4-312"><a href="#cb4-312" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">select</span>(<span class="fu">all_of</span>(new_order)) </span>
<span id="cb4-313"><a href="#cb4-313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-314"><a href="#cb4-314" aria-hidden="true" tabindex="-1"></a><span class="do">### make sure nrow is same between Before and After Processing. </span></span>
<span id="cb4-315"><a href="#cb4-315" aria-hidden="true" tabindex="-1"></a>TARGET_pretreatment <span class="sc">|&gt;</span> <span class="fu">nrow</span>()</span>
<span id="cb4-316"><a href="#cb4-316" aria-hidden="true" tabindex="-1"></a><span class="do">### [1]  3222</span></span>
<span id="cb4-317"><a href="#cb4-317" aria-hidden="true" tabindex="-1"></a>TARGET_info_after_processing <span class="sc">|&gt;</span> <span class="fu">nrow</span>() </span>
<span id="cb4-318"><a href="#cb4-318" aria-hidden="true" tabindex="-1"></a><span class="do">### [1] 3222</span></span>
<span id="cb4-319"><a href="#cb4-319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-320"><a href="#cb4-320" aria-hidden="true" tabindex="-1"></a><span class="do">### make sure IOBR/TME data just contain sample which target_sheet gave. </span></span>
<span id="cb4-321"><a href="#cb4-321" aria-hidden="true" tabindex="-1"></a>detected_id_TARGET <span class="ot">&lt;-</span> TARGET_info_after_processing<span class="sc">$</span>Run</span>
<span id="cb4-322"><a href="#cb4-322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-323"><a href="#cb4-323" aria-hidden="true" tabindex="-1"></a><span class="do">### save info</span></span>
<span id="cb4-324"><a href="#cb4-324" aria-hidden="true" tabindex="-1"></a>cohorts <span class="ot">&lt;-</span> <span class="fu">unique</span>(TARGET_info_after_processing<span class="sc">$</span>cohort)</span>
<span id="cb4-325"><a href="#cb4-325" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-326"><a href="#cb4-326" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (cohort1 <span class="cf">in</span> cohorts) {</span>
<span id="cb4-327"><a href="#cb4-327" aria-hidden="true" tabindex="-1"></a>  cohort_data <span class="ot">&lt;-</span> <span class="fu">subset</span>(TARGET_info_after_processing, cohort <span class="sc">==</span> cohort1)</span>
<span id="cb4-328"><a href="#cb4-328" aria-hidden="true" tabindex="-1"></a>  name <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"-"</span>, <span class="st">"_"</span>, cohort1)</span>
<span id="cb4-329"><a href="#cb4-329" aria-hidden="true" tabindex="-1"></a>  file_name <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"/home/data2/Projects/Fusion/fusiondb_v3/Clininfo/"</span>, name ,<span class="st">"_info.tsv"</span>)</span>
<span id="cb4-330"><a href="#cb4-330" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fwrite</span>(cohort_data , <span class="at">file =</span> file_name, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)  </span>
<span id="cb4-331"><a href="#cb4-331" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-332"><a href="#cb4-332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-333"><a href="#cb4-333" aria-hidden="true" tabindex="-1"></a><span class="do">##Gene</span></span>
<span id="cb4-334"><a href="#cb4-334" aria-hidden="true" tabindex="-1"></a>metafusion1 <span class="ot">&lt;-</span> metafusion <span class="sc">|&gt;</span> </span>
<span id="cb4-335"><a href="#cb4-335" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb4-336"><a href="#cb4-336" aria-hidden="true" tabindex="-1"></a>    <span class="at">cohort =</span> <span class="fu">case_when</span>(</span>
<span id="cb4-337"><a href="#cb4-337" aria-hidden="true" tabindex="-1"></a>      Sample_ID <span class="sc">%in%</span> TARGET_info_after_processing<span class="sc">$</span>Run <span class="sc">&amp;</span> <span class="fu">grepl</span>(<span class="st">"TARGET"</span>, cohort) <span class="sc">~</span> TARGET_info_after_processing<span class="sc">$</span>cohort[<span class="fu">match</span>(Sample_ID, TARGET_info_after_processing<span class="sc">$</span>Run)],</span>
<span id="cb4-338"><a href="#cb4-338" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> cohort</span>
<span id="cb4-339"><a href="#cb4-339" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb4-340"><a href="#cb4-340" aria-hidden="true" tabindex="-1"></a>  )  </span>
<span id="cb4-341"><a href="#cb4-341" aria-hidden="true" tabindex="-1"></a>  cp_file_name1 <span class="ot">&lt;-</span> <span class="st">"/home/data2/Projects/Fusion/fusiondb_v3/metafusion_calls_matched.cptac.target.tsv"</span></span>
<span id="cb4-342"><a href="#cb4-342" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fwrite</span>(metafusion1, <span class="at">file =</span> cp_file_name1, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>) </span>
<span id="cb4-343"><a href="#cb4-343" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-344"><a href="#cb4-344" aria-hidden="true" tabindex="-1"></a><span class="do">##TME</span></span>
<span id="cb4-345"><a href="#cb4-345" aria-hidden="true" tabindex="-1"></a>NBL_TME <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/home/data2/Projects/Fusion/fusiondb/IOBR/TARGET-NBL_TME.tsv.gz"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>) <span class="sc">|&gt;</span> <span class="fu">mutate</span>(<span class="at">Project_id =</span> <span class="st">"TARGET-NBL"</span>)</span>
<span id="cb4-346"><a href="#cb4-346" aria-hidden="true" tabindex="-1"></a>AML_TME <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/home/data2/Projects/Fusion/fusiondb/IOBR/TARGET-AML_TME.tsv.gz"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>) <span class="sc">|&gt;</span> <span class="fu">mutate</span>(<span class="at">Project_id =</span> <span class="st">"TARGET-AML"</span>)</span>
<span id="cb4-347"><a href="#cb4-347" aria-hidden="true" tabindex="-1"></a>TARGET_TME <span class="ot">&lt;-</span> <span class="fu">rbind</span>(NBL_TME,AML_TME) <span class="sc">|&gt;</span></span>
<span id="cb4-348"><a href="#cb4-348" aria-hidden="true" tabindex="-1"></a>              <span class="fu">mutate</span>(<span class="at">dash_count =</span> <span class="fu">str_count</span>(ID, <span class="st">"-"</span>))</span>
<span id="cb4-349"><a href="#cb4-349" aria-hidden="true" tabindex="-1"></a><span class="do">### Manually check and delete composite samples from the info dataset.(TARGET-30-PAPTFZ-01A-01R-TARGET-30-PAPEAV-01A-01R)</span></span>
<span id="cb4-350"><a href="#cb4-350" aria-hidden="true" tabindex="-1"></a>TARGET_TME <span class="sc">|&gt;</span> </span>
<span id="cb4-351"><a href="#cb4-351" aria-hidden="true" tabindex="-1"></a><span class="fu">filter</span>(dash_count <span class="sc">==</span> <span class="dv">9</span>) <span class="sc">|&gt;</span> </span>
<span id="cb4-352"><a href="#cb4-352" aria-hidden="true" tabindex="-1"></a><span class="fu">select</span>(ID) <span class="sc">|&gt;</span></span>
<span id="cb4-353"><a href="#cb4-353" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>()</span>
<span id="cb4-354"><a href="#cb4-354" aria-hidden="true" tabindex="-1"></a><span class="do">### 1: TARGET-30-PASYPX-01A-01R-TARGET-30-PAIXIF-01A-01R</span></span>
<span id="cb4-355"><a href="#cb4-355" aria-hidden="true" tabindex="-1"></a><span class="do">### 2: TARGET-30-PANKFE-01A-01R-TARGET-30-PAPUAR-01A-01R</span></span>
<span id="cb4-356"><a href="#cb4-356" aria-hidden="true" tabindex="-1"></a><span class="do">### 3: TARGET-30-PANUKV-01A-01R-TARGET-30-PASUML-01A-01R</span></span>
<span id="cb4-357"><a href="#cb4-357" aria-hidden="true" tabindex="-1"></a><span class="do">### 4: TARGET-30-PAPTFZ-01A-01R-TARGET-30-PAPEAV-01A-01R</span></span>
<span id="cb4-358"><a href="#cb4-358" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-359"><a href="#cb4-359" aria-hidden="true" tabindex="-1"></a>TARGET_TME_pretreatment <span class="ot">&lt;-</span> TARGET_TME <span class="sc">|&gt;</span></span>
<span id="cb4-360"><a href="#cb4-360" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">filter</span>(dash_count <span class="sc">!=</span> <span class="dv">9</span>) <span class="sc">|&gt;</span></span>
<span id="cb4-361"><a href="#cb4-361" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">select</span>(<span class="sc">-</span>dash_count)</span>
<span id="cb4-362"><a href="#cb4-362" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-363"><a href="#cb4-363" aria-hidden="true" tabindex="-1"></a><span class="do">### Filter TARGET Replicate Samples</span></span>
<span id="cb4-364"><a href="#cb4-364" aria-hidden="true" tabindex="-1"></a><span class="do">### The filter rule following broad institute says:</span></span>
<span id="cb4-365"><a href="#cb4-365" aria-hidden="true" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb4-366"><a href="#cb4-366" aria-hidden="true" tabindex="-1"></a><span class="do">### In many instances there is more than one aliquot for a given combination of individual, platform, and data type. However, only one aliquot may be ingested into Firehose. Therefore, a set of precedence rules are applied to select the most scientifically advantageous one among them. Two filters are applied to achieve this aim: an Analyte Replicate Filter and a Sort Replicate Filter.</span></span>
<span id="cb4-367"><a href="#cb4-367" aria-hidden="true" tabindex="-1"></a><span class="do">### </span></span>
<span id="cb4-368"><a href="#cb4-368" aria-hidden="true" tabindex="-1"></a><span class="do">### Analyte Replicate Filter</span></span>
<span id="cb4-369"><a href="#cb4-369" aria-hidden="true" tabindex="-1"></a><span class="do">### The following precedence rules are applied when the aliquots have differing analytes. For RNA aliquots, T analytes are dropped in preference to H and R analytes, since T is the inferior extraction protocol. If H and R are encountered, H is the chosen analyte. This is somewhat arbitrary and subject to change, since it is not clear at present whether H or R is the better protocol. If there are multiple aliquots associated with the chosen RNA analyte, the aliquot with the later plate number is chosen. For DNA aliquots, D analytes (native DNA) are preferred over G, W, or X (whole-genome amplified) analytes, unless the G, W, or X analyte sample has a higher plate number.</span></span>
<span id="cb4-370"><a href="#cb4-370" aria-hidden="true" tabindex="-1"></a><span class="do">### </span></span>
<span id="cb4-371"><a href="#cb4-371" aria-hidden="true" tabindex="-1"></a><span class="do">### Sort Replicate Filter</span></span>
<span id="cb4-372"><a href="#cb4-372" aria-hidden="true" tabindex="-1"></a><span class="do">### The following precedence rules are applied when the analyte filter still produces more than one sample. The sort filter chooses the aliquot with the highest lexicographical sort value, to ensure that the barcode with the highest portion and/or plate number is selected when all other barcode fields are identical.</span></span>
<span id="cb4-373"><a href="#cb4-373" aria-hidden="true" tabindex="-1"></a><span class="do">### </span></span>
<span id="cb4-374"><a href="#cb4-374" aria-hidden="true" tabindex="-1"></a><span class="do">### Ref Link: &lt;https://confluence.broadinstitute.org/display/GDAC/FAQ#FAQ-sampleTypesQWhatTCGAsampletypesareFirehosepipelinesexecutedupon&gt;</span></span>
<span id="cb4-375"><a href="#cb4-375" aria-hidden="true" tabindex="-1"></a><span class="do">###           &lt;https://gist.github.com/ShixiangWang/33b2f9b49b77eaa8f773b428480f9101&gt;</span></span>
<span id="cb4-376"><a href="#cb4-376" aria-hidden="true" tabindex="-1"></a>TARGET_TME_after_replicate <span class="ot">&lt;-</span> TARGET_TME_pretreatment <span class="sc">|&gt;</span></span>
<span id="cb4-377"><a href="#cb4-377" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">mutate</span>(</span>
<span id="cb4-378"><a href="#cb4-378" aria-hidden="true" tabindex="-1"></a>                              <span class="at">prefixes =</span> <span class="fu">str_remove</span>(ID, <span class="st">"-[^-]+$"</span>),  </span>
<span id="cb4-379"><a href="#cb4-379" aria-hidden="true" tabindex="-1"></a>                              <span class="at">suffixes =</span> <span class="fu">str_extract</span>(ID, <span class="st">"[^-]+$"</span>), </span>
<span id="cb4-380"><a href="#cb4-380" aria-hidden="true" tabindex="-1"></a>                              <span class="at">second_chars =</span> <span class="fu">str_extract</span>(suffixes, <span class="st">"[0-9]+"</span>),</span>
<span id="cb4-381"><a href="#cb4-381" aria-hidden="true" tabindex="-1"></a>                              <span class="at">second_chars =</span> <span class="fu">as.integer</span>(second_chars)) <span class="sc">|&gt;</span></span>
<span id="cb4-382"><a href="#cb4-382" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">group_by</span>(prefixes) <span class="sc">|&gt;</span></span>
<span id="cb4-383"><a href="#cb4-383" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">slice</span>(<span class="fu">which.max</span>(second_chars)) <span class="sc">|&gt;</span></span>
<span id="cb4-384"><a href="#cb4-384" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">ungroup</span>() <span class="sc">|&gt;</span></span>
<span id="cb4-385"><a href="#cb4-385" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">mutate</span>(<span class="at">ID =</span> prefixes) <span class="sc">|&gt;</span> </span>
<span id="cb4-386"><a href="#cb4-386" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">filter</span>(ID <span class="sc">%in%</span> detected_id_TARGET)  <span class="sc">|&gt;</span> <span class="do">### Ensure that all IDs in the TME data can be found in the TARGET info data. </span></span>
<span id="cb4-387"><a href="#cb4-387" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(prefixes,suffixes,second_chars))</span>
<span id="cb4-388"><a href="#cb4-388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-389"><a href="#cb4-389" aria-hidden="true" tabindex="-1"></a><span class="do">### Using TARGET info data cohort column to annotate TARGET TME data </span></span>
<span id="cb4-390"><a href="#cb4-390" aria-hidden="true" tabindex="-1"></a>cohort_data   <span class="ot">&lt;-</span> TARGET_info_after_processing <span class="sc">|&gt;</span> </span>
<span id="cb4-391"><a href="#cb4-391" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">select</span>(Run, cohort)   <span class="sc">|&gt;</span></span>
<span id="cb4-392"><a href="#cb4-392" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">unique</span>()   </span>
<span id="cb4-393"><a href="#cb4-393" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-394"><a href="#cb4-394" aria-hidden="true" tabindex="-1"></a>TARGET_TME_after_cohort <span class="ot">&lt;-</span> TARGET_TME_after_replicate <span class="sc">|&gt;</span></span>
<span id="cb4-395"><a href="#cb4-395" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">mutate</span>(<span class="at">Run =</span> ID)  <span class="sc">|&gt;</span></span>
<span id="cb4-396"><a href="#cb4-396" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">left_join</span>(cohort_data, <span class="at">by =</span> <span class="st">"Run"</span>)  <span class="sc">|&gt;</span></span>
<span id="cb4-397"><a href="#cb4-397" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">select</span>(<span class="sc">-</span>Run)</span>
<span id="cb4-398"><a href="#cb4-398" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-399"><a href="#cb4-399" aria-hidden="true" tabindex="-1"></a><span class="do">### save TME</span></span>
<span id="cb4-400"><a href="#cb4-400" aria-hidden="true" tabindex="-1"></a>cohorts_TME <span class="ot">&lt;-</span> <span class="fu">unique</span>(TARGET_TME_after_cohort<span class="sc">$</span>cohort)</span>
<span id="cb4-401"><a href="#cb4-401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-402"><a href="#cb4-402" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (cohort1 <span class="cf">in</span> cohorts_TME) {</span>
<span id="cb4-403"><a href="#cb4-403" aria-hidden="true" tabindex="-1"></a>  cohort_data <span class="ot">&lt;-</span> <span class="fu">subset</span>(TARGET_TME_after_cohort, cohort <span class="sc">==</span> cohort1) <span class="sc">|&gt;</span> <span class="fu">select</span>(<span class="sc">-</span>cohort)</span>
<span id="cb4-404"><a href="#cb4-404" aria-hidden="true" tabindex="-1"></a>  file_name <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"/home/data2/Projects/Fusion/fusiondb_v3/IOBR/"</span>, cohort1 ,<span class="st">"_TME.tsv.gz"</span>)</span>
<span id="cb4-405"><a href="#cb4-405" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(cohort_data, </span>
<span id="cb4-406"><a href="#cb4-406" aria-hidden="true" tabindex="-1"></a>              <span class="at">file =</span> <span class="fu">gzfile</span>(file_name), </span>
<span id="cb4-407"><a href="#cb4-407" aria-hidden="true" tabindex="-1"></a>              <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, </span>
<span id="cb4-408"><a href="#cb4-408" aria-hidden="true" tabindex="-1"></a>              <span class="at">row.names =</span> <span class="cn">FALSE</span>, </span>
<span id="cb4-409"><a href="#cb4-409" aria-hidden="true" tabindex="-1"></a>              <span class="at">quote =</span> <span class="cn">FALSE</span>) </span>
<span id="cb4-410"><a href="#cb4-410" aria-hidden="true" tabindex="-1"></a>}  </span>
<span id="cb4-411"><a href="#cb4-411" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-412"><a href="#cb4-412" aria-hidden="true" tabindex="-1"></a><span class="do">## IOBR</span></span>
<span id="cb4-413"><a href="#cb4-413" aria-hidden="true" tabindex="-1"></a>NBL_IOBR_signature <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/home/data2/Projects/Fusion/fusiondb/IOBR/TARGET-NBL_IOBR_signature.tsv.gz"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>) <span class="sc">|&gt;</span> <span class="fu">mutate</span>(<span class="at">Project_id =</span> <span class="st">"TARGET-NBL"</span>)</span>
<span id="cb4-414"><a href="#cb4-414" aria-hidden="true" tabindex="-1"></a>AML_IOBR_signature <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/home/data2/Projects/Fusion/fusiondb/IOBR/TARGET-AML_IOBR_signature.tsv.gz"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>) <span class="sc">|&gt;</span> <span class="fu">mutate</span>(<span class="at">Project_id =</span> <span class="st">"TARGET-AML"</span>)</span>
<span id="cb4-415"><a href="#cb4-415" aria-hidden="true" tabindex="-1"></a>TARGET_IOBR <span class="ot">&lt;-</span> <span class="fu">rbind</span>(NBL_IOBR_signature,AML_IOBR_signature) <span class="sc">|&gt;</span></span>
<span id="cb4-416"><a href="#cb4-416" aria-hidden="true" tabindex="-1"></a>               <span class="fu">mutate</span>(<span class="at">dash_count =</span> <span class="fu">str_count</span>(ID, <span class="st">"-"</span>))</span>
<span id="cb4-417"><a href="#cb4-417" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-418"><a href="#cb4-418" aria-hidden="true" tabindex="-1"></a><span class="do">### Manually check and delete composite samples from the info dataset.(TARGET-30-PAPTFZ-01A-01R-TARGET-30-PAPEAV-01A-01R)</span></span>
<span id="cb4-419"><a href="#cb4-419" aria-hidden="true" tabindex="-1"></a>TARGET_IOBR <span class="sc">|&gt;</span> </span>
<span id="cb4-420"><a href="#cb4-420" aria-hidden="true" tabindex="-1"></a><span class="fu">filter</span>(dash_count <span class="sc">==</span> <span class="dv">9</span>) <span class="sc">|&gt;</span> </span>
<span id="cb4-421"><a href="#cb4-421" aria-hidden="true" tabindex="-1"></a><span class="fu">select</span>(ID) <span class="sc">|&gt;</span></span>
<span id="cb4-422"><a href="#cb4-422" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>()</span>
<span id="cb4-423"><a href="#cb4-423" aria-hidden="true" tabindex="-1"></a><span class="do">### 1: TARGET-30-PASYPX-01A-01R-TARGET-30-PAIXIF-01A-01R</span></span>
<span id="cb4-424"><a href="#cb4-424" aria-hidden="true" tabindex="-1"></a><span class="do">### 2: TARGET-30-PANKFE-01A-01R-TARGET-30-PAPUAR-01A-01R</span></span>
<span id="cb4-425"><a href="#cb4-425" aria-hidden="true" tabindex="-1"></a><span class="do">### 3: TARGET-30-PANUKV-01A-01R-TARGET-30-PASUML-01A-01R</span></span>
<span id="cb4-426"><a href="#cb4-426" aria-hidden="true" tabindex="-1"></a><span class="do">### 4: TARGET-30-PAPTFZ-01A-01R-TARGET-30-PAPEAV-01A-01R</span></span>
<span id="cb4-427"><a href="#cb4-427" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-428"><a href="#cb4-428" aria-hidden="true" tabindex="-1"></a>TARGET_IOBR_pretreatment <span class="ot">&lt;-</span> TARGET_IOBR <span class="sc">|&gt;</span></span>
<span id="cb4-429"><a href="#cb4-429" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">filter</span>(dash_count <span class="sc">!=</span> <span class="dv">9</span>) <span class="sc">|&gt;</span></span>
<span id="cb4-430"><a href="#cb4-430" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">select</span>(<span class="sc">-</span>dash_count)</span>
<span id="cb4-431"><a href="#cb4-431" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-432"><a href="#cb4-432" aria-hidden="true" tabindex="-1"></a>TARGET_IOBR_after_replicate <span class="ot">&lt;-</span> TARGET_IOBR_pretreatment <span class="sc">|&gt;</span></span>
<span id="cb4-433"><a href="#cb4-433" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">mutate</span>(</span>
<span id="cb4-434"><a href="#cb4-434" aria-hidden="true" tabindex="-1"></a>                            <span class="at">prefixes =</span> <span class="fu">str_remove</span>(ID, <span class="st">"-[^-]+$"</span>),  </span>
<span id="cb4-435"><a href="#cb4-435" aria-hidden="true" tabindex="-1"></a>                            <span class="at">suffixes =</span> <span class="fu">str_extract</span>(ID, <span class="st">"[^-]+$"</span>), </span>
<span id="cb4-436"><a href="#cb4-436" aria-hidden="true" tabindex="-1"></a>                            <span class="at">second_chars =</span> <span class="fu">str_extract</span>(suffixes, <span class="st">"[0-9]+"</span>),</span>
<span id="cb4-437"><a href="#cb4-437" aria-hidden="true" tabindex="-1"></a>                            <span class="at">second_chars =</span> <span class="fu">as.integer</span>(second_chars)) <span class="sc">|&gt;</span></span>
<span id="cb4-438"><a href="#cb4-438" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">group_by</span>(prefixes) <span class="sc">|&gt;</span></span>
<span id="cb4-439"><a href="#cb4-439" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">slice</span>(<span class="fu">which.max</span>(second_chars)) <span class="sc">|&gt;</span></span>
<span id="cb4-440"><a href="#cb4-440" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">ungroup</span>() <span class="sc">|&gt;</span></span>
<span id="cb4-441"><a href="#cb4-441" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">mutate</span>(<span class="at">ID =</span> prefixes) <span class="sc">|&gt;</span> </span>
<span id="cb4-442"><a href="#cb4-442" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">filter</span>(ID <span class="sc">%in%</span> detected_id_TARGET)  <span class="sc">|&gt;</span> <span class="do">### Ensure that all IDs in the TME data can be found in the TARGET info data. </span></span>
<span id="cb4-443"><a href="#cb4-443" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(prefixes,suffixes,second_chars))</span>
<span id="cb4-444"><a href="#cb4-444" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-445"><a href="#cb4-445" aria-hidden="true" tabindex="-1"></a><span class="do">### Using TARGET info data cohort column to annotate TARGET TME data </span></span>
<span id="cb4-446"><a href="#cb4-446" aria-hidden="true" tabindex="-1"></a>cohort_data   <span class="ot">&lt;-</span> TARGET_info_after_processing <span class="sc">|&gt;</span> </span>
<span id="cb4-447"><a href="#cb4-447" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">select</span>(Run, cohort)   <span class="sc">|&gt;</span></span>
<span id="cb4-448"><a href="#cb4-448" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">unique</span>()   </span>
<span id="cb4-449"><a href="#cb4-449" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-450"><a href="#cb4-450" aria-hidden="true" tabindex="-1"></a>TARGET_IOBR_after_cohort <span class="ot">&lt;-</span> TARGET_IOBR_after_replicate <span class="sc">|&gt;</span></span>
<span id="cb4-451"><a href="#cb4-451" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">mutate</span>(<span class="at">Run =</span> ID)  <span class="sc">|&gt;</span></span>
<span id="cb4-452"><a href="#cb4-452" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">left_join</span>(cohort_data, <span class="at">by =</span> <span class="st">"Run"</span>)  <span class="sc">|&gt;</span></span>
<span id="cb4-453"><a href="#cb4-453" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">select</span>(<span class="sc">-</span>Run)</span>
<span id="cb4-454"><a href="#cb4-454" aria-hidden="true" tabindex="-1"></a><span class="do">### save IOBR</span></span>
<span id="cb4-455"><a href="#cb4-455" aria-hidden="true" tabindex="-1"></a>cohorts_IOBR <span class="ot">&lt;-</span> <span class="fu">unique</span>(TARGET_IOBR_after_cohort<span class="sc">$</span>cohort)</span>
<span id="cb4-456"><a href="#cb4-456" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-457"><a href="#cb4-457" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (cohort1 <span class="cf">in</span> cohorts_IOBR) {</span>
<span id="cb4-458"><a href="#cb4-458" aria-hidden="true" tabindex="-1"></a>  cohort_data <span class="ot">&lt;-</span> <span class="fu">subset</span>(TARGET_IOBR_after_cohort, cohort <span class="sc">==</span> cohort1) <span class="sc">|&gt;</span> <span class="fu">select</span>(<span class="sc">-</span>cohort)</span>
<span id="cb4-459"><a href="#cb4-459" aria-hidden="true" tabindex="-1"></a>  file_name <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"/home/data2/Projects/Fusion/fusiondb_v3/IOBR/"</span>, cohort1 ,<span class="st">"_IOBR_signature.tsv.gz"</span>)</span>
<span id="cb4-460"><a href="#cb4-460" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(cohort_data, </span>
<span id="cb4-461"><a href="#cb4-461" aria-hidden="true" tabindex="-1"></a>              <span class="at">file =</span> <span class="fu">gzfile</span>(file_name), </span>
<span id="cb4-462"><a href="#cb4-462" aria-hidden="true" tabindex="-1"></a>              <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, </span>
<span id="cb4-463"><a href="#cb4-463" aria-hidden="true" tabindex="-1"></a>              <span class="at">row.names =</span> <span class="cn">FALSE</span>, </span>
<span id="cb4-464"><a href="#cb4-464" aria-hidden="true" tabindex="-1"></a>              <span class="at">quote =</span> <span class="cn">FALSE</span>) </span>
<span id="cb4-465"><a href="#cb4-465" aria-hidden="true" tabindex="-1"></a>}  </span>
<span id="cb4-466"><a href="#cb4-466" aria-hidden="true" tabindex="-1"></a></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="cptac-1" class="level3">
<h3 class="anchored" data-anchor-id="cptac-1">CPTAC</h3>
<div class="sourceCode" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>file_meta <span class="ot">&lt;-</span> <span class="st">"/home/data2/Projects/Fusion/fusiondb_v3/metafusion_calls_matched.tsv"</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>metafusion <span class="ot">&lt;-</span> <span class="fu">fread</span>(file_meta, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="co"># CPTAC</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="do">## info</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>CPTAC_sheet <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/home/data2/Projects/Fusion/unclean/CPTAC_gdc_sample_sheet.2023-11-15.tsv"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a><span class="do">### Composite Sample </span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a><span class="do">### Composite Sample use comma to separate like C3L-03400-05, C3L-03400-02</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a><span class="do">### </span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a><span class="do">### It leads one question is whether these composite samples using comma to separate from identical patient?</span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>rows_with_comma <span class="ot">&lt;-</span> <span class="fu">grepl</span>(<span class="st">", "</span>, CPTAC_sheet<span class="sc">$</span><span class="st">`</span><span class="at">Sample ID</span><span class="st">`</span>)</span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>rows_with_comma_data <span class="ot">&lt;-</span> CPTAC_sheet[rows_with_comma, ]</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>check_same_chars <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>  parts <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(x, <span class="st">", "</span>))</span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(parts[<span class="dv">1</span>] <span class="sc">==</span> parts[<span class="dv">2</span>])</span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>same_chars <span class="ot">&lt;-</span> <span class="fu">sapply</span>(rows_with_comma_data<span class="sc">$</span><span class="st">`</span><span class="at">Case ID</span><span class="st">`</span>, check_same_chars)</span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">all</span>(same_chars)) {</span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"TRUE</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"FALSE</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a><span class="do">### They are from identical patient.</span></span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a><span class="do">### For the TARGET databases, check whether there is any overlap between composite and normal Sample_ID in the info data. If there is an overlap, retain only the normal Sample_ID. </span></span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a>check_cptac_overlap <span class="ot">&lt;-</span> CPTAC_sheet <span class="sc">|&gt;</span>  </span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">select</span>(<span class="fu">c</span>(<span class="st">`</span><span class="at">Case ID</span><span class="st">`</span>, <span class="st">`</span><span class="at">Sample ID</span><span class="st">`</span>, <span class="st">`</span><span class="at">Sample Type</span><span class="st">`</span>, <span class="st">`</span><span class="at">Project ID</span><span class="st">`</span>)) <span class="sc">|&gt;</span></span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">unique</span>() <span class="sc">|&gt;</span> </span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">separate_rows</span>(<span class="st">`</span><span class="at">Sample ID</span><span class="st">`</span>, <span class="at">sep =</span> <span class="st">",</span><span class="sc">\\</span><span class="st">s*"</span>) </span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb5-39"><a href="#cb5-39" aria-hidden="true" tabindex="-1"></a>check_cptac_overlap <span class="ot">&lt;-</span> check_cptac_overlap <span class="sc">|&gt;</span></span>
<span id="cb5-40"><a href="#cb5-40" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">mutate</span>(<span class="at">Sample_Type =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">", "</span>, <span class="st">`</span><span class="at">Sample ID</span><span class="st">`</span>), <span class="st">"Composite"</span>, <span class="st">"Normal"</span>))</span>
<span id="cb5-41"><a href="#cb5-41" aria-hidden="true" tabindex="-1"></a>                      </span>
<span id="cb5-42"><a href="#cb5-42" aria-hidden="true" tabindex="-1"></a>overlap_samples <span class="ot">&lt;-</span> check_cptac_overlap <span class="sc">|&gt;</span></span>
<span id="cb5-43"><a href="#cb5-43" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">group_by</span>(<span class="st">`</span><span class="at">Sample ID</span><span class="st">`</span>) <span class="sc">|&gt;</span></span>
<span id="cb5-44"><a href="#cb5-44" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">summarise</span>(<span class="at">Count =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">'drop'</span>) <span class="sc">|&gt;</span></span>
<span id="cb5-45"><a href="#cb5-45" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">filter</span>(Count <span class="sc">&gt;</span> <span class="dv">1</span>)</span>
<span id="cb5-46"><a href="#cb5-46" aria-hidden="true" tabindex="-1"></a><span class="do">### &gt; overlap_samples </span></span>
<span id="cb5-47"><a href="#cb5-47" aria-hidden="true" tabindex="-1"></a><span class="do">### A tibble: 0 × 2</span></span>
<span id="cb5-48"><a href="#cb5-48" aria-hidden="true" tabindex="-1"></a><span class="do">### ℹ 2 variables: Sample ID &lt;chr&gt;, Count &lt;int&gt; </span></span>
<span id="cb5-49"><a href="#cb5-49" aria-hidden="true" tabindex="-1"></a><span class="do">### no this situation</span></span>
<span id="cb5-50"><a href="#cb5-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-51"><a href="#cb5-51" aria-hidden="true" tabindex="-1"></a><span class="do">### Due to the less rigorous patient naming conventions(like 604) in the CPTAC2 project, check for any overlapping patient names between CPTAC2 and CPTAC3.</span></span>
<span id="cb5-52"><a href="#cb5-52" aria-hidden="true" tabindex="-1"></a>CPTAC_sheet_rigorous  <span class="ot">&lt;-</span> CPTAC_sheet <span class="sc">|&gt;</span> </span>
<span id="cb5-53"><a href="#cb5-53" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">select</span>(<span class="fu">c</span>(<span class="st">`</span><span class="at">Case ID</span><span class="st">`</span>, <span class="st">`</span><span class="at">Sample ID</span><span class="st">`</span>, <span class="st">`</span><span class="at">Sample Type</span><span class="st">`</span>, <span class="st">`</span><span class="at">Project ID</span><span class="st">`</span>)) <span class="sc">|&gt;</span>  </span>
<span id="cb5-54"><a href="#cb5-54" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">unique</span>() </span>
<span id="cb5-55"><a href="#cb5-55" aria-hidden="true" tabindex="-1"></a>duplicates <span class="ot">&lt;-</span> CPTAC_sheet_rigorous <span class="sc">|&gt;</span></span>
<span id="cb5-56"><a href="#cb5-56" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">group_by</span>(<span class="st">`</span><span class="at">Case ID</span><span class="st">`</span>, <span class="st">`</span><span class="at">Sample ID</span><span class="st">`</span>) <span class="sc">|&gt;</span></span>
<span id="cb5-57"><a href="#cb5-57" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">summarise</span>(<span class="at">n_projects =</span> <span class="fu">n</span>(), <span class="at">projects =</span> <span class="fu">paste</span>(<span class="st">`</span><span class="at">Project ID</span><span class="st">`</span>, <span class="at">collapse =</span> <span class="st">", "</span>)) <span class="sc">|&gt;</span></span>
<span id="cb5-58"><a href="#cb5-58" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">filter</span>(n_projects <span class="sc">&gt;</span> <span class="dv">1</span>)</span>
<span id="cb5-59"><a href="#cb5-59" aria-hidden="true" tabindex="-1"></a><span class="do">### nrow(duplicates)</span></span>
<span id="cb5-60"><a href="#cb5-60" aria-hidden="true" tabindex="-1"></a><span class="do">### [1] 0</span></span>
<span id="cb5-61"><a href="#cb5-61" aria-hidden="true" tabindex="-1"></a><span class="do">### no same patient names between CPTAC2 and CPTAC3</span></span>
<span id="cb5-62"><a href="#cb5-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-63"><a href="#cb5-63" aria-hidden="true" tabindex="-1"></a><span class="do">### CPTAC_sheet pretreatment</span></span>
<span id="cb5-64"><a href="#cb5-64" aria-hidden="true" tabindex="-1"></a>matched_CPTAC_pretreatment <span class="ot">&lt;-</span> CPTAC_sheet <span class="sc">|&gt;</span> </span>
<span id="cb5-65"><a href="#cb5-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(<span class="st">`</span><span class="at">Case ID</span><span class="st">`</span>, <span class="st">`</span><span class="at">Sample ID</span><span class="st">`</span>, <span class="st">`</span><span class="at">Sample Type</span><span class="st">`</span>, <span class="st">`</span><span class="at">Project ID</span><span class="st">`</span>)) <span class="sc">|&gt;</span></span>
<span id="cb5-66"><a href="#cb5-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">"Patient_ID"</span> <span class="ot">=</span> <span class="st">`</span><span class="at">Case ID</span><span class="st">`</span>,</span>
<span id="cb5-67"><a href="#cb5-67" aria-hidden="true" tabindex="-1"></a>         <span class="st">"Sample_ID"</span> <span class="ot">=</span> <span class="st">`</span><span class="at">Sample ID</span><span class="st">`</span>,</span>
<span id="cb5-68"><a href="#cb5-68" aria-hidden="true" tabindex="-1"></a>         <span class="st">"Sample_type"</span>  <span class="ot">=</span>  <span class="st">`</span><span class="at">Sample Type</span><span class="st">`</span>,</span>
<span id="cb5-69"><a href="#cb5-69" aria-hidden="true" tabindex="-1"></a>         <span class="st">"Project_id"</span> <span class="ot">=</span> <span class="st">`</span><span class="at">Project ID</span><span class="st">`</span>) <span class="sc">|&gt;</span> </span>
<span id="cb5-70"><a href="#cb5-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate_rows</span>(Patient_ID, <span class="at">sep =</span> <span class="st">",</span><span class="sc">\\</span><span class="st">s*"</span>) <span class="sc">|&gt;</span>  <span class="co">#because they are from identical patient.</span></span>
<span id="cb5-71"><a href="#cb5-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>() </span>
<span id="cb5-72"><a href="#cb5-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-73"><a href="#cb5-73" aria-hidden="true" tabindex="-1"></a><span class="do">### CPTAC_info pretreatment</span></span>
<span id="cb5-74"><a href="#cb5-74" aria-hidden="true" tabindex="-1"></a>CPTAC_info <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/home/data2/Projects/Fusion/fusiondb/Clininfo/CPTAC_info.tsv"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb5-75"><a href="#cb5-75" aria-hidden="true" tabindex="-1"></a><span class="co"># &gt; CPTAC_info$Cancer |&gt; unique()</span></span>
<span id="cb5-76"><a href="#cb5-76" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] "BRCA"       "CCRCC"      "COAD"       "GBM"        "HGSC"       "HNSCC"      "LSCC"       "LUAD"       "PDAC"       "UCEC"       "GTEX-Brain"</span></span>
<span id="cb5-77"><a href="#cb5-77" aria-hidden="true" tabindex="-1"></a><span class="do">### We use our naming standard mutate a new column cohort.</span></span>
<span id="cb5-78"><a href="#cb5-78" aria-hidden="true" tabindex="-1"></a>cptac_replacement_table2 <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb5-79"><a href="#cb5-79" aria-hidden="true" tabindex="-1"></a>  <span class="st">"BRCA"</span> <span class="ot">=</span> <span class="st">"CPTAC-BRCA"</span>,</span>
<span id="cb5-80"><a href="#cb5-80" aria-hidden="true" tabindex="-1"></a>  <span class="st">"CCRCC"</span> <span class="ot">=</span> <span class="st">"CPTAC-KIRC"</span>,</span>
<span id="cb5-81"><a href="#cb5-81" aria-hidden="true" tabindex="-1"></a>  <span class="st">"COAD"</span> <span class="ot">=</span>  <span class="st">"CPTAC-COAD"</span>,</span>
<span id="cb5-82"><a href="#cb5-82" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GBM"</span> <span class="ot">=</span> <span class="st">"CPTAC-GBM"</span>,</span>
<span id="cb5-83"><a href="#cb5-83" aria-hidden="true" tabindex="-1"></a>  <span class="st">"HGSC"</span> <span class="ot">=</span> <span class="st">"CPTAC-HGSC"</span>,</span>
<span id="cb5-84"><a href="#cb5-84" aria-hidden="true" tabindex="-1"></a>  <span class="st">"HNSCC"</span> <span class="ot">=</span> <span class="st">"CPTAC-HNSC"</span>,</span>
<span id="cb5-85"><a href="#cb5-85" aria-hidden="true" tabindex="-1"></a>  <span class="st">"LSCC"</span> <span class="ot">=</span> <span class="st">"CPTAC-LUSC"</span> ,</span>
<span id="cb5-86"><a href="#cb5-86" aria-hidden="true" tabindex="-1"></a>  <span class="st">"LUAD"</span> <span class="ot">=</span> <span class="st">"CPTAC-LUAD"</span>,</span>
<span id="cb5-87"><a href="#cb5-87" aria-hidden="true" tabindex="-1"></a>  <span class="st">"PDAC"</span> <span class="ot">=</span> <span class="st">"CPTAC-PDAC"</span>,</span>
<span id="cb5-88"><a href="#cb5-88" aria-hidden="true" tabindex="-1"></a>  <span class="st">"UCEC"</span> <span class="ot">=</span> <span class="st">"CPTAC-UCEC"</span>,</span>
<span id="cb5-89"><a href="#cb5-89" aria-hidden="true" tabindex="-1"></a>  <span class="st">"GTEX-Brain"</span> <span class="ot">=</span> <span class="st">"CPTAC-GTEX-Brain"</span></span>
<span id="cb5-90"><a href="#cb5-90" aria-hidden="true" tabindex="-1"></a>)     </span>
<span id="cb5-91"><a href="#cb5-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-92"><a href="#cb5-92" aria-hidden="true" tabindex="-1"></a>matched_CPTAC_info <span class="ot">&lt;-</span> CPTAC_info<span class="sc">$</span>Cancer <span class="sc">%in%</span> <span class="fu">names</span>(cptac_replacement_table2)</span>
<span id="cb5-93"><a href="#cb5-93" aria-hidden="true" tabindex="-1"></a>CPTAC_info_pretreatment <span class="ot">&lt;-</span> CPTAC_info <span class="sc">%&gt;%</span></span>
<span id="cb5-94"><a href="#cb5-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">cohort =</span> <span class="fu">case_when</span>(</span>
<span id="cb5-95"><a href="#cb5-95" aria-hidden="true" tabindex="-1"></a>    matched_CPTAC_info <span class="sc">~</span> cptac_replacement_table2[Cancer],</span>
<span id="cb5-96"><a href="#cb5-96" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="fu">as.character</span>(<span class="cn">NA</span>)  </span>
<span id="cb5-97"><a href="#cb5-97" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">|&gt;</span> </span>
<span id="cb5-98"><a href="#cb5-98" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>Sample_ID)</span>
<span id="cb5-99"><a href="#cb5-99" aria-hidden="true" tabindex="-1"></a><span class="do">### CPTAC_info_pretreatment  |&gt; filter(is.na(cohort)) |&gt; nrow()</span></span>
<span id="cb5-100"><a href="#cb5-100" aria-hidden="true" tabindex="-1"></a><span class="do">### [1] 0</span></span>
<span id="cb5-101"><a href="#cb5-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-102"><a href="#cb5-102" aria-hidden="true" tabindex="-1"></a>matched_CPTAC <span class="ot">&lt;-</span> matched_CPTAC_pretreatment <span class="sc">|&gt;</span></span>
<span id="cb5-103"><a href="#cb5-103" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">left_join</span>(CPTAC_info_pretreatment, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Patient_ID"</span> <span class="ot">=</span> <span class="st">"Patient_ID"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb5-104"><a href="#cb5-104" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">mutate</span>(<span class="at">Run =</span> <span class="fu">str_replace_all</span>(Sample_ID, <span class="st">", "</span>, <span class="st">"-"</span>))</span>
<span id="cb5-105"><a href="#cb5-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-106"><a href="#cb5-106" aria-hidden="true" tabindex="-1"></a><span class="do">### CPTAC Sample type</span></span>
<span id="cb5-107"><a href="#cb5-107" aria-hidden="true" tabindex="-1"></a><span class="do">### matched_CPTAC_pretreatment$Sample_type |&gt; unique()</span></span>
<span id="cb5-108"><a href="#cb5-108" aria-hidden="true" tabindex="-1"></a><span class="do">### [1] "Primary Tumor"                                                             "Solid Tissue Normal"                                                      </span></span>
<span id="cb5-109"><a href="#cb5-109" aria-hidden="true" tabindex="-1"></a><span class="do">### [3] "Primary Tumor, Primary Tumor"                                              "Primary Tumor, Primary Tumor, Primary Tumor"                              </span></span>
<span id="cb5-110"><a href="#cb5-110" aria-hidden="true" tabindex="-1"></a><span class="do">### [5] "Solid Tissue Normal, Solid Tissue Normal, Solid Tissue Normal"             "Solid Tissue Normal, Solid Tissue Normal"                                 </span></span>
<span id="cb5-111"><a href="#cb5-111" aria-hidden="true" tabindex="-1"></a><span class="do">### [7] "Primary Tumor, Primary Tumor, Primary Tumor, Primary Tumor, Primary Tumor" "Primary Tumor, Primary Tumor, Primary Tumor, Primary Tumor"   </span></span>
<span id="cb5-112"><a href="#cb5-112" aria-hidden="true" tabindex="-1"></a><span class="do">### </span></span>
<span id="cb5-113"><a href="#cb5-113" aria-hidden="true" tabindex="-1"></a><span class="do">### Just two sample type so if grepling "Solid Tissue Normal" is normal sample, grepling "Primary Tumor" is tumor sample</span></span>
<span id="cb5-114"><a href="#cb5-114" aria-hidden="true" tabindex="-1"></a>matched_CPTAC_after_processing <span class="ot">&lt;-</span> matched_CPTAC <span class="sc">|&gt;</span></span>
<span id="cb5-115"><a href="#cb5-115" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">cohort =</span> <span class="fu">case_when</span>(</span>
<span id="cb5-116"><a href="#cb5-116" aria-hidden="true" tabindex="-1"></a>    <span class="fu">grepl</span>(<span class="st">"Solid Tissue Normal"</span>, Sample_type) <span class="sc">~</span> <span class="st">"CPTAC-NORMAL"</span>,</span>
<span id="cb5-117"><a href="#cb5-117" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> cohort  </span>
<span id="cb5-118"><a href="#cb5-118" aria-hidden="true" tabindex="-1"></a>  )) </span>
<span id="cb5-119"><a href="#cb5-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-120"><a href="#cb5-120" aria-hidden="true" tabindex="-1"></a><span class="do">### make sure nrow is same between Before and After Processing. </span></span>
<span id="cb5-121"><a href="#cb5-121" aria-hidden="true" tabindex="-1"></a>matched_CPTAC_after_processing <span class="sc">|&gt;</span> <span class="fu">nrow</span>()</span>
<span id="cb5-122"><a href="#cb5-122" aria-hidden="true" tabindex="-1"></a><span class="do">### [1] 2558</span></span>
<span id="cb5-123"><a href="#cb5-123" aria-hidden="true" tabindex="-1"></a>matched_CPTAC_pretreatment <span class="sc">|&gt;</span> <span class="fu">nrow</span>() </span>
<span id="cb5-124"><a href="#cb5-124" aria-hidden="true" tabindex="-1"></a><span class="do">### [1] 2558</span></span>
<span id="cb5-125"><a href="#cb5-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-126"><a href="#cb5-126" aria-hidden="true" tabindex="-1"></a><span class="do">### make sure IOBR/TME data just contain sample which cptac_sheet gave. </span></span>
<span id="cb5-127"><a href="#cb5-127" aria-hidden="true" tabindex="-1"></a>detected_id_CPTAC <span class="ot">&lt;-</span> matched_CPTAC_after_processing<span class="sc">$</span>Run <span class="sc">|&gt;</span> <span class="fu">unique</span>()</span>
<span id="cb5-128"><a href="#cb5-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-129"><a href="#cb5-129" aria-hidden="true" tabindex="-1"></a><span class="do">### save info</span></span>
<span id="cb5-130"><a href="#cb5-130" aria-hidden="true" tabindex="-1"></a>cohorts <span class="ot">&lt;-</span> <span class="fu">unique</span>(matched_CPTAC_after_processing<span class="sc">$</span>cohort)</span>
<span id="cb5-131"><a href="#cb5-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-132"><a href="#cb5-132" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (cohort1 <span class="cf">in</span> cohorts) {</span>
<span id="cb5-133"><a href="#cb5-133" aria-hidden="true" tabindex="-1"></a>  cohort_data <span class="ot">&lt;-</span> <span class="fu">subset</span>(matched_CPTAC_after_processing, cohort <span class="sc">==</span> cohort1)</span>
<span id="cb5-134"><a href="#cb5-134" aria-hidden="true" tabindex="-1"></a>  name <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"-"</span>, <span class="st">"_"</span>, cohort1)</span>
<span id="cb5-135"><a href="#cb5-135" aria-hidden="true" tabindex="-1"></a>  file_name <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"/home/data2/Projects/Fusion/fusiondb_v3/Clininfo/"</span>, name ,<span class="st">"_info.tsv"</span>)</span>
<span id="cb5-136"><a href="#cb5-136" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fwrite</span>(cohort_data , <span class="at">file =</span> file_name, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)  </span>
<span id="cb5-137"><a href="#cb5-137" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-138"><a href="#cb5-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-139"><a href="#cb5-139" aria-hidden="true" tabindex="-1"></a><span class="do">## TME</span></span>
<span id="cb5-140"><a href="#cb5-140" aria-hidden="true" tabindex="-1"></a>CPTAC2_TME <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/home/data2/Projects/Fusion/fusiondb/IOBR/CPTAC-2_TME.tsv.gz"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>) <span class="sc">|&gt;</span> <span class="fu">mutate</span>(<span class="at">Project_id =</span> <span class="st">"CPTAC2"</span>)</span>
<span id="cb5-141"><a href="#cb5-141" aria-hidden="true" tabindex="-1"></a>CPTAC3_TME <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/home/data2/Projects/Fusion/fusiondb/IOBR/CPTAC-3_TME.tsv.gz"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>) <span class="sc">|&gt;</span> <span class="fu">mutate</span>(<span class="at">Project_id =</span> <span class="st">"CPTAC3"</span>)</span>
<span id="cb5-142"><a href="#cb5-142" aria-hidden="true" tabindex="-1"></a>CPTAC_TME <span class="ot">&lt;-</span> <span class="fu">rbind</span>(CPTAC2_TME, CPTAC3_TME)</span>
<span id="cb5-143"><a href="#cb5-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-144"><a href="#cb5-144" aria-hidden="true" tabindex="-1"></a><span class="do">### Match matched_CPTAC_after_processing$Run to CPTAC_TME$ID</span></span>
<span id="cb5-145"><a href="#cb5-145" aria-hidden="true" tabindex="-1"></a><span class="do">### </span></span>
<span id="cb5-146"><a href="#cb5-146" aria-hidden="true" tabindex="-1"></a><span class="do">### CPTAC_TME can match</span></span>
<span id="cb5-147"><a href="#cb5-147" aria-hidden="true" tabindex="-1"></a>matched_can <span class="ot">&lt;-</span> CPTAC_TME <span class="sc">%&gt;%</span></span>
<span id="cb5-148"><a href="#cb5-148" aria-hidden="true" tabindex="-1"></a>               <span class="fu">filter</span>(ID <span class="sc">%in%</span> matched_CPTAC_after_processing<span class="sc">$</span>Run) </span>
<span id="cb5-149"><a href="#cb5-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-150"><a href="#cb5-150" aria-hidden="true" tabindex="-1"></a><span class="do">### </span></span>
<span id="cb5-151"><a href="#cb5-151" aria-hidden="true" tabindex="-1"></a><span class="do">### CPTAC_TME can't match</span></span>
<span id="cb5-152"><a href="#cb5-152" aria-hidden="true" tabindex="-1"></a>matched_not <span class="ot">&lt;-</span> CPTAC_TME <span class="sc">%&gt;%</span></span>
<span id="cb5-153"><a href="#cb5-153" aria-hidden="true" tabindex="-1"></a>               <span class="fu">filter</span>(<span class="sc">!</span>(ID <span class="sc">%in%</span> matched_CPTAC_after_processing<span class="sc">$</span>Run))</span>
<span id="cb5-154"><a href="#cb5-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-155"><a href="#cb5-155" aria-hidden="true" tabindex="-1"></a><span class="do">### </span></span>
<span id="cb5-156"><a href="#cb5-156" aria-hidden="true" tabindex="-1"></a><span class="do">### situtation 1: Due to the barcode naming convention and etc, some portion character are added to the endning of the IOBR/TME sample name.</span></span>
<span id="cb5-157"><a href="#cb5-157" aria-hidden="true" tabindex="-1"></a><span class="do">### str_count(matched_last$ID, "_")</span></span>
<span id="cb5-158"><a href="#cb5-158" aria-hidden="true" tabindex="-1"></a><span class="do">### [1] 1 2 0</span></span>
<span id="cb5-159"><a href="#cb5-159" aria-hidden="true" tabindex="-1"></a><span class="do">### 1.1 "CPT000817-0008" and "CPT000817". If the value of str_count(matched_last$ID, "_") for a row is 0, then the value of the prefixes column for that row should be the characters before the last -, and the value of the suffixes column should be the characters after the last -.</span></span>
<span id="cb5-160"><a href="#cb5-160" aria-hidden="true" tabindex="-1"></a><span class="do">### 1.2 "f6a7f764-2495-4a1d-89a2-e12f51_D7_1" and "f6a7f764-2495-4a1d-89a2-e12f51". If the value of str_count(matched_last$ID, "_") for a row is 1, then the value of the prefixes column for that row should be the characters before the first _, and the value of the suffixes column should be the characters after the first _.</span></span>
<span id="cb5-161"><a href="#cb5-161" aria-hidden="true" tabindex="-1"></a><span class="do">### 1.3 "CPT008756_0001" and "CPT008756". f6a7f764-2495-4a1d-89a2-e12f51_D7_1, .If the value of str_count(matched_last$ID, "_") for a row is 2, then the value of the prefixes column for that row should be the characters before the first _, and the value of the suffixes column should be the characters after the first _.</span></span>
<span id="cb5-162"><a href="#cb5-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-163"><a href="#cb5-163" aria-hidden="true" tabindex="-1"></a>matched_not <span class="ot">&lt;-</span> matched_not <span class="sc">%&gt;%</span></span>
<span id="cb5-164"><a href="#cb5-164" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span>  </span>
<span id="cb5-165"><a href="#cb5-165" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb5-166"><a href="#cb5-166" aria-hidden="true" tabindex="-1"></a>    <span class="at">prefixes =</span> <span class="fu">case_when</span>(</span>
<span id="cb5-167"><a href="#cb5-167" aria-hidden="true" tabindex="-1"></a>      <span class="fu">str_count</span>(ID, <span class="st">"_"</span>) <span class="sc">==</span> <span class="dv">0</span> <span class="sc">~</span> <span class="fu">str_split</span>(ID, <span class="st">"-"</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>],  </span>
<span id="cb5-168"><a href="#cb5-168" aria-hidden="true" tabindex="-1"></a>      <span class="fu">str_count</span>(ID, <span class="st">"_"</span>) <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="fu">str_split</span>(ID, <span class="st">"_"</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>],  </span>
<span id="cb5-169"><a href="#cb5-169" aria-hidden="true" tabindex="-1"></a>      <span class="fu">str_count</span>(ID, <span class="st">"_"</span>) <span class="sc">==</span> <span class="dv">2</span> <span class="sc">~</span> <span class="fu">str_split</span>(ID, <span class="st">"_"</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>],  </span>
<span id="cb5-170"><a href="#cb5-170" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="cn">NA_character_</span></span>
<span id="cb5-171"><a href="#cb5-171" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb5-172"><a href="#cb5-172" aria-hidden="true" tabindex="-1"></a>    <span class="at">suffixes =</span> <span class="fu">case_when</span>(</span>
<span id="cb5-173"><a href="#cb5-173" aria-hidden="true" tabindex="-1"></a>      <span class="fu">str_count</span>(ID, <span class="st">"_"</span>) <span class="sc">==</span> <span class="dv">0</span> <span class="sc">~</span> <span class="fu">str_split</span>(ID, <span class="st">"-"</span>)[[<span class="dv">1</span>]][<span class="fu">length</span>(<span class="fu">str_split</span>(ID, <span class="st">"-"</span>)[[<span class="dv">1</span>]])],  </span>
<span id="cb5-174"><a href="#cb5-174" aria-hidden="true" tabindex="-1"></a>      <span class="fu">str_count</span>(ID, <span class="st">"_"</span>) <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="fu">str_split</span>(ID, <span class="st">"_"</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>],  </span>
<span id="cb5-175"><a href="#cb5-175" aria-hidden="true" tabindex="-1"></a>      <span class="fu">str_count</span>(ID, <span class="st">"_"</span>) <span class="sc">==</span> <span class="dv">2</span> <span class="sc">~</span> <span class="fu">str_split</span>(ID, <span class="st">"_"</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>],  </span>
<span id="cb5-176"><a href="#cb5-176" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="cn">NA_character_</span></span>
<span id="cb5-177"><a href="#cb5-177" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb5-178"><a href="#cb5-178" aria-hidden="true" tabindex="-1"></a>    <span class="at">name1 =</span> prefixes</span>
<span id="cb5-179"><a href="#cb5-179" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb5-180"><a href="#cb5-180" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() </span>
<span id="cb5-181"><a href="#cb5-181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-182"><a href="#cb5-182" aria-hidden="true" tabindex="-1"></a><span class="do">### situtation 1 after processing can match</span></span>
<span id="cb5-183"><a href="#cb5-183" aria-hidden="true" tabindex="-1"></a>matched_in_situtation1 <span class="ot">&lt;-</span>  matched_not <span class="sc">%&gt;%</span></span>
<span id="cb5-184"><a href="#cb5-184" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">filter</span>((name1 <span class="sc">%in%</span> matched_CPTAC_after_processing<span class="sc">$</span>Run))<span class="sc">%&gt;%</span></span>
<span id="cb5-185"><a href="#cb5-185" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">mutate</span>(<span class="at">ID =</span> name1)  <span class="sc">%&gt;%</span></span>
<span id="cb5-186"><a href="#cb5-186" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(suffixes,prefixes,name1) )     </span>
<span id="cb5-187"><a href="#cb5-187" aria-hidden="true" tabindex="-1"></a>matched_can <span class="ot">&lt;-</span> <span class="fu">rbind</span>(matched_can, matched_in_situtation1)</span>
<span id="cb5-188"><a href="#cb5-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-189"><a href="#cb5-189" aria-hidden="true" tabindex="-1"></a><span class="do">### not in situtation 1 </span></span>
<span id="cb5-190"><a href="#cb5-190" aria-hidden="true" tabindex="-1"></a>matched_not_situtation1 <span class="ot">&lt;-</span> matched_not <span class="sc">%&gt;%</span></span>
<span id="cb5-191"><a href="#cb5-191" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">filter</span>(<span class="sc">!</span>(name1 <span class="sc">%in%</span> matched_CPTAC_after_processing<span class="sc">$</span>Run))</span>
<span id="cb5-192"><a href="#cb5-192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-193"><a href="#cb5-193" aria-hidden="true" tabindex="-1"></a><span class="do">### situtation 2: Due to IOBR package analyze code, an additional "X" is added to the beginning of the IOBR/TME sample name.</span></span>
<span id="cb5-194"><a href="#cb5-194" aria-hidden="true" tabindex="-1"></a><span class="do">### "X1104806" and "1104806", "X9f905736-f662-41d6-b3ac-16758d" and "9f905736-f662-41d6-b3ac-16758d". </span></span>
<span id="cb5-195"><a href="#cb5-195" aria-hidden="true" tabindex="-1"></a>matched_not_situtation1  <span class="ot">&lt;-</span> matched_not_situtation1 <span class="sc">%&gt;%</span></span>
<span id="cb5-196"><a href="#cb5-196" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">mutate</span>(<span class="at">name2 =</span> <span class="fu">str_remove</span>(ID, <span class="st">"^X"</span>))  </span>
<span id="cb5-197"><a href="#cb5-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-198"><a href="#cb5-198" aria-hidden="true" tabindex="-1"></a><span class="do">### situtation 2 after processing can match</span></span>
<span id="cb5-199"><a href="#cb5-199" aria-hidden="true" tabindex="-1"></a>matched_in_situtation2 <span class="ot">&lt;-</span> matched_not_situtation1 <span class="sc">%&gt;%</span></span>
<span id="cb5-200"><a href="#cb5-200" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">filter</span>((name2 <span class="sc">%in%</span> matched_CPTAC_after_processing<span class="sc">$</span>Run))    <span class="sc">%&gt;%</span></span>
<span id="cb5-201"><a href="#cb5-201" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">mutate</span>(<span class="at">ID =</span> name2 )  <span class="sc">%&gt;%</span></span>
<span id="cb5-202"><a href="#cb5-202" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(name1,prefixes,suffixes,name2)) </span>
<span id="cb5-203"><a href="#cb5-203" aria-hidden="true" tabindex="-1"></a>matched_can <span class="ot">&lt;-</span> <span class="fu">rbind</span>(matched_can, matched_in_situtation2)</span>
<span id="cb5-204"><a href="#cb5-204" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-205"><a href="#cb5-205" aria-hidden="true" tabindex="-1"></a><span class="do">### situtation 2 after processing can't match</span></span>
<span id="cb5-206"><a href="#cb5-206" aria-hidden="true" tabindex="-1"></a>matched_not_situtation2 <span class="ot">&lt;-</span> matched_not_situtation1 <span class="sc">%&gt;%</span></span>
<span id="cb5-207"><a href="#cb5-207" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">filter</span>(<span class="sc">!</span>(name2 <span class="sc">%in%</span> matched_CPTAC_after_processing<span class="sc">$</span>Run))</span>
<span id="cb5-208"><a href="#cb5-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-209"><a href="#cb5-209" aria-hidden="true" tabindex="-1"></a><span class="do">### situtation 3: Mixed situation1 and situation2.</span></span>
<span id="cb5-210"><a href="#cb5-210" aria-hidden="true" tabindex="-1"></a><span class="do">### X161317f3-1a39-438e-af45-cd2502_D7 and 161317f3-1a39-438e-af45-cd2502</span></span>
<span id="cb5-211"><a href="#cb5-211" aria-hidden="true" tabindex="-1"></a>matched_not_situtation2  <span class="ot">&lt;-</span> matched_not_situtation2 <span class="sc">%&gt;%</span></span>
<span id="cb5-212"><a href="#cb5-212" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">mutate</span>(<span class="at">name3 =</span> <span class="fu">str_remove</span>(prefixes, <span class="st">"^X"</span>)) </span>
<span id="cb5-213"><a href="#cb5-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-214"><a href="#cb5-214" aria-hidden="true" tabindex="-1"></a><span class="do">### situtation 3 after processing can match    </span></span>
<span id="cb5-215"><a href="#cb5-215" aria-hidden="true" tabindex="-1"></a>matched_in_situtation3 <span class="ot">&lt;-</span> matched_not_situtation2  <span class="sc">%&gt;%</span></span>
<span id="cb5-216"><a href="#cb5-216" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">filter</span>((name3 <span class="sc">%in%</span> matched_CPTAC_after_processing<span class="sc">$</span>Run))  <span class="sc">%&gt;%</span></span>
<span id="cb5-217"><a href="#cb5-217" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">mutate</span>(<span class="at">ID =</span> name3) <span class="sc">%&gt;%</span></span>
<span id="cb5-218"><a href="#cb5-218" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(name1,name2,prefixes,suffixes, name3))           </span>
<span id="cb5-219"><a href="#cb5-219" aria-hidden="true" tabindex="-1"></a>matched_can <span class="ot">&lt;-</span> <span class="fu">rbind</span>(matched_can, matched_in_situtation3)</span>
<span id="cb5-220"><a href="#cb5-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-221"><a href="#cb5-221" aria-hidden="true" tabindex="-1"></a><span class="do">### situtation 3 after processing can't match</span></span>
<span id="cb5-222"><a href="#cb5-222" aria-hidden="true" tabindex="-1"></a>matched_not_situtation3 <span class="ot">&lt;-</span> matched_not_situtation2  <span class="sc">%&gt;%</span></span>
<span id="cb5-223"><a href="#cb5-223" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">filter</span>(<span class="sc">!</span>(name3 <span class="sc">%in%</span> matched_CPTAC_after_processing<span class="sc">$</span>Run))</span>
<span id="cb5-224"><a href="#cb5-224" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(matched_not_situtation3 )</span>
<span id="cb5-225"><a href="#cb5-225" aria-hidden="true" tabindex="-1"></a><span class="do">### [1] 81</span></span>
<span id="cb5-226"><a href="#cb5-226" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-227"><a href="#cb5-227" aria-hidden="true" tabindex="-1"></a><span class="do">### Check the reason why it still can't match after Type 3 processing</span></span>
<span id="cb5-228"><a href="#cb5-228" aria-hidden="true" tabindex="-1"></a><span class="do">### After manual review, these Sample_ID are not included in either the info dataset or the gdc_sheet.</span></span>
<span id="cb5-229"><a href="#cb5-229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-230"><a href="#cb5-230" aria-hidden="true" tabindex="-1"></a><span class="do">### Check if CPTAC TME data ID are any samples that have been divided into multiple portions.</span></span>
<span id="cb5-231"><a href="#cb5-231" aria-hidden="true" tabindex="-1"></a>matched_can <span class="sc">|&gt;</span> <span class="fu">select</span>(ID) <span class="sc">|&gt;</span>  <span class="fu">nrow</span>()</span>
<span id="cb5-232"><a href="#cb5-232" aria-hidden="true" tabindex="-1"></a><span class="do">### [1] 2558</span></span>
<span id="cb5-233"><a href="#cb5-233" aria-hidden="true" tabindex="-1"></a>matched_can <span class="sc">|&gt;</span> <span class="fu">select</span>(ID) <span class="sc">|&gt;</span>  <span class="fu">unique</span>() <span class="sc">|&gt;</span> <span class="fu">nrow</span>()</span>
<span id="cb5-234"><a href="#cb5-234" aria-hidden="true" tabindex="-1"></a><span class="do">### [1] 2558</span></span>
<span id="cb5-235"><a href="#cb5-235" aria-hidden="true" tabindex="-1"></a><span class="do">### No this situation.</span></span>
<span id="cb5-236"><a href="#cb5-236" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-237"><a href="#cb5-237" aria-hidden="true" tabindex="-1"></a><span class="do">### Check if CPTAC TME data has same norw with CPTAC info data.</span></span>
<span id="cb5-238"><a href="#cb5-238" aria-hidden="true" tabindex="-1"></a>matched_can <span class="sc">|&gt;</span> <span class="fu">select</span>(ID) <span class="sc">|&gt;</span>   <span class="fu">unique</span>() <span class="sc">|&gt;</span> <span class="fu">nrow</span>()</span>
<span id="cb5-239"><a href="#cb5-239" aria-hidden="true" tabindex="-1"></a><span class="do">### [1] 2558</span></span>
<span id="cb5-240"><a href="#cb5-240" aria-hidden="true" tabindex="-1"></a>matched_CPTAC_after_processing <span class="sc">|&gt;</span> <span class="fu">select</span>(Run) <span class="sc">|&gt;</span>  <span class="fu">unique</span>() <span class="sc">|&gt;</span> <span class="fu">nrow</span>()</span>
<span id="cb5-241"><a href="#cb5-241" aria-hidden="true" tabindex="-1"></a><span class="do">### [1] 2558</span></span>
<span id="cb5-242"><a href="#cb5-242" aria-hidden="true" tabindex="-1"></a><span class="do">### Sure.</span></span>
<span id="cb5-243"><a href="#cb5-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-244"><a href="#cb5-244" aria-hidden="true" tabindex="-1"></a>cohort_data <span class="ot">&lt;-</span> matched_CPTAC_after_processing <span class="sc">|&gt;</span> </span>
<span id="cb5-245"><a href="#cb5-245" aria-hidden="true" tabindex="-1"></a>               <span class="fu">select</span>(Run, cohort)   <span class="sc">|&gt;</span></span>
<span id="cb5-246"><a href="#cb5-246" aria-hidden="true" tabindex="-1"></a>               <span class="fu">unique</span>()   </span>
<span id="cb5-247"><a href="#cb5-247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-248"><a href="#cb5-248" aria-hidden="true" tabindex="-1"></a>matched_can <span class="ot">&lt;-</span> matched_can  <span class="sc">|&gt;</span></span>
<span id="cb5-249"><a href="#cb5-249" aria-hidden="true" tabindex="-1"></a>               <span class="fu">mutate</span>(<span class="at">Run =</span> ID)  <span class="sc">|&gt;</span></span>
<span id="cb5-250"><a href="#cb5-250" aria-hidden="true" tabindex="-1"></a>               <span class="fu">left_join</span>(cohort_data, <span class="at">by =</span> <span class="st">"Run"</span>)  <span class="sc">|&gt;</span></span>
<span id="cb5-251"><a href="#cb5-251" aria-hidden="true" tabindex="-1"></a>               <span class="fu">select</span>(<span class="sc">-</span>Run)</span>
<span id="cb5-252"><a href="#cb5-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-253"><a href="#cb5-253" aria-hidden="true" tabindex="-1"></a>cohorts_TME <span class="ot">&lt;-</span> <span class="fu">unique</span>(matched_can<span class="sc">$</span>cohort)</span>
<span id="cb5-254"><a href="#cb5-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-255"><a href="#cb5-255" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (cohort1 <span class="cf">in</span> cohorts_TME) {</span>
<span id="cb5-256"><a href="#cb5-256" aria-hidden="true" tabindex="-1"></a>  cohort_data <span class="ot">&lt;-</span> <span class="fu">subset</span>(matched_can, cohort <span class="sc">==</span> cohort1) <span class="sc">|&gt;</span> <span class="fu">select</span>(<span class="sc">-</span>cohort)</span>
<span id="cb5-257"><a href="#cb5-257" aria-hidden="true" tabindex="-1"></a>  file_name <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"/home/data2/Projects/Fusion/fusiondb_v3/IOBR/"</span>, cohort1 ,<span class="st">"_TME.tsv.gz"</span>)</span>
<span id="cb5-258"><a href="#cb5-258" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(cohort_data, </span>
<span id="cb5-259"><a href="#cb5-259" aria-hidden="true" tabindex="-1"></a>              <span class="at">file =</span> <span class="fu">gzfile</span>(file_name), </span>
<span id="cb5-260"><a href="#cb5-260" aria-hidden="true" tabindex="-1"></a>              <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, </span>
<span id="cb5-261"><a href="#cb5-261" aria-hidden="true" tabindex="-1"></a>              <span class="at">row.names =</span> <span class="cn">FALSE</span>, </span>
<span id="cb5-262"><a href="#cb5-262" aria-hidden="true" tabindex="-1"></a>              <span class="at">quote =</span> <span class="cn">FALSE</span>) </span>
<span id="cb5-263"><a href="#cb5-263" aria-hidden="true" tabindex="-1"></a>}  </span>
<span id="cb5-264"><a href="#cb5-264" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-265"><a href="#cb5-265" aria-hidden="true" tabindex="-1"></a><span class="co">#IOBR</span></span>
<span id="cb5-266"><a href="#cb5-266" aria-hidden="true" tabindex="-1"></a>CPTAC2_Signature <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/home/data2/Projects/Fusion/fusiondb/IOBR/CPTAC-2_IOBR_signature.tsv.gz"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>) <span class="sc">|&gt;</span> <span class="fu">mutate</span>(<span class="at">Project_id =</span> <span class="st">"CPTAC2"</span>)</span>
<span id="cb5-267"><a href="#cb5-267" aria-hidden="true" tabindex="-1"></a>CPTAC3_Signature <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/home/data2/Projects/Fusion/fusiondb/IOBR/CPTAC-3_IOBR_signature.tsv.gz"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>) <span class="sc">|&gt;</span> <span class="fu">mutate</span>(<span class="at">Project_id =</span> <span class="st">"CPTAC3"</span>)</span>
<span id="cb5-268"><a href="#cb5-268" aria-hidden="true" tabindex="-1"></a>CPTAC_Signature <span class="ot">&lt;-</span> <span class="fu">rbind</span>(CPTAC2_Signature, CPTAC3_Signature)</span>
<span id="cb5-269"><a href="#cb5-269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-270"><a href="#cb5-270" aria-hidden="true" tabindex="-1"></a><span class="do">### Match matched_CPTAC_after_processing$Run to CPTAC_TME$ID</span></span>
<span id="cb5-271"><a href="#cb5-271" aria-hidden="true" tabindex="-1"></a><span class="do">### </span></span>
<span id="cb5-272"><a href="#cb5-272" aria-hidden="true" tabindex="-1"></a><span class="do">### CPTAC_TME can match</span></span>
<span id="cb5-273"><a href="#cb5-273" aria-hidden="true" tabindex="-1"></a>matched_can <span class="ot">&lt;-</span> CPTAC_Signature <span class="sc">%&gt;%</span></span>
<span id="cb5-274"><a href="#cb5-274" aria-hidden="true" tabindex="-1"></a>               <span class="fu">filter</span>(ID <span class="sc">%in%</span> matched_CPTAC_after_processing<span class="sc">$</span>Run) </span>
<span id="cb5-275"><a href="#cb5-275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-276"><a href="#cb5-276" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-277"><a href="#cb5-277" aria-hidden="true" tabindex="-1"></a><span class="do">### </span></span>
<span id="cb5-278"><a href="#cb5-278" aria-hidden="true" tabindex="-1"></a><span class="do">### CPTAC_TME can't match</span></span>
<span id="cb5-279"><a href="#cb5-279" aria-hidden="true" tabindex="-1"></a>matched_not <span class="ot">&lt;-</span>  CPTAC_Signature <span class="sc">%&gt;%</span></span>
<span id="cb5-280"><a href="#cb5-280" aria-hidden="true" tabindex="-1"></a>                <span class="fu">filter</span>(<span class="sc">!</span>(ID <span class="sc">%in%</span> matched_CPTAC_after_processing<span class="sc">$</span>Run))</span>
<span id="cb5-281"><a href="#cb5-281" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-282"><a href="#cb5-282" aria-hidden="true" tabindex="-1"></a><span class="do">### </span></span>
<span id="cb5-283"><a href="#cb5-283" aria-hidden="true" tabindex="-1"></a><span class="do">### situtation 1:like TME </span></span>
<span id="cb5-284"><a href="#cb5-284" aria-hidden="true" tabindex="-1"></a>matched_not <span class="ot">&lt;-</span> matched_not <span class="sc">%&gt;%</span></span>
<span id="cb5-285"><a href="#cb5-285" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span>  </span>
<span id="cb5-286"><a href="#cb5-286" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb5-287"><a href="#cb5-287" aria-hidden="true" tabindex="-1"></a>    <span class="at">prefixes =</span> <span class="fu">case_when</span>(</span>
<span id="cb5-288"><a href="#cb5-288" aria-hidden="true" tabindex="-1"></a>      <span class="fu">str_count</span>(ID, <span class="st">"_"</span>) <span class="sc">==</span> <span class="dv">0</span> <span class="sc">~</span> <span class="fu">str_split</span>(ID, <span class="st">"-"</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>],  </span>
<span id="cb5-289"><a href="#cb5-289" aria-hidden="true" tabindex="-1"></a>      <span class="fu">str_count</span>(ID, <span class="st">"_"</span>) <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="fu">str_split</span>(ID, <span class="st">"_"</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>],  </span>
<span id="cb5-290"><a href="#cb5-290" aria-hidden="true" tabindex="-1"></a>      <span class="fu">str_count</span>(ID, <span class="st">"_"</span>) <span class="sc">==</span> <span class="dv">2</span> <span class="sc">~</span> <span class="fu">str_split</span>(ID, <span class="st">"_"</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>], </span>
<span id="cb5-291"><a href="#cb5-291" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="cn">NA_character_</span></span>
<span id="cb5-292"><a href="#cb5-292" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb5-293"><a href="#cb5-293" aria-hidden="true" tabindex="-1"></a>    <span class="at">suffixes =</span> <span class="fu">case_when</span>(</span>
<span id="cb5-294"><a href="#cb5-294" aria-hidden="true" tabindex="-1"></a>      <span class="fu">str_count</span>(ID, <span class="st">"_"</span>) <span class="sc">==</span> <span class="dv">0</span> <span class="sc">~</span> <span class="fu">str_split</span>(ID, <span class="st">"-"</span>)[[<span class="dv">1</span>]][<span class="fu">length</span>(<span class="fu">str_split</span>(ID, <span class="st">"-"</span>)[[<span class="dv">1</span>]])],  </span>
<span id="cb5-295"><a href="#cb5-295" aria-hidden="true" tabindex="-1"></a>      <span class="fu">str_count</span>(ID, <span class="st">"_"</span>) <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="fu">str_split</span>(ID, <span class="st">"_"</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>],  </span>
<span id="cb5-296"><a href="#cb5-296" aria-hidden="true" tabindex="-1"></a>      <span class="fu">str_count</span>(ID, <span class="st">"_"</span>) <span class="sc">==</span> <span class="dv">2</span> <span class="sc">~</span> <span class="fu">str_split</span>(ID, <span class="st">"_"</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>],  </span>
<span id="cb5-297"><a href="#cb5-297" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="cn">NA_character_</span></span>
<span id="cb5-298"><a href="#cb5-298" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb5-299"><a href="#cb5-299" aria-hidden="true" tabindex="-1"></a>    <span class="at">name1 =</span> prefixes</span>
<span id="cb5-300"><a href="#cb5-300" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb5-301"><a href="#cb5-301" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()  </span>
<span id="cb5-302"><a href="#cb5-302" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-303"><a href="#cb5-303" aria-hidden="true" tabindex="-1"></a><span class="do">### situtation 1 after processing can match</span></span>
<span id="cb5-304"><a href="#cb5-304" aria-hidden="true" tabindex="-1"></a>matched_in_situtation1 <span class="ot">&lt;-</span> matched_not <span class="sc">%&gt;%</span></span>
<span id="cb5-305"><a href="#cb5-305" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">filter</span>((name1 <span class="sc">%in%</span> matched_CPTAC_after_processing<span class="sc">$</span>Run))<span class="sc">%&gt;%</span></span>
<span id="cb5-306"><a href="#cb5-306" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">mutate</span>(<span class="at">ID =</span> name1)  <span class="sc">%&gt;%</span></span>
<span id="cb5-307"><a href="#cb5-307" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(suffixes,prefixes,name1) )    </span>
<span id="cb5-308"><a href="#cb5-308" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-309"><a href="#cb5-309" aria-hidden="true" tabindex="-1"></a>matched_can <span class="ot">&lt;-</span> <span class="fu">rbind</span>(matched_can, matched_in_situtation1)</span>
<span id="cb5-310"><a href="#cb5-310" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-311"><a href="#cb5-311" aria-hidden="true" tabindex="-1"></a><span class="do">### situtation 1 after processing can't match</span></span>
<span id="cb5-312"><a href="#cb5-312" aria-hidden="true" tabindex="-1"></a>matched_not_situtation1 <span class="ot">&lt;-</span> matched_not <span class="sc">%&gt;%</span></span>
<span id="cb5-313"><a href="#cb5-313" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">filter</span>(<span class="sc">!</span>(name1 <span class="sc">%in%</span> matched_CPTAC_after_processing<span class="sc">$</span>Run))</span>
<span id="cb5-314"><a href="#cb5-314" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-315"><a href="#cb5-315" aria-hidden="true" tabindex="-1"></a><span class="do">### situtation 2:like TME </span></span>
<span id="cb5-316"><a href="#cb5-316" aria-hidden="true" tabindex="-1"></a>matched_not_situtation1  <span class="ot">&lt;-</span> matched_not_situtation1 <span class="sc">%&gt;%</span></span>
<span id="cb5-317"><a href="#cb5-317" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">mutate</span>(<span class="at">name2 =</span> <span class="fu">str_remove</span>(ID, <span class="st">"^X"</span>))  </span>
<span id="cb5-318"><a href="#cb5-318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-319"><a href="#cb5-319" aria-hidden="true" tabindex="-1"></a><span class="do">### situtation 2 after processing can match</span></span>
<span id="cb5-320"><a href="#cb5-320" aria-hidden="true" tabindex="-1"></a>matched_in_situtation2 <span class="ot">&lt;-</span> matched_not_situtation1 <span class="sc">%&gt;%</span></span>
<span id="cb5-321"><a href="#cb5-321" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">filter</span>((name2 <span class="sc">%in%</span> matched_CPTAC_after_processing<span class="sc">$</span>Run))    <span class="sc">%&gt;%</span></span>
<span id="cb5-322"><a href="#cb5-322" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">mutate</span>(<span class="at">ID =</span> name2 )  <span class="sc">%&gt;%</span></span>
<span id="cb5-323"><a href="#cb5-323" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(name1,prefixes,suffixes,name2)) </span>
<span id="cb5-324"><a href="#cb5-324" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-325"><a href="#cb5-325" aria-hidden="true" tabindex="-1"></a>matched_can <span class="ot">&lt;-</span> <span class="fu">rbind</span>(matched_can, matched_in_situtation2)</span>
<span id="cb5-326"><a href="#cb5-326" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-327"><a href="#cb5-327" aria-hidden="true" tabindex="-1"></a><span class="do">### situtation 2 after processing can't match</span></span>
<span id="cb5-328"><a href="#cb5-328" aria-hidden="true" tabindex="-1"></a>matched_not_situtation2 <span class="ot">&lt;-</span> matched_not_situtation1 <span class="sc">%&gt;%</span></span>
<span id="cb5-329"><a href="#cb5-329" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">filter</span>(<span class="sc">!</span>(name2 <span class="sc">%in%</span> matched_CPTAC_after_processing<span class="sc">$</span>Run))</span>
<span id="cb5-330"><a href="#cb5-330" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-331"><a href="#cb5-331" aria-hidden="true" tabindex="-1"></a><span class="do">### situtation 3:like TME </span></span>
<span id="cb5-332"><a href="#cb5-332" aria-hidden="true" tabindex="-1"></a>matched_not_situtation2   <span class="ot">&lt;-</span> matched_not_situtation2  <span class="sc">%&gt;%</span></span>
<span id="cb5-333"><a href="#cb5-333" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">mutate</span>(<span class="at">name3 =</span> <span class="fu">str_remove</span>(prefixes, <span class="st">"^X"</span>)) </span>
<span id="cb5-334"><a href="#cb5-334" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-335"><a href="#cb5-335" aria-hidden="true" tabindex="-1"></a><span class="do">### situtation 3 after processing can match   </span></span>
<span id="cb5-336"><a href="#cb5-336" aria-hidden="true" tabindex="-1"></a>matched_in_situtation3 <span class="ot">&lt;-</span> matched_not_situtation2  <span class="sc">%&gt;%</span></span>
<span id="cb5-337"><a href="#cb5-337" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">filter</span>((name3 <span class="sc">%in%</span> matched_CPTAC_after_processing<span class="sc">$</span>Run))  <span class="sc">%&gt;%</span></span>
<span id="cb5-338"><a href="#cb5-338" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">mutate</span>(<span class="at">ID =</span> name3) <span class="sc">%&gt;%</span></span>
<span id="cb5-339"><a href="#cb5-339" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(name1,name2,prefixes,suffixes, name3))     </span>
<span id="cb5-340"><a href="#cb5-340" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-341"><a href="#cb5-341" aria-hidden="true" tabindex="-1"></a>matched_can <span class="ot">&lt;-</span> <span class="fu">rbind</span>(matched_can, matched_in_situtation3)</span>
<span id="cb5-342"><a href="#cb5-342" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-343"><a href="#cb5-343" aria-hidden="true" tabindex="-1"></a><span class="do">### situtation 3 after processing can't match  </span></span>
<span id="cb5-344"><a href="#cb5-344" aria-hidden="true" tabindex="-1"></a>matched_not_situtation3 <span class="ot">&lt;-</span> matched_not_situtation2  <span class="sc">%&gt;%</span></span>
<span id="cb5-345"><a href="#cb5-345" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">filter</span>(<span class="sc">!</span>(name3 <span class="sc">%in%</span> matched_CPTAC_after_processing<span class="sc">$</span>Run))</span>
<span id="cb5-346"><a href="#cb5-346" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-347"><a href="#cb5-347" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(matched_not_situtation3 )</span>
<span id="cb5-348"><a href="#cb5-348" aria-hidden="true" tabindex="-1"></a><span class="do">### [1] 81</span></span>
<span id="cb5-349"><a href="#cb5-349" aria-hidden="true" tabindex="-1"></a>                          </span>
<span id="cb5-350"><a href="#cb5-350" aria-hidden="true" tabindex="-1"></a><span class="do">### Check the reason why it still can't match after Type 3 processing</span></span>
<span id="cb5-351"><a href="#cb5-351" aria-hidden="true" tabindex="-1"></a><span class="do">### After manual review, these Sample_ID are not included in either the info dataset or the gdc_sheet.</span></span>
<span id="cb5-352"><a href="#cb5-352" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-353"><a href="#cb5-353" aria-hidden="true" tabindex="-1"></a><span class="do">### Check if CPTAC IOBR data ID are any samples that have been divided into multiple portions.</span></span>
<span id="cb5-354"><a href="#cb5-354" aria-hidden="true" tabindex="-1"></a>matched_can <span class="sc">|&gt;</span> <span class="fu">select</span>(ID) <span class="sc">|&gt;</span>  <span class="fu">nrow</span>()</span>
<span id="cb5-355"><a href="#cb5-355" aria-hidden="true" tabindex="-1"></a><span class="do">### [1] 2558</span></span>
<span id="cb5-356"><a href="#cb5-356" aria-hidden="true" tabindex="-1"></a>matched_can <span class="sc">|&gt;</span> <span class="fu">select</span>(ID) <span class="sc">|&gt;</span>  <span class="fu">unique</span>() <span class="sc">|&gt;</span> <span class="fu">nrow</span>()</span>
<span id="cb5-357"><a href="#cb5-357" aria-hidden="true" tabindex="-1"></a><span class="do">### [1] 2558</span></span>
<span id="cb5-358"><a href="#cb5-358" aria-hidden="true" tabindex="-1"></a><span class="do">### No this situation.</span></span>
<span id="cb5-359"><a href="#cb5-359" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-360"><a href="#cb5-360" aria-hidden="true" tabindex="-1"></a><span class="do">### Check if CPTAC IOBR data has same norw with CPTAC info data.</span></span>
<span id="cb5-361"><a href="#cb5-361" aria-hidden="true" tabindex="-1"></a>matched_can <span class="sc">|&gt;</span> <span class="fu">select</span>(ID) <span class="sc">|&gt;</span>   <span class="fu">unique</span>() <span class="sc">|&gt;</span> <span class="fu">nrow</span>()</span>
<span id="cb5-362"><a href="#cb5-362" aria-hidden="true" tabindex="-1"></a><span class="do">### [1] 2558</span></span>
<span id="cb5-363"><a href="#cb5-363" aria-hidden="true" tabindex="-1"></a>matched_CPTAC_after_processing <span class="sc">|&gt;</span> <span class="fu">select</span>(Run) <span class="sc">|&gt;</span>  <span class="fu">unique</span>() <span class="sc">|&gt;</span> <span class="fu">nrow</span>()</span>
<span id="cb5-364"><a href="#cb5-364" aria-hidden="true" tabindex="-1"></a><span class="do">### [1] 2558</span></span>
<span id="cb5-365"><a href="#cb5-365" aria-hidden="true" tabindex="-1"></a><span class="do">### Sure.</span></span>
<span id="cb5-366"><a href="#cb5-366" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-367"><a href="#cb5-367" aria-hidden="true" tabindex="-1"></a>cohort_data   <span class="ot">&lt;-</span> matched_CPTAC_after_processing <span class="sc">|&gt;</span> </span>
<span id="cb5-368"><a href="#cb5-368" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">select</span>(Run, cohort)   <span class="sc">|&gt;</span></span>
<span id="cb5-369"><a href="#cb5-369" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">unique</span>()   </span>
<span id="cb5-370"><a href="#cb5-370" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-371"><a href="#cb5-371" aria-hidden="true" tabindex="-1"></a>matched_can <span class="ot">&lt;-</span> matched_can <span class="sc">|&gt;</span></span>
<span id="cb5-372"><a href="#cb5-372" aria-hidden="true" tabindex="-1"></a>               <span class="fu">mutate</span>(<span class="at">Run =</span> ID)  <span class="sc">|&gt;</span></span>
<span id="cb5-373"><a href="#cb5-373" aria-hidden="true" tabindex="-1"></a>               <span class="fu">left_join</span>(cohort_data, <span class="at">by =</span> <span class="st">"Run"</span>)  <span class="sc">|&gt;</span></span>
<span id="cb5-374"><a href="#cb5-374" aria-hidden="true" tabindex="-1"></a>               <span class="fu">select</span>(<span class="sc">-</span>Run)</span>
<span id="cb5-375"><a href="#cb5-375" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-376"><a href="#cb5-376" aria-hidden="true" tabindex="-1"></a>cohorts_IOBR <span class="ot">&lt;-</span> <span class="fu">unique</span>(matched_can<span class="sc">$</span>cohort)</span>
<span id="cb5-377"><a href="#cb5-377" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-378"><a href="#cb5-378" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (cohort1 <span class="cf">in</span> cohorts_IOBR) {</span>
<span id="cb5-379"><a href="#cb5-379" aria-hidden="true" tabindex="-1"></a>  cohort_data <span class="ot">&lt;-</span> <span class="fu">subset</span>(matched_can, cohort <span class="sc">==</span> cohort1) <span class="sc">|&gt;</span> <span class="fu">select</span>(<span class="sc">-</span>cohort)</span>
<span id="cb5-380"><a href="#cb5-380" aria-hidden="true" tabindex="-1"></a>  file_name <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"/home/data2/Projects/Fusion/fusiondb_v3/IOBR/"</span>, cohort1 ,<span class="st">"_IOBR_signature.tsv.gz"</span>)</span>
<span id="cb5-381"><a href="#cb5-381" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(cohort_data, </span>
<span id="cb5-382"><a href="#cb5-382" aria-hidden="true" tabindex="-1"></a>              <span class="at">file =</span> <span class="fu">gzfile</span>(file_name), </span>
<span id="cb5-383"><a href="#cb5-383" aria-hidden="true" tabindex="-1"></a>              <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, </span>
<span id="cb5-384"><a href="#cb5-384" aria-hidden="true" tabindex="-1"></a>              <span class="at">row.names =</span> <span class="cn">FALSE</span>, </span>
<span id="cb5-385"><a href="#cb5-385" aria-hidden="true" tabindex="-1"></a>              <span class="at">quote =</span> <span class="cn">FALSE</span>) </span>
<span id="cb5-386"><a href="#cb5-386" aria-hidden="true" tabindex="-1"></a>}  </span>
<span id="cb5-387"><a href="#cb5-387" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb5-388"><a href="#cb5-388" aria-hidden="true" tabindex="-1"></a><span class="do">## Gene</span></span>
<span id="cb5-389"><a href="#cb5-389" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-390"><a href="#cb5-390" aria-hidden="true" tabindex="-1"></a>metafusion1 <span class="ot">&lt;-</span> metafusion <span class="sc">|&gt;</span> </span>
<span id="cb5-391"><a href="#cb5-391" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb5-392"><a href="#cb5-392" aria-hidden="true" tabindex="-1"></a>    <span class="at">cohort =</span> <span class="fu">case_when</span>(</span>
<span id="cb5-393"><a href="#cb5-393" aria-hidden="true" tabindex="-1"></a>      Sample_ID <span class="sc">%in%</span> matched_CPTAC_after_processing<span class="sc">$</span>Run  <span class="sc">&amp;</span> <span class="fu">grepl</span>(<span class="st">"CPTAC"</span>, cohort) <span class="sc">~</span> matched_CPTAC_after_processing<span class="sc">$</span>cohort[<span class="fu">match</span>(Sample_ID, matched_CPTAC_after_processing<span class="sc">$</span>Run)],</span>
<span id="cb5-394"><a href="#cb5-394" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> cohort</span>
<span id="cb5-395"><a href="#cb5-395" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb5-396"><a href="#cb5-396" aria-hidden="true" tabindex="-1"></a>  )  </span>
<span id="cb5-397"><a href="#cb5-397" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-398"><a href="#cb5-398" aria-hidden="true" tabindex="-1"></a><span class="do">### In the past, the metafusion file had an issue with the sample_type being incorrectly labeled.</span></span>
<span id="cb5-399"><a href="#cb5-399" aria-hidden="true" tabindex="-1"></a><span class="do">### After further investigation, it was found that only the composite Sample_ID in CPTAC had this issue, where samples labeled as 'Normal' were mistakenly identified as 'Tumor' in the sample_type column.</span></span>
<span id="cb5-400"><a href="#cb5-400" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-401"><a href="#cb5-401" aria-hidden="true" tabindex="-1"></a>normal_id <span class="ot">&lt;-</span>   matched_CPTAC_after_processing <span class="sc">|&gt;</span> </span>
<span id="cb5-402"><a href="#cb5-402" aria-hidden="true" tabindex="-1"></a>               <span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"CPTAC-NORMAL"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb5-403"><a href="#cb5-403" aria-hidden="true" tabindex="-1"></a>               <span class="fu">select</span>(Run) <span class="sc">|&gt;</span></span>
<span id="cb5-404"><a href="#cb5-404" aria-hidden="true" tabindex="-1"></a>               <span class="fu">unique</span>() <span class="sc">|&gt;</span></span>
<span id="cb5-405"><a href="#cb5-405" aria-hidden="true" tabindex="-1"></a>               <span class="fu">pull</span>(Run)</span>
<span id="cb5-406"><a href="#cb5-406" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-407"><a href="#cb5-407" aria-hidden="true" tabindex="-1"></a>meta_normal <span class="ot">&lt;-</span> metafusion1 <span class="sc">|&gt;</span></span>
<span id="cb5-408"><a href="#cb5-408" aria-hidden="true" tabindex="-1"></a>               <span class="fu">filter</span>(Sample_ID <span class="sc">%in%</span> normal_id)</span>
<span id="cb5-409"><a href="#cb5-409" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-410"><a href="#cb5-410" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(meta_normal<span class="sc">$</span>sample_type)</span>
<span id="cb5-411"><a href="#cb5-411" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] "Normal" "Tumor" </span></span>
<span id="cb5-412"><a href="#cb5-412" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-413"><a href="#cb5-413" aria-hidden="true" tabindex="-1"></a>meta_processing  <span class="ot">&lt;-</span> metafusion1 <span class="sc">|&gt;</span></span>
<span id="cb5-414"><a href="#cb5-414" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">mutate</span>(</span>
<span id="cb5-415"><a href="#cb5-415" aria-hidden="true" tabindex="-1"></a>                      <span class="at">sample_type =</span> <span class="fu">ifelse</span>(</span>
<span id="cb5-416"><a href="#cb5-416" aria-hidden="true" tabindex="-1"></a>                        sample_type <span class="sc">!=</span> <span class="st">"Normal"</span> <span class="sc">&amp;</span> Sample_ID <span class="sc">%in%</span> normal_id,</span>
<span id="cb5-417"><a href="#cb5-417" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Normal"</span>,</span>
<span id="cb5-418"><a href="#cb5-418" aria-hidden="true" tabindex="-1"></a>                        sample_type</span>
<span id="cb5-419"><a href="#cb5-419" aria-hidden="true" tabindex="-1"></a>                      )</span>
<span id="cb5-420"><a href="#cb5-420" aria-hidden="true" tabindex="-1"></a>                    )</span>
<span id="cb5-421"><a href="#cb5-421" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-422"><a href="#cb5-422" aria-hidden="true" tabindex="-1"></a>cp_file_name1 <span class="ot">&lt;-</span> <span class="st">"/home/data2/Projects/Fusion/fusiondb_v3/metafusion_calls_matched.cptac.tsv"</span></span>
<span id="cb5-423"><a href="#cb5-423" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(meta_processing, <span class="at">file =</span> cp_file_name1, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="tcga-1" class="level3">
<h3 class="anchored" data-anchor-id="tcga-1">TCGA</h3>
<div class="sourceCode" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>file_meta <span class="ot">&lt;-</span> <span class="st">"/home/data2/Projects/Fusion/fusiondb_v3/metafusion_calls_matched.cptac.target.tsv"</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>metafusion <span class="ot">&lt;-</span> <span class="fu">fread</span>(file_meta  , <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="co"># TCGA</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a><span class="do">## info</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>TCGA_info <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/home/data2/Projects/Fusion/fusiondb/Clininfo/TCGA_info.tsv"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="do">### TCGA_sheet pretreatment</span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>TCGA_sheet <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/home/data2/Projects/Fusion/unclean/TCGA_sample_sheet.2024-05-14.tsv"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>              <span class="fu">select</span>(<span class="fu">c</span>(<span class="st">`</span><span class="at">Sample ID</span><span class="st">`</span>, <span class="st">`</span><span class="at">Sample Type</span><span class="st">`</span>,<span class="st">`</span><span class="at">Project ID</span><span class="st">`</span>)) </span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(TCGA_sheet) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Sample_ID"</span>, <span class="st">"Sample_type"</span>,<span class="st">"Project_id"</span>)</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>TCGA_sheet_pretreatment <span class="ot">&lt;-</span> TCGA_sheet  <span class="sc">|&gt;</span></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">mutate</span>(<span class="at">Sample_ID =</span> <span class="fu">substr</span>(Sample_ID, <span class="dv">1</span>, <span class="dv">15</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">unique</span>()</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a><span class="do">### We use our naming standard mutate a new column cohort. </span></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>TCGA_after_processing <span class="ot">&lt;-</span> TCGA_sheet_pretreatment <span class="sc">|&gt;</span></span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(TCGA_info, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Sample_ID"</span> <span class="ot">=</span> <span class="st">"Sample_ID"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">cohort =</span> Project_id,</span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">Run =</span> Sample_ID,</span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">Patient_ID =</span> <span class="fu">substr</span>(Sample_ID, <span class="dv">1</span>, <span class="dv">12</span>),  </span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a><span class="do">### TCGA Sample type</span></span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a><span class="do">### TCGA_after_processing$Sample_type |&gt; unique()</span></span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a><span class="do">###   [1] "Primary Tumor"                                   "Solid Tissue Normal"                             "Metastatic"                                     </span></span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a><span class="do">###   [4] "Additional - New Primary"                        "Primary Blood Derived Cancer - Peripheral Blood" "Additional Metastatic"                          </span></span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a><span class="do">###   [7] "Recurrent Tumor"   </span></span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a><span class="do">### </span></span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a><span class="do">### "Solid Tissue Normal" is normal sample, others is tumor sample</span></span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true" tabindex="-1"></a>TCGA_NORMAL <span class="ot">&lt;-</span> TCGA_after_processing <span class="sc">|&gt;</span> </span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true" tabindex="-1"></a>               <span class="fu">filter</span>(Sample_type <span class="sc">==</span> <span class="st">"Solid Tissue Normal"</span>) </span>
<span id="cb6-38"><a href="#cb6-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-39"><a href="#cb6-39" aria-hidden="true" tabindex="-1"></a>NORMAL_id <span class="ot">&lt;-</span> TCGA_NORMAL<span class="sc">$</span>Run <span class="sc">|&gt;</span> <span class="fu">unique</span>()</span>
<span id="cb6-40"><a href="#cb6-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-41"><a href="#cb6-41" aria-hidden="true" tabindex="-1"></a>TCGA_after_processing <span class="ot">&lt;-</span> TCGA_after_processing <span class="sc">|&gt;</span> </span>
<span id="cb6-42"><a href="#cb6-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb6-43"><a href="#cb6-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">Project_id =</span> cohort,</span>
<span id="cb6-44"><a href="#cb6-44" aria-hidden="true" tabindex="-1"></a>    <span class="at">cohort =</span> <span class="fu">case_when</span>(</span>
<span id="cb6-45"><a href="#cb6-45" aria-hidden="true" tabindex="-1"></a>      Run <span class="sc">%in%</span> NORMAL_id <span class="sc">~</span> <span class="st">"TCGA-NORMAL"</span>,</span>
<span id="cb6-46"><a href="#cb6-46" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> cohort)) </span>
<span id="cb6-47"><a href="#cb6-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-48"><a href="#cb6-48" aria-hidden="true" tabindex="-1"></a><span class="do">### make sure nrow is same between Before and After Processing. </span></span>
<span id="cb6-49"><a href="#cb6-49" aria-hidden="true" tabindex="-1"></a>TCGA_after_processing <span class="sc">|&gt;</span> <span class="fu">nrow</span>()</span>
<span id="cb6-50"><a href="#cb6-50" aria-hidden="true" tabindex="-1"></a><span class="do">### [1] 10545</span></span>
<span id="cb6-51"><a href="#cb6-51" aria-hidden="true" tabindex="-1"></a>TCGA_sheet_pretreatment <span class="sc">|&gt;</span> <span class="fu">nrow</span>() </span>
<span id="cb6-52"><a href="#cb6-52" aria-hidden="true" tabindex="-1"></a><span class="do">### [1] 10545</span></span>
<span id="cb6-53"><a href="#cb6-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-54"><a href="#cb6-54" aria-hidden="true" tabindex="-1"></a><span class="do">### make sure IOBR/TME data just contain sample which tcga_sheet gave. </span></span>
<span id="cb6-55"><a href="#cb6-55" aria-hidden="true" tabindex="-1"></a>detected_id_TCGA <span class="ot">&lt;-</span> TCGA_after_processing<span class="sc">$</span>Run <span class="sc">|&gt;</span> <span class="fu">unique</span>()</span>
<span id="cb6-56"><a href="#cb6-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-57"><a href="#cb6-57" aria-hidden="true" tabindex="-1"></a><span class="do">### save info</span></span>
<span id="cb6-58"><a href="#cb6-58" aria-hidden="true" tabindex="-1"></a>cohorts <span class="ot">&lt;-</span> <span class="fu">unique</span>(TCGA_after_processing<span class="sc">$</span>cohort)</span>
<span id="cb6-59"><a href="#cb6-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-60"><a href="#cb6-60" aria-hidden="true" tabindex="-1"></a>file_name1 <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"/home/data2/Projects/Fusion/fusiondb_v3/Clininfo/"</span>)</span>
<span id="cb6-61"><a href="#cb6-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-62"><a href="#cb6-62" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (cohort1 <span class="cf">in</span> cohorts) {</span>
<span id="cb6-63"><a href="#cb6-63" aria-hidden="true" tabindex="-1"></a>  cohort_data <span class="ot">&lt;-</span> <span class="fu">subset</span>(TCGA_after_processing, cohort <span class="sc">==</span> cohort1)</span>
<span id="cb6-64"><a href="#cb6-64" aria-hidden="true" tabindex="-1"></a>  name <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"-"</span>, <span class="st">"_"</span>, cohort1)</span>
<span id="cb6-65"><a href="#cb6-65" aria-hidden="true" tabindex="-1"></a>  file_name <span class="ot">&lt;-</span> <span class="fu">paste0</span>(file_name1, name ,<span class="st">"_info.tsv"</span>)</span>
<span id="cb6-66"><a href="#cb6-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fwrite</span>(cohort_data , <span class="at">file =</span> file_name, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)  </span>
<span id="cb6-67"><a href="#cb6-67" aria-hidden="true" tabindex="-1"></a>}  </span>
<span id="cb6-68"><a href="#cb6-68" aria-hidden="true" tabindex="-1"></a><span class="do">## Gene</span></span>
<span id="cb6-69"><a href="#cb6-69" aria-hidden="true" tabindex="-1"></a>metafusion1 <span class="ot">&lt;-</span> metafusion <span class="sc">|&gt;</span> </span>
<span id="cb6-70"><a href="#cb6-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb6-71"><a href="#cb6-71" aria-hidden="true" tabindex="-1"></a>    <span class="at">cohort =</span> <span class="fu">case_when</span>(</span>
<span id="cb6-72"><a href="#cb6-72" aria-hidden="true" tabindex="-1"></a>      Sample_ID <span class="sc">%in%</span> TCGA_after_processing<span class="sc">$</span>Run <span class="sc">&amp;</span> <span class="fu">grepl</span>(<span class="st">"TCGA"</span>, cohort) <span class="sc">~</span> TCGA_after_processing<span class="sc">$</span>cohort[<span class="fu">match</span>(Sample_ID, TCGA_after_processing<span class="sc">$</span>Run)],</span>
<span id="cb6-73"><a href="#cb6-73" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> cohort</span>
<span id="cb6-74"><a href="#cb6-74" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb6-75"><a href="#cb6-75" aria-hidden="true" tabindex="-1"></a>  ) </span>
<span id="cb6-76"><a href="#cb6-76" aria-hidden="true" tabindex="-1"></a>cp_file_name1 <span class="ot">&lt;-</span> <span class="st">"/home/data2/Projects/Fusion/fusiondb_v3/metafusion_gdc_divide.tsv"</span></span>
<span id="cb6-77"><a href="#cb6-77" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(metafusion1, <span class="at">file =</span> cp_file_name1, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>) </span>
<span id="cb6-78"><a href="#cb6-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-79"><a href="#cb6-79" aria-hidden="true" tabindex="-1"></a><span class="do">## IOBR</span></span>
<span id="cb6-80"><a href="#cb6-80" aria-hidden="true" tabindex="-1"></a>folder_path <span class="ot">&lt;-</span> <span class="st">"/home/data2/Projects/Fusion/fusiondb/IOBR"</span></span>
<span id="cb6-81"><a href="#cb6-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-82"><a href="#cb6-82" aria-hidden="true" tabindex="-1"></a>file_pattern <span class="ot">&lt;-</span> <span class="st">"^TCGA.*_IOBR_signature.tsv.gz$"</span></span>
<span id="cb6-83"><a href="#cb6-83" aria-hidden="true" tabindex="-1"></a>file_list <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> folder_path, <span class="at">pattern =</span> file_pattern, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb6-84"><a href="#cb6-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-85"><a href="#cb6-85" aria-hidden="true" tabindex="-1"></a>combined_data <span class="ot">&lt;-</span> <span class="fu">data.table</span>()</span>
<span id="cb6-86"><a href="#cb6-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-87"><a href="#cb6-87" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (file <span class="cf">in</span> file_list) {</span>
<span id="cb6-88"><a href="#cb6-88" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-89"><a href="#cb6-89" aria-hidden="true" tabindex="-1"></a>  file_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(file, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb6-90"><a href="#cb6-90" aria-hidden="true" tabindex="-1"></a>  file_name <span class="ot">&lt;-</span> <span class="fu">basename</span>(file)  </span>
<span id="cb6-91"><a href="#cb6-91" aria-hidden="true" tabindex="-1"></a>  project_id <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"^(TCGA-[^_]+).*"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, file_name)  </span>
<span id="cb6-92"><a href="#cb6-92" aria-hidden="true" tabindex="-1"></a>  file_data <span class="ot">&lt;-</span> file_data <span class="sc">%&gt;%</span></span>
<span id="cb6-93"><a href="#cb6-93" aria-hidden="true" tabindex="-1"></a>               <span class="fu">mutate</span>(<span class="at">Project_id =</span> project_id)</span>
<span id="cb6-94"><a href="#cb6-94" aria-hidden="true" tabindex="-1"></a>  combined_data <span class="ot">&lt;-</span> <span class="fu">rbind</span>(combined_data, file_data)</span>
<span id="cb6-95"><a href="#cb6-95" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-96"><a href="#cb6-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-97"><a href="#cb6-97" aria-hidden="true" tabindex="-1"></a><span class="do">### Filter TCGA Replicate Samples</span></span>
<span id="cb6-98"><a href="#cb6-98" aria-hidden="true" tabindex="-1"></a><span class="do">### The filter rule following broad institute says:</span></span>
<span id="cb6-99"><a href="#cb6-99" aria-hidden="true" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb6-100"><a href="#cb6-100" aria-hidden="true" tabindex="-1"></a><span class="do">### In many instances there is more than one aliquot for a given combination of individual, platform, and data type. However, only one aliquot may be ingested into Firehose. Therefore, a set of precedence rules are applied to select the most scientifically advantageous one among them. Two filters are applied to achieve this aim: an Analyte Replicate Filter and a Sort Replicate Filter.</span></span>
<span id="cb6-101"><a href="#cb6-101" aria-hidden="true" tabindex="-1"></a><span class="do">### </span></span>
<span id="cb6-102"><a href="#cb6-102" aria-hidden="true" tabindex="-1"></a><span class="do">### Analyte Replicate Filter</span></span>
<span id="cb6-103"><a href="#cb6-103" aria-hidden="true" tabindex="-1"></a><span class="do">### The following precedence rules are applied when the aliquots have differing analytes. For RNA aliquots, T analytes are dropped in preference to H and R analytes, since T is the inferior extraction protocol. If H and R are encountered, H is the chosen analyte. This is somewhat arbitrary and subject to change, since it is not clear at present whether H or R is the better protocol. If there are multiple aliquots associated with the chosen RNA analyte, the aliquot with the later plate number is chosen. For DNA aliquots, D analytes (native DNA) are preferred over G, W, or X (whole-genome amplified) analytes, unless the G, W, or X analyte sample has a higher plate number.</span></span>
<span id="cb6-104"><a href="#cb6-104" aria-hidden="true" tabindex="-1"></a><span class="do">### </span></span>
<span id="cb6-105"><a href="#cb6-105" aria-hidden="true" tabindex="-1"></a><span class="do">### Sort Replicate Filter</span></span>
<span id="cb6-106"><a href="#cb6-106" aria-hidden="true" tabindex="-1"></a><span class="do">### The following precedence rules are applied when the analyte filter still produces more than one sample. The sort filter chooses the aliquot with the highest lexicographical sort value, to ensure that the barcode with the highest portion and/or plate number is selected when all other barcode fields are identical.</span></span>
<span id="cb6-107"><a href="#cb6-107" aria-hidden="true" tabindex="-1"></a><span class="do">### </span></span>
<span id="cb6-108"><a href="#cb6-108" aria-hidden="true" tabindex="-1"></a><span class="do">### Ref Link: &lt;https://confluence.broadinstitute.org/display/GDAC/FAQ#FAQ-sampleTypesQWhatTCGAsampletypesareFirehosepipelinesexecutedupon&gt;</span></span>
<span id="cb6-109"><a href="#cb6-109" aria-hidden="true" tabindex="-1"></a><span class="do">###           &lt;https://gist.github.com/ShixiangWang/33b2f9b49b77eaa8f773b428480f9101&gt;</span></span>
<span id="cb6-110"><a href="#cb6-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-111"><a href="#cb6-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-112"><a href="#cb6-112" aria-hidden="true" tabindex="-1"></a>result_data <span class="ot">&lt;-</span> combined_data <span class="sc">|&gt;</span></span>
<span id="cb6-113"><a href="#cb6-113" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb6-114"><a href="#cb6-114" aria-hidden="true" tabindex="-1"></a>    <span class="at">prefixes =</span> <span class="fu">str_split</span>(ID, <span class="st">"-"</span>, <span class="at">simplify =</span> <span class="cn">TRUE</span>)[, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>] <span class="sc">%&gt;%</span> <span class="fu">apply</span>(<span class="dv">1</span>, paste, <span class="at">collapse =</span> <span class="st">"-"</span>),</span>
<span id="cb6-115"><a href="#cb6-115" aria-hidden="true" tabindex="-1"></a>    <span class="at">suffixes =</span> <span class="fu">str_split</span>(ID, <span class="st">"-"</span>, <span class="at">simplify =</span> <span class="cn">TRUE</span>)[, <span class="dv">5</span>],</span>
<span id="cb6-116"><a href="#cb6-116" aria-hidden="true" tabindex="-1"></a>    <span class="at">second_chars =</span> <span class="fu">str_extract</span>(suffixes, <span class="st">"[0-9]+"</span>),</span>
<span id="cb6-117"><a href="#cb6-117" aria-hidden="true" tabindex="-1"></a>    <span class="at">second_chars =</span> <span class="fu">as.integer</span>(second_chars))</span>
<span id="cb6-118"><a href="#cb6-118" aria-hidden="true" tabindex="-1"></a>result_IOBR <span class="ot">&lt;-</span> result_data  <span class="sc">|&gt;</span></span>
<span id="cb6-119"><a href="#cb6-119" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(prefixes) <span class="sc">|&gt;</span></span>
<span id="cb6-120"><a href="#cb6-120" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice</span>(<span class="fu">which.max</span>(second_chars)) <span class="sc">|&gt;</span></span>
<span id="cb6-121"><a href="#cb6-121" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">|&gt;</span></span>
<span id="cb6-122"><a href="#cb6-122" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">cohort =</span> Project_id,</span>
<span id="cb6-123"><a href="#cb6-123" aria-hidden="true" tabindex="-1"></a>         <span class="at">ID =</span> <span class="fu">substr</span>(prefixes, <span class="dv">1</span>, <span class="dv">15</span>),</span>
<span id="cb6-124"><a href="#cb6-124" aria-hidden="true" tabindex="-1"></a>         <span class="at">cohort =</span> <span class="fu">case_when</span>(ID <span class="sc">%in%</span> NORMAL_id <span class="sc">~</span> <span class="st">"TCGA-NORMAL"</span>,</span>
<span id="cb6-125"><a href="#cb6-125" aria-hidden="true" tabindex="-1"></a>                            <span class="cn">TRUE</span> <span class="sc">~</span> cohort)</span>
<span id="cb6-126"><a href="#cb6-126" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb6-127"><a href="#cb6-127" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(ID <span class="sc">%in%</span> detected_id_TCGA)  <span class="sc">|&gt;</span> </span>
<span id="cb6-128"><a href="#cb6-128" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">"suffixes"</span>, <span class="st">"second_chars"</span>,<span class="st">"prefixes"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb6-129"><a href="#cb6-129" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>() </span>
<span id="cb6-130"><a href="#cb6-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-131"><a href="#cb6-131" aria-hidden="true" tabindex="-1"></a>cohorts_IOBR <span class="ot">&lt;-</span> <span class="fu">unique</span>(result_IOBR<span class="sc">$</span>cohort)</span>
<span id="cb6-132"><a href="#cb6-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-133"><a href="#cb6-133" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (cohort1 <span class="cf">in</span> cohorts_IOBR) {</span>
<span id="cb6-134"><a href="#cb6-134" aria-hidden="true" tabindex="-1"></a>  cohort_data <span class="ot">&lt;-</span> <span class="fu">subset</span>(result_IOBR, cohort <span class="sc">==</span> cohort1) <span class="sc">|&gt;</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(cohort,Project_id) )</span>
<span id="cb6-135"><a href="#cb6-135" aria-hidden="true" tabindex="-1"></a>  file_name <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"/home/data2/Projects/Fusion/fusiondb_v3/IOBR/"</span>, cohort1 ,<span class="st">"_IOBR_signature.tsv.gz"</span>)</span>
<span id="cb6-136"><a href="#cb6-136" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(cohort_data, </span>
<span id="cb6-137"><a href="#cb6-137" aria-hidden="true" tabindex="-1"></a>              <span class="at">file =</span> <span class="fu">gzfile</span>(file_name), </span>
<span id="cb6-138"><a href="#cb6-138" aria-hidden="true" tabindex="-1"></a>              <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, </span>
<span id="cb6-139"><a href="#cb6-139" aria-hidden="true" tabindex="-1"></a>              <span class="at">row.names =</span> <span class="cn">FALSE</span>, </span>
<span id="cb6-140"><a href="#cb6-140" aria-hidden="true" tabindex="-1"></a>              <span class="at">quote =</span> <span class="cn">FALSE</span>) </span>
<span id="cb6-141"><a href="#cb6-141" aria-hidden="true" tabindex="-1"></a>}  </span>
<span id="cb6-142"><a href="#cb6-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-143"><a href="#cb6-143" aria-hidden="true" tabindex="-1"></a><span class="do">## TME</span></span>
<span id="cb6-144"><a href="#cb6-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-145"><a href="#cb6-145" aria-hidden="true" tabindex="-1"></a>folder_path <span class="ot">&lt;-</span> <span class="st">"/home/data2/Projects/Fusion/fusiondb/IOBR"</span></span>
<span id="cb6-146"><a href="#cb6-146" aria-hidden="true" tabindex="-1"></a>file_pattern <span class="ot">&lt;-</span> <span class="st">"^TCGA.*_TME.tsv.gz$"</span></span>
<span id="cb6-147"><a href="#cb6-147" aria-hidden="true" tabindex="-1"></a>file_list <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> folder_path, <span class="at">pattern =</span> file_pattern, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb6-148"><a href="#cb6-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-149"><a href="#cb6-149" aria-hidden="true" tabindex="-1"></a>combined_data1 <span class="ot">&lt;-</span> <span class="fu">data.table</span>()</span>
<span id="cb6-150"><a href="#cb6-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-151"><a href="#cb6-151" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (file <span class="cf">in</span> file_list) {</span>
<span id="cb6-152"><a href="#cb6-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-153"><a href="#cb6-153" aria-hidden="true" tabindex="-1"></a>  file_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(file, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb6-154"><a href="#cb6-154" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-155"><a href="#cb6-155" aria-hidden="true" tabindex="-1"></a>  file_name <span class="ot">&lt;-</span> <span class="fu">basename</span>(file)  </span>
<span id="cb6-156"><a href="#cb6-156" aria-hidden="true" tabindex="-1"></a>  project_id <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"^(TCGA-[^_]+).*"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, file_name)  </span>
<span id="cb6-157"><a href="#cb6-157" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-158"><a href="#cb6-158" aria-hidden="true" tabindex="-1"></a>  file_data <span class="ot">&lt;-</span> file_data <span class="sc">%&gt;%</span></span>
<span id="cb6-159"><a href="#cb6-159" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Project_id =</span> project_id)</span>
<span id="cb6-160"><a href="#cb6-160" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-161"><a href="#cb6-161" aria-hidden="true" tabindex="-1"></a>  combined_data1 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(combined_data1, file_data)</span>
<span id="cb6-162"><a href="#cb6-162" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-163"><a href="#cb6-163" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-164"><a href="#cb6-164" aria-hidden="true" tabindex="-1"></a>result_data1 <span class="ot">&lt;-</span> combined_data1 <span class="sc">|&gt;</span></span>
<span id="cb6-165"><a href="#cb6-165" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb6-166"><a href="#cb6-166" aria-hidden="true" tabindex="-1"></a>    <span class="at">prefixes =</span> <span class="fu">str_split</span>(ID, <span class="st">"-"</span>, <span class="at">simplify =</span> <span class="cn">TRUE</span>)[, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>] <span class="sc">%&gt;%</span> <span class="fu">apply</span>(<span class="dv">1</span>, paste, <span class="at">collapse =</span> <span class="st">"-"</span>),</span>
<span id="cb6-167"><a href="#cb6-167" aria-hidden="true" tabindex="-1"></a>    <span class="at">suffixes =</span> <span class="fu">str_split</span>(ID, <span class="st">"-"</span>, <span class="at">simplify =</span> <span class="cn">TRUE</span>)[, <span class="dv">5</span>],</span>
<span id="cb6-168"><a href="#cb6-168" aria-hidden="true" tabindex="-1"></a>    <span class="at">second_chars =</span> <span class="fu">str_extract</span>(suffixes, <span class="st">"[0-9]+"</span>),</span>
<span id="cb6-169"><a href="#cb6-169" aria-hidden="true" tabindex="-1"></a>    <span class="at">second_chars =</span> <span class="fu">as.integer</span>(second_chars))</span>
<span id="cb6-170"><a href="#cb6-170" aria-hidden="true" tabindex="-1"></a>result_TME <span class="ot">&lt;-</span> result_data1  <span class="sc">|&gt;</span></span>
<span id="cb6-171"><a href="#cb6-171" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(prefixes) <span class="sc">|&gt;</span></span>
<span id="cb6-172"><a href="#cb6-172" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice</span>(<span class="fu">which.max</span>(second_chars)) <span class="sc">|&gt;</span></span>
<span id="cb6-173"><a href="#cb6-173" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>() <span class="sc">|&gt;</span></span>
<span id="cb6-174"><a href="#cb6-174" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">cohort =</span> Project_id,</span>
<span id="cb6-175"><a href="#cb6-175" aria-hidden="true" tabindex="-1"></a>             <span class="at">ID =</span> <span class="fu">substr</span>(prefixes, <span class="dv">1</span>, <span class="dv">15</span>),</span>
<span id="cb6-176"><a href="#cb6-176" aria-hidden="true" tabindex="-1"></a>             <span class="at">cohort =</span> <span class="fu">case_when</span>(ID <span class="sc">%in%</span> NORMAL_id <span class="sc">~</span> <span class="st">"TCGA-NORMAL"</span>,</span>
<span id="cb6-177"><a href="#cb6-177" aria-hidden="true" tabindex="-1"></a>                                <span class="cn">TRUE</span> <span class="sc">~</span> cohort)</span>
<span id="cb6-178"><a href="#cb6-178" aria-hidden="true" tabindex="-1"></a>      ) <span class="sc">|&gt;</span></span>
<span id="cb6-179"><a href="#cb6-179" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(ID <span class="sc">%in%</span> detected_id_TCGA)  <span class="sc">|&gt;</span> </span>
<span id="cb6-180"><a href="#cb6-180" aria-hidden="true" tabindex="-1"></a>      <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">"suffixes"</span>, <span class="st">"second_chars"</span>,<span class="st">"prefixes"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb6-181"><a href="#cb6-181" aria-hidden="true" tabindex="-1"></a>      <span class="fu">unique</span>() </span>
<span id="cb6-182"><a href="#cb6-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-183"><a href="#cb6-183" aria-hidden="true" tabindex="-1"></a>cohorts_TME <span class="ot">&lt;-</span> <span class="fu">unique</span>(result_TME<span class="sc">$</span>cohort)</span>
<span id="cb6-184"><a href="#cb6-184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-185"><a href="#cb6-185" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (cohort1 <span class="cf">in</span> cohorts_TME) {</span>
<span id="cb6-186"><a href="#cb6-186" aria-hidden="true" tabindex="-1"></a>  cohort_data <span class="ot">&lt;-</span> <span class="fu">subset</span>(result_TME, cohort <span class="sc">==</span> cohort1) <span class="sc">|&gt;</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(cohort,Project_id) )</span>
<span id="cb6-187"><a href="#cb6-187" aria-hidden="true" tabindex="-1"></a>  file_name <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"/home/data2/Projects/Fusion/fusiondb_v3/IOBR/"</span>, cohort1 ,<span class="st">"_TME.tsv.gz"</span>)</span>
<span id="cb6-188"><a href="#cb6-188" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(cohort_data, </span>
<span id="cb6-189"><a href="#cb6-189" aria-hidden="true" tabindex="-1"></a>              <span class="at">file =</span> <span class="fu">gzfile</span>(file_name), </span>
<span id="cb6-190"><a href="#cb6-190" aria-hidden="true" tabindex="-1"></a>              <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, </span>
<span id="cb6-191"><a href="#cb6-191" aria-hidden="true" tabindex="-1"></a>              <span class="at">row.names =</span> <span class="cn">FALSE</span>, </span>
<span id="cb6-192"><a href="#cb6-192" aria-hidden="true" tabindex="-1"></a>              <span class="at">quote =</span> <span class="cn">FALSE</span>) </span>
<span id="cb6-193"><a href="#cb6-193" aria-hidden="true" tabindex="-1"></a>} </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/OncoHarmony-Network\.github\.io\/doc-immunofusion\/");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>